---
title: "Microeuk-survey"
author: "Sarah Hu"
date: "12/17/2020"
output: 
  html_document:
    number_sections: true
    theme: spacelab
    highlight: monochrome
    collapsed: false
    toc: true
    toc_depth: 3
    toc_float: true
    code_folding: hide
  pdf_document:
    toc: true
    toc_depth: '3'
knit: (function(input_file, encoding) {
  out_dir <- 'docs';
  rmarkdown::render(input_file,
 encoding=encoding,
 output_file=file.path(dirname(input_file), out_dir, 'index.html'))})
editor_options: 
  markdown: 
    wrap: 72
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, tidy = FALSE, message = FALSE)
```

```{r Library loading initial, echo=FALSE, message=FALSE}
library(tidyverse); library(cowplot)
library(phyloseq); library(decontam)
library(compositions); library(patchwork); 
library(ggupset); library(gt)
library(plotly); library(viridis); library(vegan)
```

Preliminary analyses for microeukaryote tag-sequence survey.

*Description*: Investigate the diversity of single-celled microbial
eukaryotic communities across several deep-sea hydrothermal vent sites
(including NE Pacific, Caribbean). We plan to address questions related
to the environmental factors that shape protistan community dynamics,
and determine if patterns in species diversity and distribution vary at
different deep-sea habitats. These questions will be addressed using
similarly generated metabarcoding data from several distinct
hydrothermal vents. Along with characterizing community structure, we
plan to evaluate interactions between protist species (to identify
putative predator-prey or parasite-host relationships) and their
environment (to explore their relationship to geochemical properties).

*Questions to address*

> What is the general biogeography and distribution of the deep-sea
> hydrothermal vent microbial eukaryotic community?

> What community structure features (i.e., species richness, proportion
> cosmopolitan versus endemic, species evenness) are shared across or
> unique to deep-sea hydrothermal vent sites?

> What environmental features (i.e., temperature, geochemistry)
> influence microbial eukaryotic community diversity? Can we identify if
> certain environmental factors select for putative vent endemics?

# Environmental characterization of all vent sites

Three sites in separate ocean regions.

```{r Import metadata for environmental measurements}
metadata <- read.delim("data-input/samplelist-metadata.txt", na.strings = "")
# View(metadata)
# ?read.delim()
# colnames(metadata)
```

Filter to environmental data only.

```{r}
env_deepsea <- metadata %>% 
  mutate_all(as.character) %>%
  filter(Sample_or_Control == "Sample") %>% 
  filter(!(SAMPLETYPE == "Incubation")) %>% 
  filter(!(SAMPLETYPE == "Microcolonizer")) %>% 
  select(VENT, COORDINATES, SITE, SAMPLEID, DEPTH, SAMPLETYPE, YEAR, TEMP = starts_with("TEMP"), pH, PercSeawater, Mg = starts_with("Mg"), H2 = starts_with("H2."), H2S = starts_with("H2S"), CH4 = starts_with("CH4"), ProkConc) %>%  
  pivot_longer(cols = TEMP:ProkConc, names_to = "VARIABLE", values_to = "VALUE", values_drop_na = FALSE) %>% 
  distinct() %>% 
  group_by(VENT, COORDINATES, SITE, DEPTH, SAMPLETYPE, YEAR, VARIABLE) 

# head(env_deepsea)
# unique(env_deepsea$VARIABLE)
# str(env_deepsea)
```

Units for the variables are as follows:
- Temp = Celsius
- Percent Seawater = %
- Mg = mmol/kg (or mM)
- H2 = µmol/L (or µM)
- H2S = mmol/L (or mM)
- CH4 = µmol/kg (or mM)
- ProkConc = cells/ml

_last updated Dec 29, 2021_ 
* Needs updated H2 numbers for MCR from Jeff
* All of this chemistry needs to be double checked again... 


## Visualize geochemistry

```{r, fig.height=9, fig.width=10}
rm <- c("-", "", "nd", "bd", NA)

geochem_1 <- env_deepsea %>% 
  filter(VARIABLE != "ProkConc") %>% 
  filter(!(VALUE %in% rm)) %>% 
  mutate(VALUE = as.numeric(as.character(VALUE))) %>% 
  ggplot(aes(x = SAMPLETYPE, y = VALUE, fill = SITE, shape = SAMPLETYPE))+
  geom_jitter(size = 3) +
  facet_wrap(VARIABLE ~ ., scales = "free") +
  scale_shape_manual(values = c(21, 23, 24)) +
  scale_fill_manual(values = c("#fdbb84", "#31a354", "#ef3b2c", "#02818a")) +
  guides(fill = guide_legend(override.aes = list(shape = 21) ),
              shape = guide_legend(override.aes = list(fill = "black"))) +
  theme_bw() +
    theme(axis.text = element_text(color="black", size=12),
          legend.title = element_blank(),
          axis.title = element_text(color="black", size=14),
          legend.text = element_text(color = "black", size = 14),
          plot.margin = margin(2, 1, 2, 1, "cm"),
          strip.background = element_blank()) +
  labs(x = "", y = "")
geochem_1


geochem_violin <- env_deepsea %>% 
  filter(VARIABLE != "ProkConc") %>% 
  filter(!(VALUE %in% rm)) %>% 
  mutate(VALUE = as.numeric(as.character(VALUE))) %>% 
  ggplot(aes(x = SAMPLETYPE, y = VALUE, fill = SITE, shape = SAMPLETYPE)) +
  geom_boxplot(alpha = 0.3, aes(group = SAMPLETYPE), fill = "grey", width = 0.3) +
  geom_jitter(size = 2, width = 0.2) +
  facet_wrap(VARIABLE ~ ., scales = "free") +
  scale_shape_manual(values = c(21, 23, 24)) +
  scale_fill_manual(values = c("#fdbb84", "#31a354", "#ef3b2c", "#02818a")) +
  guides(fill = guide_legend(override.aes = list(shape = 21) ),
              shape = guide_legend(override.aes = list(fill = "black"))) +
  theme_bw() +
    theme(axis.text = element_text(color="black", size=12),
          legend.title = element_blank(),
          axis.title = element_text(color="black", size=14),
          legend.text = element_text(color = "black", size = 14),
          plot.margin = margin(2, 1, 2, 1, "cm"),
          strip.background = element_blank()) +
  labs(x = "", y = "")
geochem_violin
```

```{r, fig.height=4, fig.width=5}
geochem_2 <- env_deepsea %>% 
  filter(VARIABLE == "ProkConc") %>% 
  filter(!(VALUE %in% rm)) %>% 
  mutate(VALUE = as.numeric(as.character(VALUE))) %>% 
  ggplot(aes(x = SAMPLETYPE, y = VALUE, fill = SITE, shape = SAMPLETYPE)) +
  geom_boxplot(alpha = 0.3, aes(group = SAMPLETYPE), fill = "grey", width = 0.3) +
  geom_jitter(size = 2, width = 0.2) +
  facet_wrap(VARIABLE ~ ., scales = "free") +
  scale_y_log10() +
  scale_shape_manual(values = c(21, 23, 24)) +
  scale_fill_manual(values = c("#fdbb84", "#31a354", "#ef3b2c", "#02818a")) +
  guides(fill = guide_legend(override.aes = list(shape = 21) ),
              shape = guide_legend(override.aes = list(fill = "black"))) +
  theme_bw() +
    theme(axis.text = element_text(color="black", size=12),
          legend.title = element_blank(),
          axis.title = element_text(color="black", size=14),
          legend.text = element_text(color = "black", size = 14),
          plot.margin = margin(2, 1, 2, 1, "cm"),
          strip.background = element_blank()) +
  labs(x = "", y = "")
geochem_2
```

### Set colors and symbols for dataset
```{r}
sampletype_order <- c("Background", "Plume", "Vent")
sampletype_symbol<- c(21, 23, 24)

site_order <- c("Axial", "GordaRidge", "Piccard", "VonDamm")
site_color <- c("#fdbb84", "#31a354", "#ef3b2c", "#02818a")
```


## **Table S1** Geochemistry values

Generate output table for all parameters measured

```{r}
# colnames(metadata)
geomchem_table <- metadata %>% 
  mutate_all(as.character) %>%
  filter(Sample_or_Control == "Sample") %>% 
  filter(!(SAMPLETYPE == "Incubation")) %>% 
  filter(!(SAMPLETYPE == "Microcolonizer")) %>% 
  select(SAMPLE, VENT, COORDINATES, SITE, SAMPLEID, DEPTH, SAMPLETYPE, YEAR, TEMP = starts_with("TEMP"), pH, PercSeawater, Mg = starts_with("Mg"), H2 = starts_with("H2."), H2S = starts_with("H2S"), CH4 = starts_with("CH4"), ProkConc)
geomchem_table
# write_delim(geomchem_table, file = "table-geochem-params.txt", delim = "\t")
```

# Import exisiting sequence data

Datasets included: - Gorda Ridge 2019 cruise - Axial Seamount time
series - 2013, 2014, & 2015 - Mid-Cayman Rise 2020 cruise

All data generated from extracted RNA, reverse transcribed to cDNA and
amplified with primers that target the V4 hypervariable region on the
18S rRNA gene.

Analysis done with QIIME2, kept 40-60% of the sequences through the QC
process and generated Amplicon Sequence Variants (ASVs) with DADA2.
Taxonomic assignment done with vsearch using the PR2 database (v4.14) at
80% identity. See the `seq-analysis` directory for QIIME2 code.

## Merged data from QIIME2 analysis

After determining ASVs for each sequence run, ASV tables were merged.

```{r Import taxonomy and ASV table, message=FALSE, eval=FALSE}
merged_tax <- read_delim("data-input/taxonomy.tsv", delim = "\t")
merged_asv <- read_delim("data-input/microeuk-merged-asv-table.tsv", delim = "\t", skip = 1)
# head(merged_tax)
```

### Import metadata & reformat

Still want to find more metadata. As of Dec 30, got as much geochem data available, but can add in more metadata for the MCR cruise. Need to confirm values from Jeff 


```{r}
metadata_formatted <- metadata %>% 
  mutate_all(as.character) %>%
  filter(Sample_or_Control == "Sample") %>% 
  filter(!(SAMPLETYPE == "Incubation")) %>% 
  filter(!(SAMPLETYPE == "Microcolonizer")) %>% 
  select(SAMPLE, VENT, COORDINATES, SITE, SAMPLEID, DEPTH, SAMPLETYPE, YEAR, TEMP = starts_with("TEMP"), pH, PercSeawater, Mg = starts_with("Mg"), H2 = starts_with("H2."), H2S = starts_with("H2S"), CH4 = starts_with("CH4"), ProkConc, Sample_or_Control)
```

Remove samples from Gorda Ridge microcolonizers and from the FLP
experiments (Gorda Ridge and Mid-Cayman Rise).

```{r Join taxonomy and ASV data, eval=FALSE}
asv_wtax <- merged_asv %>%
  select(FeatureID = '#OTU ID', everything()) %>%
  pivot_longer(cols = !FeatureID,
               names_to = "SAMPLE", values_to = "value") %>%
  left_join(merged_tax, by = c("FeatureID" = "Feature ID")) %>%
  left_join(metadata_formatted) %>%
  filter(!grepl("Siders_", SAMPLE)) %>% 
  filter(SAMPLETYPE != "Incubation") %>% 
  filter(SAMPLETYPE != "Microcolonizer") %>%
  mutate(DATASET = case_when(
    grepl("_GR_", SAMPLE) ~ "GR",
    grepl("Gorda", SAMPLE) ~ "GR",
    grepl("_MCR_", SAMPLE) ~ "MCR",
    grepl("Axial", SAMPLE) ~ "Axial",
  TRUE ~ "Control or blank")) %>%
    separate(Taxon, c("Domain", "Supergroup",
                  "Phylum", "Class", "Order",
                  "Family", "Genus", "Species"), sep = ";", remove = FALSE) %>% 
  unite(SAMPLENAME, SITE, SAMPLETYPE, YEAR, VENT, SAMPLEID, sep = " ", remove = FALSE)

# View(asv_wtax)
# head(asv_wtax) ## Complete ASV table with full taxonomy names and annotated sample information
```

### Total number of sequences

Barplots to show total number of sequences and total number of ASVs.

Total number of sequences and ASVs parallel each other. The Axial and
Gorda Ridge data were run on the same sequence run, with Mid-Cayman Rise
run on a separate MiSeq run - so the average number of sequences (and
ASVs) varies between these two runs. A few samples have too few
sequences, they will be removed below.

This newest version of PR2 has bacteria and archaea in it. Very, very
few were assigned to this. Majority assigned to eukaryotes.

```{r Plot total ASV and sequences, eval=FALSE, fig.height=11, fig.width=14}
# head(asv_wtax)
library(viridis)
plot_grid(
  # Total number of ASVs
  asv_wtax %>% 
  filter(value > 0) %>% 
  filter(Sample_or_Control == "Sample") %>% 
  ggplot(aes(x = SAMPLENAME)) +
  geom_bar(stat = "count", width = 0.9) +
  labs(y = "Total ASVs per sample", x = "") +
  coord_flip() +
    scale_y_continuous(position = "right") +
  theme_linedraw() +
  facet_grid(DATASET ~ ., scale = "free", space = "free") +
  theme(axis.text.x = element_text(angle = 0, hjust = 1, vjust = 1),
        axis.text.y = element_text(angle = 0, hjust = 1, vjust = 1),
        strip.background = element_blank(), strip.text = element_text(color = "black")),
  asv_wtax %>% 
    filter(Sample_or_Control == "Sample") %>%
  group_by(SAMPLENAME, SITE, Domain, DATASET) %>% 
  summarise(SUM_SEQ_DOMAIN = sum(value)) %>% 
  ggplot(aes(x = SAMPLENAME, y = SUM_SEQ_DOMAIN, fill = Domain)) +
  geom_bar(stat = "identity", color = "black", width = 0.9) +
  labs(y = "Total sequences per sample", x = "") +
  coord_flip() +
  viridis::scale_fill_viridis(discrete = TRUE) +
  scale_y_continuous(position = "right") +
  theme_linedraw() +
  facet_grid(DATASET ~ ., scale = "free", space = "free") +
  theme(axis.text.x = element_text(angle = 0, hjust = 1, vjust = 1),
        axis.text.y = element_text(angle = 0, hjust = 1, vjust = 1),
        strip.background = element_blank(), strip.text = element_text(color = "black"),
        legend.position = "right"),
  ncol = 2, align = c("hv"), axis = c("lr"))
```

### Table reporting total ASVs and sequences

```{r Total ASVs raw, eval=FALSE}
table_raw_stats <- asv_wtax %>% filter(value > 0) %>%
       group_by(SAMPLENAME, DATASET, SITE) %>%
       summarise(SEQ_SUM = sum(value),
                 ASV_COUNT = n()) %>% 
  ungroup() %>% 
  gt(
    groupname_col = c("DATASET", "SITE"),
    rowname_col = "SAMPLENAME"
  )
table_raw_stats
gtsave(table_raw_stats, filename = "seq_asv_count_nonQC.html", path = "output-tables/")
```

> After removing contaminate ASVs below, I will set threshold of 10,000
> sequences- if a sample has fewer than this, chuck it.

## Decontaminate sequence library

Import sample description text file, import as phyloseq library, and
remove potential contaminate ASVs and sequences. Catalog total number of
ASVs and sequences removed from analysis.

```{r Load libraries for decontam, eval=FALSE}
# library(decontam); library(phyloseq)
```

### Import as phyloseq objects

```{r Extract taxa and sample info as matrix for phyloseq input, eval=FALSE}
tax_matrix <- merged_tax %>% 
  select(FeatureID = `Feature ID`, Taxon) %>% 
  separate(Taxon, c("Domain", "Supergroup", 
                  "Phylum", "Class", "Order",
                  "Family", "Genus", "Species"), sep = ";", remove = FALSE) %>% 
  column_to_rownames(var = "FeatureID") %>% 
  as.matrix

asv_matrix <- merged_asv %>% 
  select(FeatureID = '#OTU ID', everything()) %>% 
  column_to_rownames(var = "FeatureID") %>% 
  as.matrix

# Align row names for each matrix
rownames(tax_matrix) <- row.names(asv_matrix)

# Set rownames of metadata table to SAMPLE information
row.names(metadata) <- metadata$SAMPLE
```

```{r Phyloseq import, eval=FALSE}
# Import asv and tax matrices
ASV = otu_table(asv_matrix, taxa_are_rows = TRUE)
TAX = tax_table(tax_matrix)
phylo_obj <- phyloseq(ASV, TAX)

# Import metadata as sample data in phyloseq
samplenames <- sample_data(metadata)

# join as phyloseq object
physeq_wnames = merge_phyloseq(phylo_obj, samplenames)
# colnames(ASV)

## Check
# physeq_wnames
```

### Identify contaminant ASVs

```{r Assign control samples, eval=FALSE}
# When "Control" appears in "Sample_or_Control column, this is a negative control"
sample_data(physeq_wnames)$is.neg <- sample_data(physeq_wnames)$Sample_or_Control == "Control"
```

```{r Identify contaminant ASVs, eval=FALSE}
# ID contaminants using Prevalence information
contam_prev <- isContaminant(physeq_wnames, 
                               method="prevalence", 
                               neg="is.neg", 
                               threshold = 0.5, normalize = TRUE) 

# Report number of ASVs IDed as contaminants
table(contam_prev$contaminant)
```

> 0.5 - this threshold will ID contaminants in all samples that are more
> prevalent in negative controls than in positive samples.

*As of Dec 30 2021*: 56 ASVs deemed to be contaminant and will be
removed.

### Remove problematic ASVs

```{r Assign contaminant ASVs, eval=FALSE}
# Subset contaminant ASVs
contams <- filter(contam_prev, contaminant == "TRUE")
list_of_contam_asvs <- as.character(row.names(contams))
# length(list_of_contam_asvs)

taxa_contam <- as.data.frame(tax_matrix) %>% 
  rownames_to_column(var = "FeatureID") %>% 
  filter(FeatureID %in% list_of_contam_asvs)
# head(taxa_contam)
```

```{r Compare decontaminated data with original, eval=FALSE}
# View(asv_wtax)
asv_wtax_decon <- asv_wtax %>% 
  filter(!(FeatureID %in% list_of_contam_asvs)) %>% 
  filter(!(Sample_or_Control == "Control"))

tmp_orig <- (asv_wtax %>% filter(!(Sample_or_Control == "Control")))

# Stats on lost
x <- length(unique(tmp_orig$FeatureID)); x
y <- length(unique(asv_wtax_decon$FeatureID)); y
y-x
100*((y-x)/x) # 56 total ASVs lost
a <- sum(tmp_orig$value);a #3.817 million
b <- sum(asv_wtax_decon$value);b #3.799 million 
100*((b-a)/a)
# Lost 0.47% of sequences from whole dataset.

## Subsample to clean ASVs
asv_wtax_wstats <- asv_wtax %>% 
  mutate(DECONTAM = case_when(
    FeatureID %in% list_of_contam_asvs ~ "FAIL",
    TRUE ~ "PASS"
  ))
```

>   Started with 17934 ASVs, post-decontamination, we have 17878 (a loss
    of 56 ASVs).

>   Data started with 3817219 sequences, after removing 56 ASVs, we have
    3788791 total sequences. There was a total loss of 0.74% of
    sequences.

```{r, fig.height=18, fig.width=18, eval=FALSE}
plot_grid(asv_wtax_wstats %>% 
  filter(value > 0) %>% 
  ggplot(aes(x = SAMPLE, fill = DECONTAM)) +
  geom_bar(stat = "count", width = 0.9, color = "black") +
  labs(y = "Total ASVs") +
  coord_flip() +
  theme_linedraw() +
  facet_grid(DATASET ~ ., scale = "free", space = "free") +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 1),
        strip.background = element_blank(), strip.text = element_text(color = "black"),
        legend.position = "bottom"),
  asv_wtax_wstats %>% 
  group_by(SAMPLE, SITE, DECONTAM, DATASET) %>% 
  summarise(SUM_SEQ_DOMAIN = sum(value)) %>% 
  ggplot(aes(x = SAMPLE, y = SUM_SEQ_DOMAIN, fill = DECONTAM)) +
  geom_bar(stat = "identity", color = "black", width = 0.9) +
  labs(y = "Total Sequences") +
  coord_flip() +
  theme_linedraw() +
  facet_grid(DATASET ~ ., scale = "free", space = "free") +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 1),
        strip.background = element_blank(), strip.text = element_text(color = "black"),
        legend.position = "bottom"),
  ncol = 2)
```

This plot shows the distribution of ASVs and sequences that failed or
passed the decontamination step. Most obvious are the control samples
that indicated the potentially contaminate ASVs.

# Survey of sequences from *in situ* samples

```{r Isolate in situ only, eval=FALSE}
# colnames(asv_wtax_wstats)
# unique(asv_wtax_wstats$SAMPLE)
sites <- c("Piccard", "VonDamm", "Axial", "GordaRidge")
asv_insitu <- asv_wtax_wstats %>% filter(Sample_or_Control != "Control") %>% 
       filter(SITE %in% sites) %>% 
       filter(!grepl("_expTf_", SAMPLE)) %>% 
       filter(value > 0) %>% 
       filter(DECONTAM == "PASS")

# Get quick stats on totals
sum(asv_insitu$value) # 3.8 million sequences
length(unique(asv_insitu$FeatureID)) #12,378 ASVs
```
> Final in situ dataset includes 3.79 million sequences and 12,378 ASVs total.

Additional sample QC, check replicates, and determine if replicates
should be averaged.

```{r Investigate sequence and sample QC, fig.height=8, fig.width=15, eval=FALSE}
plot_grid(asv_insitu %>% 
  group_by(SAMPLENAME, VENT, DATASET, Domain) %>% 
  summarise(seqsum_var = sum(value),
            asvcount_var = n()) %>% 
  pivot_longer(ends_with("_var"), names_to = "VARIABLE") %>% 
  ggplot(aes(x = SAMPLENAME, y = value, fill = Domain)) +
    geom_bar(color = "black", stat = "identity", position = "fill") +
    facet_grid(VARIABLE ~ DATASET, space = "free", scales = "free") +
  scale_y_continuous(expand = c(0,0)) +
  theme_linedraw() +
  scale_fill_brewer(palette = "Paired") +
  theme(strip.background = element_blank(), strip.text = element_text(color = "black"),
        axis.text.x = element_text(color = "black", angle = 90, hjust = 1, vjust = 0.5),
        legend.position = "bottom"),
  asv_insitu %>% 
  group_by(SAMPLENAME, VENT, DATASET, Domain) %>% 
  summarise(seqsum_var = sum(value),
            asvcount_var = n()) %>% 
  pivot_longer(ends_with("_var"), names_to = "VARIABLE") %>% 
  ggplot(aes(x = SAMPLENAME, y = value, fill = Domain)) +
    geom_bar(color = "black", stat = "identity", position = "stack") +
    facet_grid(VARIABLE ~ DATASET, space = "free_x", scales = "free") +
  scale_y_continuous(expand = c(0,0)) +
  theme_linedraw() +
  scale_fill_brewer(palette = "Paired") +
  theme(strip.background = element_blank(), strip.text = element_text(color = "black"),
        axis.text.x = element_text(color = "black", angle = 90, hjust = 1, vjust = 0.5),
        legend.position = "bottom"),
ncol = 2)
```

## Base bar plot - taxonomy

```{r Review taxonomy and make bar plot, fig.width=15, fig.height=8, eval=FALSE}
asv_insitu %>% 
  filter(Domain == "Eukaryota") %>%
  # unite(SampleIdentifier, VENT, SAMPLETYPE, sep = " ", remove = FALSE) %>% 
  mutate(Supergroup = ifelse(is.na(Supergroup), "Unknown Eukaryota", Supergroup),
         Phylum = ifelse(is.na(Phylum), "Unknown", Phylum),
         Phylum = ifelse(Phylum == "Alveolata_X", "Ellobiopsidae", Phylum),
         Supergroup = ifelse(Supergroup == "Alveolata", paste(Supergroup, Phylum, sep = "-"), Supergroup)) %>% 
  group_by(FeatureID, Taxon, Domain, Supergroup, Phylum, Class, Order, Family, Genus, Species,
           VENT, SITE, SAMPLETYPE, YEAR, DATASET) %>% 
    summarise(SEQ_AVG_REP = mean(value)) %>% 
  ungroup() %>% 
  group_by(SITE, SAMPLETYPE, VENT, Supergroup) %>% 
    summarise(SEQ_SUM = sum(SEQ_AVG_REP)) %>% 
  ggplot(aes(x = VENT, y = SEQ_SUM, fill = Supergroup)) +
    geom_bar(stat = "identity", position = "stack", color = "black", width = 0.9) +
    facet_grid(. ~ SITE + SAMPLETYPE, scale = "free", space = "free") +
  theme_linedraw() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 1),
        strip.background = element_blank(), strip.text = element_text(color = "black")) +
  scale_y_continuous(expand = c(0,0)) +
  scale_fill_manual(values = c("#f1eef6", "#d7b5d8", "#df65b0", "#ce1256", "#fc9272", "#ef3b2c", 
    "#800026", "#fff7bc", "#fec44f", "#d95f0e", "#a63603", "#74c476", "#238b45", 
    "#00441b", "#7fcdbb", "#084081", "#c6dbef", "#2b8cbe", "#016c59", "#bcbddc", 
    "#807dba", "#54278f", "#bdbdbd", "black"))
```

Repeat taxonomy barplot, but with relative abundance

```{r Review taxonomy and make bar plot relative abundance, fig.width=15, fig.height=8, eval=FALSE}
asv_insitu %>% 
  filter(Domain == "Eukaryota") %>%
  # unite(SampleIdentifier, VENT, SAMPLETYPE, sep = " ", remove = FALSE) %>% 
  mutate(Supergroup = ifelse(is.na(Supergroup), "Unknown Eukaryota", Supergroup),
         Phylum = ifelse(is.na(Phylum), "Unknown", Phylum),
         Phylum = ifelse(Phylum == "Alveolata_X", "Ellobiopsidae", Phylum),
         Supergroup = ifelse(Supergroup == "Alveolata", paste(Supergroup, Phylum, sep = "-"), Supergroup)) %>% 
  group_by(FeatureID, Taxon, Domain, Supergroup, Phylum, Class, Order, Family, Genus, Species,
           VENT, SITE, SAMPLETYPE, YEAR, DATASET) %>% 
    summarise(SEQ_AVG_REP = mean(value)) %>% 
  ungroup() %>% 
  group_by(SITE, SAMPLETYPE, VENT, Supergroup) %>% 
    summarise(SEQ_SUM = sum(SEQ_AVG_REP)) %>% 
  ggplot(aes(x = VENT, y = SEQ_SUM, fill = Supergroup)) +
    geom_bar(stat = "identity", position = "fill", color = "black", width = 0.9) +
    facet_grid(. ~ SITE + SAMPLETYPE, scale = "free", space = "free") +
  theme_linedraw() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 1),
        strip.background = element_blank(), strip.text = element_text(color = "black")) +
  scale_y_continuous(expand = c(0,0)) +
  scale_fill_manual(values = c("#f1eef6", "#d7b5d8", "#df65b0", "#ce1256", "#fc9272", "#ef3b2c", 
    "#800026", "#fff7bc", "#fec44f", "#d95f0e", "#a63603", "#74c476", "#238b45", 
    "#00441b", "#7fcdbb", "#084081", "#c6dbef", "#2b8cbe", "#016c59", "#bcbddc", 
    "#807dba", "#54278f", "#bdbdbd", "black"))
```

# Removal of samples - QC

Filter samples so that the total number of sequences is greater than
20,000 sequences.

```{r, eval=FALSE}
# head(asv_insitu)
# unique(asv_insitu$Sample_or_Control)
# hist(asv_insitu$value)
tmp <- (asv_insitu %>% 
          group_by(SAMPLE, SAMPLENAME) %>% 
          summarise(SUM = sum(value)) %>% 
        filter(SUM < 20000))
toofew <- as.character(unique(tmp$SAMPLE))
toofew
```

Samples: Axial_Dependable_FS900_2013 and
GordaRidge_BSW020_sterivex_2019_REPa removed due to too few sequences.

## Sequence and ASV table summary

Final table reporting total sequences and ASVs for each sample.

```{r Remake table remove samples with too few sequences, eval=FALSE}
asv_insitu_qc <- asv_insitu %>% 
  filter(!(SAMPLE %in% toofew)) %>% 
  filter(value > 0)

stats_seq_asv_postQC <- asv_insitu_qc %>% 
  group_by(SAMPLEID, VENT, DATASET, SITE, SAMPLETYPE, YEAR) %>%
    summarise(SEQ_SUM = sum(value),
     ASV_COUNT = n()) %>% 
  ungroup() %>% 
  gt(
    groupname_col = c("DATASET", "SITE", "YEAR"),
    rowname_col = "SAMPLEID"
  ) %>%
  tab_header(title = "Final sequence & ASV count")

stats_seq_asv_postQC
# sum(asv_insitu_qc$value)
# length(unique(asv_insitu_qc$FeatureID))
gtsave(stats_seq_asv_postQC, filename = "output-tables/seq_asv_count_postQC.html")

# tmp <- asv_insitu_qc %>% 
#   group_by(SAMPLEID, VENT, DATASET, SITE, SAMPLETYPE, YEAR) %>%
#     summarise(SEQ_SUM = sum(value),
#      ASV_COUNT = n())
# mean(tmp$SEQ_SUM); range(tmp$SEQ_SUM)
# mean(tmp$ASV_COUNT); range(tmp$ASV_COUNT)
# View(filter(tmp))
```

# Assess ASV presence/absence

Set up analysis to classify each ASV based on distribution

```{r Classification of ASVs by dataset and sample type, eval=FALSE}
# head(asv_insitu_qc)
# head(insitu_wide)
# unique(asv_insitu_qc$SAMPLETYPE)
# unique(asv_insitu_qc$SITE)

tax_asv_id <- asv_insitu_qc %>% 
  filter(value > 0) %>% #remove zero values
  select(FeatureID, SITE, SAMPLETYPE) %>% # isolate only ASVs that are PRESENT at a site and sampletype
  distinct() %>% #unique this, as presense = present in at least 1 rep (where applicable)
  unite(sample_id, SITE, SAMPLETYPE, sep = "_") %>% 
  # select(-SITE) %>% 
  # distinct() %>% 
  add_column(present = 1) %>%
  pivot_wider(names_from = sample_id, values_from = present, values_fill = 0) %>% 
  rowwise() %>% 
  mutate_at(vars(FeatureID), factor)
```

Is an ASV present only at the vent site? plume? or background? What
about background and plume only?

```{r Assess presence of samples, eval=FALSE}
library(purrr)
any_cols <- function(tax_asv_id) reduce(tax_asv_id, `|`)

asv_class <- tax_asv_id %>%
  mutate(vent = ifelse(any_cols(across(contains("_Vent"), ~ . > 0)), "VENT", ""),
         plume= ifelse(any_cols(across(contains("_Plume"), ~ . > 0)), "PLUME", ""),
         bsw = ifelse(any_cols(across(contains("_Background"), ~ . > 0)), "BSW", ""),
         ) %>% 
  unite(class_tmp, vent, plume, bsw, sep = "_", na.rm = TRUE) %>% 
  mutate(CLASS = case_when(
  class_tmp == "VENT__" ~ "Vent only",
  class_tmp == "_PLUME_" ~ "Plume only",
  class_tmp == "__BSW" ~ "Background only",
  class_tmp == "VENT__BSW" ~ "Vent & background",
  class_tmp == "VENT_PLUME_BSW" ~ "Vent, plume, & background",
  class_tmp == "VENT_PLUME_" ~ "Vent & plume",
  class_tmp == "_PLUME_BSW" ~ "Plume & background"
  )) %>% 
  select(FeatureID, CLASS) %>% distinct()

colnames(tax_asv_id)
```

Binary data frame with 1 indicating presence of ASV (rows) in a given
sample (columns)

Depending on prevalence of ASV, assign groupings of location.

```{r Assign ASV prevalence, eval=FALSE}
asv_class_SITE <- tax_asv_id %>%
  mutate(
    # mcr = ifelse(any_cols(across(contains("Piccard") | contains("VonDamm"), ~ . > 0)), "MCR", ""),
        picc = ifelse(any_cols(across(contains("Piccard"), ~ . > 0)), "Picc", ""),
        vd = ifelse(any_cols(across(contains("VonDamm"), ~ . > 0)), "VD", ""),
         axial = ifelse(any_cols(across(contains("Axial"), ~ . > 0)), "AxS", ""),
         gr = ifelse(any_cols(across(contains("Gorda"), ~ . > 0)), "GR", "")
         ) %>% 
  # unite(class_tmp, mcr, axial, gr, sep = "_", na.rm = TRUE) %>%
  unite(class_tmp, picc, vd, axial, gr, sep = "_", na.rm = TRUE) %>% 
  # unique(asv_class_SITE$class_tmp)
  mutate(SITE_CLASS = case_when(
  class_tmp == "___GR" ~ "Gorda Ridge only",
  class_tmp == "__AxS_" ~ "Axial only",
  class_tmp == "_VD__" ~ "Von Damm only",
  class_tmp == "Picc_VD__" ~ "Piccard & Von Damm",
  class_tmp == "Picc___" ~ "Piccard only",
  class_tmp == "Picc_VD_AxS_" ~ "MCR & Axial",
  class_tmp == "__AxS_GR" ~ "Axial & Gorda Ridge",
  class_tmp == "_VD__GR" ~ "Von Damm & Gorda Ridge",
  class_tmp == "_VD_AxS_GR" ~ "Von Damm, Axial, & Gorda Ridge",
  class_tmp == "_VD_AxS_" ~ "Von Damm & Axial",
  # class_tmp == "MCR__" ~ "Mid-Cayman Rise",
  class_tmp == "Picc_VD__GR" ~ "MCR & Gorda Ridge",
  class_tmp == "Picc__AxS_GR" ~ "Piccard, Axial, & Gorda Ridge",
  class_tmp == "Picc___GR" ~ "Piccard & Gorda Ridge",
  class_tmp == "Picc__AxS_" ~ "Piccard & Axial",
  class_tmp == "Picc_VD_AxS_GR" ~ "All sites"
  )) %>% 
  select(FeatureID, SITE_CLASS) %>% distinct()
# View(select(asv_class_SITE, SITE_CLASS) %>% distinct())
```

Combine together with original ASV table

```{r Combine ASV classifications, eval=FALSE}
insitu_asv_wClass <- asv_insitu_qc %>% 
  left_join(asv_class) %>% 
  left_join(asv_class_SITE)
# head(insitu_asv_wClass)
```

Visualize the total number of ASVs in background, plume, versus
background.

```{r, fig.height=4, fig.width=8, tidy=FALSE, eval=FALSE}
# head(asv_insitu_qc)
# svg("bubbles.svg", h = 4, w = 8)
asv_insitu_qc %>% 
  select(DATASET, FeatureID, SAMPLETYPE) %>% 
  group_by(DATASET, SAMPLETYPE) %>% 
    summarise(COUNT = n_distinct(FeatureID)) %>% 
  ggplot(aes(x = DATASET, y = SAMPLETYPE, fill = SAMPLETYPE)) +
    geom_point(aes(size = COUNT), shape = 21, color = "black") +
    scale_size_continuous(range = c(4,20)) +
  scale_fill_viridis_d(option = "B") +
  theme_void() +
  theme(legend.position = "right",
        axis.text = element_text(color = "black"))
# dev.off()
```

Bubble plot reporting the total number of ASVs found in the vent, plume,
versus background. At each site, the vent protist population had a
higher total number of ASVs.

Repeat visualization by ASV distribution category.

```{r, fig.height=8, fig.width=9, eval=FALSE}
# head(insitu_asv_wClass)
insitu_asv_wClass %>% 
  select(DATASET, FeatureID, SAMPLETYPE, CLASS) %>% 
  group_by(DATASET, SAMPLETYPE, CLASS) %>% 
    summarise(COUNT = n_distinct(FeatureID)) %>% 
  ggplot(aes(x = DATASET, y = SAMPLETYPE, fill = SAMPLETYPE)) +
    geom_point(aes(size = COUNT), shape = 21, color = "black") +
    scale_size_continuous(range = c(4,20)) +
  scale_fill_viridis_d(option = "B") +
  theme_void() +
  theme(legend.position = "right",
        axis.text.x = element_text(color = "black"),
        axis.title.y = element_blank()) +
  facet_grid(SAMPLETYPE + CLASS ~ ., scales = "free", space = "free") +
  labs(x = "", y = "", title = "Total number of ASVs by distribution & sample type")
```

Repeated bubble plot reports the total number of ASVs in the vent,
plume, and background - but now further separated by distribution (i.e.,
if an ASV was found only in the vent and plume = "Vent & plume"). The
largest portion of ASVs were found only at the vent sites (Vent only).

Categories for ASV distribution:

```{r Review classification text, eval=FALSE}
unique(insitu_asv_wClass$CLASS)
unique(insitu_asv_wClass$SITE_CLASS)
length(unique(insitu_asv_wClass$FeatureID))
sum(insitu_asv_wClass$value)
```

# Save working ASV table

Checkpoint to save working dataframes.

```{r, eval=FALSE}
save(asv_insitu, asv_insitu_qc, insitu_asv_wClass, file = "asv-tables-processed-18102021.RData")
```

# Functions to explore community diversity

To explore microbial eukaryotic community diversity at all three sites,
below functions have been written to pass 18S data for each site through
the same analysis. This will be done for all sites together and for them
individually.

Sections below highlight Axial Seamount, Mid-Cayman Rise, and Gorda
Ridge data individually.

```{r set site variables}
axial <- c("Axial")
mcr <- c("VonDamm", "Piccard")
gr <- c("GordaRidge")
all <- c("Axial", "VonDamm", "Piccard", "GordaRidge")
load("asv-tables-processed-18102021.RData", verbose = TRUE)
```

## Bar plot function

Create a bar plot showing the relative sequence abundance of 18S results
to the Supergroup and Phylum level. Function averages across replicates
and then sums to the phylum and supergroup level. Bar plot shows the
relative sequence abundance.

```{r Barplot function, tidy=FALSE, fig.height=12, fig.width=9}
make_bar_relabun <- function(df, selection){
  df_out <- df %>% 
  filter(SITE %in% selection) %>% 
  filter(Domain == "Eukaryota") %>%
  mutate(Supergroup = ifelse(is.na(Supergroup), "Unknown Eukaryota", Supergroup),
         Phylum = ifelse(is.na(Phylum), "Unknown", Phylum),
         Phylum = ifelse(Phylum == "Alveolata_X", "Ellobiopsidae", Phylum),
         Supergroup = ifelse(Supergroup == "Alveolata", paste(Supergroup, Phylum, sep = "-"), Supergroup)) %>% 
  group_by(FeatureID, Taxon, Domain, Supergroup, Phylum, Class, Order, Family, Genus, Species,
           VENT, SITE, SAMPLETYPE, YEAR, DATASET) %>% 
    summarise(SEQ_AVG_REP = mean(value)) %>% 
  ungroup()
  supergroup <- df_out %>% 
  group_by(SITE, SAMPLETYPE, VENT, YEAR, Supergroup) %>% 
    summarise(SEQ_SUM = sum(SEQ_AVG_REP)) %>% 
  ggplot(aes(x = VENT, y = SEQ_SUM, fill = Supergroup)) +
    geom_bar(stat = "identity", position = "fill", color = "black", width = 0.9) +
    facet_grid(. ~ SITE +YEAR + SAMPLETYPE, scale = "free", space = "free") +
  theme_linedraw() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1),
        strip.background = element_blank(), strip.text = element_text(color = "black"),
        legend.position = "right") +
  scale_y_continuous(expand = c(0,0)) +
  # scale_fill_brewer(palette = "Set2") +
  scale_fill_manual(values = c("#fa9fb5", "#c51b8a", "#67000d", "#ef3b2c", "#ffffcc", "#feb24c", "#c7e9b4", "#1d91c0", "#253494", "#9e9ac8", "#238b45", "#54278f", "#bdbdbd", "#252525", "#fa9fb5", "#c51b8a", "#67000d", "#ef3b2c", "#ffffcc", "#feb24c", "#c7e9b4", "#1d91c0", "#253494", "#9e9ac8", "#238b45", "#54278f", "#bdbdbd", "#252525")) +
  labs(x = "", y = "Relative abundance")
  
  phylum <- df_out %>% 
    unite(SupergroupPhylum, Supergroup, Phylum, sep = "-") %>% 
  group_by(SITE, SAMPLETYPE, VENT, YEAR, SupergroupPhylum) %>% 
    summarise(SEQ_SUM = sum(SEQ_AVG_REP)) %>% 
  ggplot(aes(x = VENT, y = SEQ_SUM, fill = SupergroupPhylum)) +
    geom_bar(stat = "identity", position = "fill", color = "black", width = 0.9) +
    facet_grid(. ~ SITE +YEAR + SAMPLETYPE, scale = "free", space = "free") +
  theme_linedraw() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1),
        strip.background = element_blank(), strip.text = element_text(color = "black"),
        legend.position = "right") +
  scale_y_continuous(expand = c(0,0)) +
  scale_fill_manual(values = c("#f1eef6", "#d7b5d8", "#df65b0", "#ce1256", "#fc9272", "#ef3b2c", 
    "#800026", "#fff7bc", "#fec44f", "#d95f0e", "#a63603", "#74c476", "#238b45", 
    "#00441b", "#7fcdbb", "#084081", "#c6dbef", "#2b8cbe", "#016c59", "#bcbddc", 
    "#807dba", "#54278f", "#bdbdbd", "black", "white", "#969696", "#525252", "#f1eef6", "#d7b5d8", "#df65b0", "#ce1256", "#fc9272", "#ef3b2c", 
    "#800026", "#fff7bc", "#fec44f", "#d95f0e", "#a63603", "#74c476", "#238b45", 
    "#00441b", "#7fcdbb", "#084081", "#c6dbef", "#2b8cbe", "#016c59", "#bcbddc", 
    "#807dba", "#54278f", "#bdbdbd", "black", "white")) +
  labs(x = "", y = "Relative abundance")
  supergroup + phylum + patchwork::plot_layout(ncol = 1)
}

# make_bar_relabun(insitu_asv_wClass, axial)
```

## Tile plot

Relative abundance plots are misleading, as this tag-sequence data is
compositional. To combat this, we can also perform a center log-ratio
transformation of the sequence counts. This tile plot (or heat map) will
show the relationship from the data mean. Positive values thus
demonstrate an increase in the taxa, while negative values illustrate
the opposite.

Ahead of the CLR transformation, average across replicates, then sum to
the Class level. THEN perform CLR transformation and plot as heat map.

```{r CLR transformation and generate tile plot}
make_clr_trans_tile <- function(df, selection){
  df_wide <- df %>%
    filter(SITE %in% selection) %>%
  # df_wide <- insitu_asv_wClass %>% 
    # filter(SITE %in% axial) %>% 
    filter(Domain == "Eukaryota") %>% 
  # Average across replicates
      group_by(FeatureID, SAMPLENAME, VENT, Domain, Supergroup, Phylum, Class, Order, Family, Genus, Species) %>% 
        summarise(AVG = mean(value)) %>% 
      ungroup() %>% 
    # Sum to the Order taxonomic classification
    unite(SAMPLENAME_2, SAMPLENAME, VENT, sep = "_") %>% 
      group_by(SAMPLENAME_2, Supergroup, Phylum, Class) %>% 
        summarise(CLASS_SUM = sum(AVG)) %>% 
    unite(CLASS, Supergroup, Phylum, Class, sep = " ") %>% 
    select(CLASS, SAMPLENAME_2, CLASS_SUM) %>% 
    pivot_wider(names_from = SAMPLENAME_2, values_from = CLASS_SUM, values_fill = 0) %>% 
    column_to_rownames(var = "CLASS")
  ## Take wide data frame and CLR transform, pivot to wide, and plot
  data.frame(compositions::clr(df_wide)) %>% 
      rownames_to_column(var = "CLASS") %>%
      pivot_longer(cols = starts_with(selection), values_to = "CLR", names_to = "SAMPLENAME_2") %>% 
      separate(SAMPLENAME_2, c("SAMPLENAME", "VENT"), sep = "_") %>% 
      separate(SAMPLENAME, c("SITE", "SAMPLETYPE", "YEAR", "Sample_tmp"), remove = TRUE) %>% 
      mutate(VENT = str_replace_all(VENT, "\\.", " ")) %>% 
      mutate(REGION = case_when(
        SITE == "GordaRidge" ~ "Gorda Ridge",
        SITE %in% mcr ~ "Mid-Cayman Rise",
        # SITE == "Piccard" ~ "Piccard",
        # SITE == "VonDamm" ~ "Von Damm",
        SITE == "Axial" ~ "Axial")) %>% 
      mutate(VENTNAME = case_when(
        REGION == "Gorda Ridge" ~ VENT,
        REGION == "Mid-Cayman Rise" ~ paste(SITE, VENT, sep = " "),
        REGION == "Axial" ~ paste(VENT, YEAR, sep = " ")
      )) %>% select(-Sample_tmp) %>% 
    unite(SAMPLE, REGION, VENTNAME, sep = " ", remove = FALSE) %>% 
    separate(CLASS, c("Supergroup", "Phylum", "Class"), sep = " ", remove = FALSE) %>%
    ggplot(aes(x = SAMPLE, y = Class, fill = CLR)) +
      geom_tile(color = "#252525") + 
      theme(legend.position = "right", 
            panel.grid.major = element_blank(),
            panel.grid.minor = element_blank(), 
            panel.border = element_blank(), 
            panel.background = element_blank(),
            axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5, color = "black",size = 8), 
            axis.text.y = element_text(color = "black", size = 8),
            strip.background = element_blank(), 
            strip.text.y = element_text(hjust = 0, vjust = 0.5, angle = 0),
            # strip.text.x = element_blank(), 
            legend.title = element_blank()) + 
      labs(x = "", y = "") +
      scale_fill_gradient2(low = "#4575b4", mid = "white", high = "#d73027", na.value = "grey50") +
      facet_grid(Supergroup + Phylum ~ SAMPLETYPE, space = "free", scales = "free")
}

```

## PCA

Similar to aove, the first step in this function transforms data using
CLR (to ASV level though). First plot will show eigen values (scree plot
to determine if 2 vs. 3 dimensions is best for data). Then function
extracts data points and creates PCA plot.

```{r function to perform CLR transformation and PCoA analysis}
make_pca <- function(df, selection){
 df_wide_asv <- df %>%
    # df_wide_asv <- insitu_asv_wClass %>%
  filter(SITE %in% selection) %>%
    # filter(SITE %in% axial) %>%
  filter(Domain == "Eukaryota") %>% 
  # Average across replicates
    group_by(FeatureID, SAMPLENAME, VENT) %>% 
    summarise(AVG = mean(value)) %>% 
    ungroup() %>% 
  separate(SAMPLENAME, c("SITE", "SAMPLETYPE", "YEAR", "Sample_tmp"), remove = TRUE) %>% 
  mutate(REGION = case_when(
    SITE == "GordaRidge" ~ "Gorda Ridge",
    SITE %in% mcr ~ "Mid-Cayman Rise",
    SITE == "Axial" ~ "Axial")) %>% 
  mutate(VENTNAME = case_when(
    REGION == "Gorda Ridge" ~ VENT,
    REGION == "Mid-Cayman Rise" ~ paste(SITE, VENT, sep = " "),
    REGION == "Axial" ~ paste(VENT, YEAR, sep = " ")
  )) %>% select(-Sample_tmp) %>% 
  unite(SAMPLE, SAMPLETYPE, REGION, VENTNAME, sep = "_", remove = FALSE) %>% 
   group_by(FeatureID, SAMPLE) %>% 
   summarise(SUM = sum(AVG)) %>% 
  pivot_wider(names_from = FeatureID, values_from = SUM, values_fill = 0) %>% 
  column_to_rownames(var = "SAMPLE")
  # look at eigenvalues
  pca_lr <- prcomp(data.frame(compositions::clr(df_wide_asv)))
  variance_lr <- (pca_lr$sdev^2)/sum(pca_lr$sdev^2)
  ## View bar plot
  barplot(variance_lr, main = "Log-Ratio PCA Screeplot", xlab = "PC Axis", ylab = "% Variance", 
    cex.names = 1.5, cex.axis = 1.5, cex.lab = 1.5, cex.main = 1.5)
  ## Extract PCR points
  data.frame(pca_lr$x, SAMPLE = rownames(pca_lr$x)) %>% 
      separate(SAMPLE, c("SAMPLETYPE", "REGION", "VENTNAME"), sep = "_", remove = FALSE) %>% 
  ## Generate PCA plot
  ggplot(aes(x = PC1, y = PC2, shape = SAMPLETYPE, fill = VENTNAME)) +
    geom_hline(yintercept = 0) + geom_vline(xintercept = 0, color = "#525252") +
    geom_point(size=4, stroke = 1, aes(fill = VENTNAME)) +
    ylab(paste0('PC2 ',round(variance_lr[2]*100,2),'%')) +
    xlab(paste0('PC1 ',round(variance_lr[1]*100,2),'%')) +
    scale_shape_manual(values = c(21, 23, 24)) +
    scale_fill_viridis(discrete = TRUE, option = "turbo") +
    # scale_fill_manual(values = fill_color) +
    # scale_color_manual(values = color_color) +
    theme_bw() +
    theme(axis.text = element_text(color="black", size=12),
          legend.title = element_blank(),
          axis.title = element_text(color="black", size=14),
          legend.text = element_text(color = "black", size = 14),
          plot.margin = margin(2, 1, 2, 1, "cm")) +
    guides(fill = guide_legend(override.aes = list(shape = 21) ),
              shape = guide_legend(override.aes = list(fill = "black")))
  }
```

## ASV richness

From complete dataset, average across replicates, then sum the total
number of ASVs in each sample. Then plot a data point for total number
of ASVs (ASV richness) by sample type - where sample type represents the
vent, plume, vs. background. Box plots show the median and range.

```{r function to plot boxplot of ASV richness}
make_asv_rich <- function(df, selection){
  df %>% 
  filter(SITE %in% selection) %>%
  filter(Domain == "Eukaryota") %>% 
  # Average across replicates
    group_by(FeatureID, SAMPLENAME, VENT, Supergroup) %>% 
    summarise(AVG = mean(value)) %>% 
  separate(SAMPLENAME, c("SITE", "SAMPLETYPE", "YEAR", "Sample_tmp"), remove = TRUE) %>% 
  mutate(REGION = case_when(
    SITE == "GordaRidge" ~ "Gorda Ridge",
    SITE %in% mcr ~ "Mid-Cayman Rise",
    SITE == "Axial" ~ "Axial")) %>% 
  mutate(VENTNAME = case_when(
    REGION == "Gorda Ridge" ~ VENT,
    REGION == "Mid-Cayman Rise" ~ paste(SITE, VENT, sep = " "),
    REGION == "Axial" ~ paste(VENT, YEAR, sep = " ")
  )) %>% select(-Sample_tmp) %>% 
  unite(SAMPLE, REGION, VENTNAME, sep = " ", remove = FALSE) %>% 
  ungroup() %>% 
  group_by(SITE, REGION, SAMPLE, SAMPLETYPE) %>% 
    summarise(NUM_ASV = n()) %>% 
  ggplot(aes(x = SAMPLETYPE, y = NUM_ASV, shape = SAMPLETYPE)) +
  geom_boxplot(aes(group = SAMPLETYPE), alpha = 0.8, width = 0.4) +
  geom_jitter(size=2, aes(fill = SITE)) +
  scale_shape_manual(values = c(21, 23, 24)) +
  theme_bw() +
  theme(axis.text = element_text(color="black", size=12),
        legend.title = element_blank(),
        axis.title = element_text(color="black", size=14),
        legend.text = element_text(color = "black", size = 14)) +
  guides(fill = guide_legend(override.aes = list(shape = 21) ),
            shape = guide_legend(override.aes = list(fill = "black") ) ) +
  labs(x = "", y = "Total number of ASVs")
}
```

## Presence-absence

Bar plot (colors correspond to Supergroup) represents the number of ASVs
shared or unique to each sample. Combination matrix below bars shows
which samples are considered for the bar plot.

```{r function to generate upsetR presence absence plot}
make_upset_plot <- function(df, selection){ 
  df %>% 
  filter(SITE %in% selection) %>%
  filter(Domain == "Eukaryota") %>% 
  # Average across replicates
    group_by(FeatureID, SAMPLENAME, VENT, Supergroup) %>% 
    summarise(AVG = mean(value)) %>% 
  separate(SAMPLENAME, c("SITE", "SAMPLETYPE", "YEAR", "Sample_tmp"), remove = TRUE) %>% 
  mutate(REGION = case_when(
    SITE == "GordaRidge" ~ "Gorda Ridge",
    SITE %in% mcr ~ "Mid-Cayman Rise",
    SITE == "Axial" ~ "Axial")) %>% 
  mutate(VENTNAME = case_when(
    REGION == "Gorda Ridge" ~ VENT,
    REGION == "Mid-Cayman Rise" ~ paste(SITE, VENT, sep = " "),
    REGION == "Axial" ~ paste(VENT, YEAR, sep = " ")
  )) %>% select(-Sample_tmp) %>% 
  unite(SAMPLE, SITE, VENTNAME, sep = " ", remove = FALSE) %>% 
  distinct(FeatureID, Supergroup, AVG, SAMPLE, .keep_all = TRUE) %>% 
  group_by(FeatureID, Supergroup) %>% 
  summarise(SAMPLE = list(SAMPLE)) %>% 
  ggplot(aes(x = SAMPLE)) +
    geom_bar(color = "black", width = 0.5, aes(fill = Supergroup)) +
    scale_x_upset(n_intersections = 35) +
  scale_y_continuous(expand = c(0,0)) +
  labs(x = "", y = "Shared ASVs") +
  theme_linedraw() +
  theme(axis.text = element_text(color="black", size=10),
        axis.title = element_text(color="black", size=10),
        legend.text = element_text(color = "black", size = 10),
        plot.margin = margin(1, 1, 1, 5, "cm")) +
    scale_fill_manual(values = c("#f1eef6", "#d7b5d8", "#df65b0", "#ce1256", "#fc9272", "#ef3b2c", 
    "#800026", "#fff7bc", "#fec44f", "#d95f0e", "#a63603", "#74c476", "#238b45", 
    "#00441b", "#7fcdbb", "#084081", "#c6dbef", "#2b8cbe", "#016c59", "#bcbddc", 
    "#807dba", "#54278f", "#bdbdbd", "black"))
}

```

## Top 15 taxa
Isolate the top 15 taxa
```{r, fig.height=8, fig.width=10}
# head(insitu_asv_wClass)

all_class <- as.character(unique(insitu_asv_wClass$CLASS))

all_class_site <- as.character(unique(insitu_asv_wClass$SITE_CLASS))

top15 <- function(df, NUM, site, level, taxa, class, category, plot_tax){
  level <- enquo(level)
  class <- enquo(class)
  plot_tax <- enquo(plot_tax)
  all_class <- as.character(unique(insitu_asv_wClass$CLASS))
  all_class_site <- as.character(unique(insitu_asv_wClass$SITE_CLASS))
out_table <- df %>% 
  # filter(SITE %in% site) %>%
  filter(!!level == taxa) %>%
  # filter(Domain %in% "Eukaryota") %>% 
  # filter(CLASS %in% all_class) %>% 
  filter(!!class %in% category) %>%
  # Average across replicates
  group_by(FeatureID, Taxon, Domain, Supergroup, Phylum, Class, Order, Family, Genus, Species,
           VENT, SITE, SAMPLETYPE, YEAR, DATASET, !!class) %>% 
    summarise(SEQ_AVG_REP = mean(value)) %>% 
  ungroup() %>% 
  # Select top 15 ASVs
  group_by(VENT, SITE, SAMPLETYPE, YEAR, DATASET) %>% 
  top_n(NUM, wt = SEQ_AVG_REP) %>% 
  unite(SAMPLENAME, SITE, SAMPLETYPE, VENT, YEAR, sep = " ", remove = FALSE)
  # 
  plot <- ggplot(out_table, aes(x = SAMPLENAME, fill = !!plot_tax)) +
    geom_bar(stat = "count", color = "black", width = 0.7) +
    coord_flip() +
    facet_grid(SAMPLETYPE + SITE ~ ., space = "free", scales = "free") +
    theme_classic() +
    scale_fill_viridis(discrete = TRUE, option = "turbo")
  ##
  plotly::ggplotly(plot)
}
  
# Function usage:
# top15 <- function(df, NUM, site, level, taxa, class, category, plot_tax)
# top15(insitu_asv_wClass, 10, all, Phylum, "Ciliophora", CLASS, all_class, Class)
# all_class
```



# Axial Seamount analysis

-   What should I do with the year to year comparison?
Individual analyses here for Axial, MCR, and GR will go in supplementary.

## Bar plot relative abundance: Axial

```{r Barplot axial, tidy=FALSE, fig.height=10, fig.width=11}
make_bar_relabun(insitu_asv_wClass, axial)
```

Axial Seamount samples from archived material - span 2013, 2014, and
2015. First, the background and plume (from 2015 only, and from plume
associated with the Anemone vent) are different from the vent samples -
overwhelmingly stramneopile and rhizaria. For the background and plume,
the stramenopiles appear to be associated with ochrophyta or opalozoa.
For the plume, the rhizaria population was associated with cercozoa,
while the background seawater was identified as belonging to radiolaria.

The major difference between the background/plume and vent sites was the
higher relative sequence abundance of ciliates and opisthokonta. For the
opisthokonta, these are primarily metazoa - and I will need to
investigate this further. Exceptions for this include the 'Dependable'
vent from 2013, which had a completely different composition, and
'Marker 113' in 2015, which the opisthokonta sequences were assigned
choanoflagellate and fungi.

*Further questions to consider*

> Any geochemical changes to Marker 113 from 2013/2014 to 2015? Could
 attribute difference of opisthokonta colonization.


## Tile plot CLR: Axial

```{r Isolate axial data and CLR, fig.height=18, fig.width=7}
make_clr_trans_tile(insitu_asv_wClass, axial)
```

Tile plot goes to the Class taxonomic level. Here at Axial, mostly the
ciliate class had higher CLR values (more enriched relative to the data
mean). Second to ciliates were cercozoa. Also noticing how Marker 113
2013 and 2015 are more similar to each other than 2014?

## PCA: Axial

```{r est eigenvalues axial, fig.height=6, fig.width=8}
make_pca(insitu_asv_wClass, axial)
```

While we only have 1 plume and bsw each for Axial, they are grouping
together - away from vents. So that is an expected signature and likely
consistent with the other sites. These colors are a little confusing, it
does look like Boca is an outlier.

## ASV richness: Axial

```{r Plot ASV richness for axial, tidy=FALSE, fig.height=4, fig.width=6}
make_asv_rich(insitu_asv_wClass, axial)
```

We only have 1 sample for background and plume from Axial Seamount. But
this shows that the vent sites have varied ASV richness,

## Presence-absence: Axial

```{r generate upsetR plot for axial, fig.width=12, fig.height=8}
make_upset_plot(insitu_asv_wClass, axial)
```
### Presence-absence at Axial - year and site
```{r, fig.width=12, fig.height=9}
alv <- c("Alveolata-Ellobiopsidae", "Alveolata-Perkinsea", "Alveolata-Unknown", "Alveolata-Chrompodellids", "Alveolata-Apicomplexa")
# svg("upsetR-bysite-sampletype-nov2.svg", h=9, w=15)
axial_loc_yr <- insitu_asv_wClass %>% 
  filter(SITE %in% axial) %>%
  filter(Domain == "Eukaryota") %>% 
  filter(Supergroup != "Opisthokonta") %>% 
  mutate(Supergroup = ifelse(is.na(Supergroup), "Unknown Eukaryota", Supergroup),
         Phylum = ifelse(is.na(Phylum), "Unknown", Phylum),
         Phylum = ifelse(Phylum == "Alveolata_X", "Ellobiopsidae", Phylum),
         Supergroup = ifelse(Supergroup == "Alveolata", paste(Supergroup, Phylum, sep = "-"), Supergroup)) %>% 
  mutate(SUPERGROUP = case_when(
    Supergroup %in% alv ~ "Other Alveolata",
    Supergroup == "Eukaryota_X" ~ "Unknown Eukaryota",
    Phylum == "Cercozoa" ~ "Rhizaria-Cercozoa",
    Phylum == "Radiolaria" ~ "Rhizaria-Radiolaria",
    Phylum == "Ochrophyta" ~ "Stramenopiles-Ochrophyta",
    Phylum == "Opalozoa" ~ "Stramenopiles-Opalozoa",
    Phylum == "Sagenista" ~ "Stramenopiles-Sagenista",
    TRUE ~ Supergroup
  )) %>% 
  # Taxa to supergroup
  mutate(SupergroupPhylum = SUPERGROUP) %>%
  # Average across replicates
    group_by(FeatureID, SAMPLENAME, VENT, SupergroupPhylum) %>% 
    summarise(AVG = mean(value)) %>% 
  ungroup() %>% 
  separate(SAMPLENAME, c("SITE", "SAMPLETYPE", "YEAR", "Sample_tmp"), remove = TRUE) %>% 
  mutate(REGION = case_when(
    SITE == "GordaRidge" ~ "Gorda Ridge",
    SITE %in% mcr ~ "Mid-Cayman Rise",
    SITE == "Axial" ~ "Axial")) %>% 
  mutate(VENTNAME = case_when(
    REGION == "Gorda Ridge" ~ VENT,
    REGION == "Mid-Cayman Rise" ~ paste(SITE, VENT, sep = " "),
    REGION == "Axial" ~ paste(VENT, YEAR, sep = " ")
  )) %>% select(-Sample_tmp) %>% 
  filter(REGION == "Axial") %>% 
  unite(SAMPLE, SITE, SAMPLETYPE, sep = " ", remove = FALSE) %>% 
  unite(YR_LOC, SAMPLETYPE, YEAR, sep = " ", remove = FALSE) %>% 
  group_by(FeatureID, SupergroupPhylum, YR_LOC) %>% 
    summarise(SUM = sum(AVG)) %>%
  # filter(SUM > 200) %>% 
  ungroup()
##  
axial_loc_yr %>% 
  distinct(FeatureID, SupergroupPhylum, SUM, YR_LOC, .keep_all = TRUE) %>% 
  group_by(FeatureID, SupergroupPhylum) %>% 
  summarise(SAMPLE = list(YR_LOC)) %>% 
  ggplot(aes(x = SAMPLE)) +
    geom_bar(color = "black", width = 0.5, aes(fill = SupergroupPhylum)) +
    scale_x_upset(n_intersections = 25) +
  scale_y_continuous(expand = c(0,0)) +
  labs(x = "", y = "Shared ASVs") +
  theme_linedraw() +
  theme(axis.text.y = element_text(color="black", size=14),
        axis.text.x = element_text(color="black", size=14),
        axis.title = element_text(color="black", size=14),
        legend.text = element_text(color = "black", size = 12),
        plot.margin = margin(1, 1, 1, 5, "cm")) +
  scale_fill_manual(values = c("#fa9fb5", "#c51b8a", "#67000d", "#ef3b2c", "#ffffcc", "#feb24c", "#c7e9b4", "#1d91c0", "#deebf7", "#253494", "#9e9ac8", "#238b45", "#54278f", "#bdbdbd", "#252525", "#fa9fb5", "#c51b8a", "#67000d", "#ef3b2c", "#ffffcc", "#feb24c", "#c7e9b4", "#1d91c0", "#253494", "#9e9ac8", "#238b45", "#54278f", "#bdbdbd", "#252525"))
# dev.off()

```


Extract total number of ASVs from Axial data, and those that were found within the vent fluid only.
```{r}
length(unique(axial_loc_yr$FeatureID))
tmp <- axial_loc_yr %>% 
  filter(grepl("Vent", YR_LOC)) %>% 
  pivot_wider(names_from = YR_LOC, values_from = SUM, values_fill = NA) %>% 
  drop_na()
head(tmp)  
length(unique(tmp$FeatureID))
```

```{r}
tmp %>% 
  pivot_longer(starts_with("Vent ")) %>% 
  group_by(name, SupergroupPhylum) %>% 
  summarise(VALUE = sum(value)) %>% 
ggplot(aes(x = name, y = VALUE, fill = SupergroupPhylum)) +
  geom_bar(stat = "identity")
```


## Top 15 taxa: Axial
```{r}
# Function usage:
# top15 <- function(df, NUM, site, level, taxa, class, category, plot_tax)

top15(insitu_asv_wClass, 15, axial, Domain, "Eukaryota", CLASS, all_class, Phylum)
```
```{r}
# unique(insitu_asv_wClass$CLASS)
vent_only <- c("Vent only")
# Isolate top 15 vent-only ASVs from axial - at the phylum level
top15(insitu_asv_wClass, 15, axial, Domain, "Eukaryota", CLASS, vent_only, Phylum)

# Top 10 ciliate only taxa, as most were ciliates
top15(insitu_asv_wClass, 10, axial, Phylum, "Ciliophora", CLASS, vent_only, Class)
```


# Mid-Cayman Rise

## Bar plot relative abundance: MCR

```{r Barplot MCR, fig.height=10, fig.width=11}
make_bar_relabun(insitu_asv_wClass, mcr)
```

## Tile plot CLR: MCR

```{r Isolate MCR data and CLR, fig.height=13, fig.width=8}
make_clr_trans_tile(insitu_asv_wClass, mcr)
```

## PCA: MCR

```{r est eigenvalues MCR, fig.height=6, fig.width=8}
make_pca(insitu_asv_wClass, mcr)
```

## ASV richness: MCR

```{r Plot ASV richness for MCR, tidy=FALSE}
make_asv_rich(insitu_asv_wClass, mcr)
```

## Presence-absence: MCR

```{r make upsetR plot for MCR, fig.width=12, fig.height=8}
make_upset_plot(insitu_asv_wClass, mcr)
```

Repeat presence-absence plot, but with a lower resolution.

```{r, fig.width=8, fig.height=4}
insitu_asv_wClass %>% 
  filter(SITE %in% mcr) %>%
  filter(Domain == "Eukaryota") %>% 
  # Average across replicates
    group_by(FeatureID, SAMPLENAME, VENT, Supergroup) %>% 
    summarise(AVG = mean(value)) %>% 
  separate(SAMPLENAME, c("SITE", "SAMPLETYPE", "YEAR", "Sample_tmp"), remove = TRUE) %>% 
  mutate(REGION = case_when(
    SITE == "GordaRidge" ~ "Gorda Ridge",
    SITE %in% mcr ~ "Mid-Cayman Rise",
    SITE == "Axial" ~ "Axial")) %>% 
  mutate(VENTNAME = case_when(
    REGION == "Gorda Ridge" ~ VENT,
    REGION == "Mid-Cayman Rise" ~ paste(SITE, VENT, sep = " "),
    REGION == "Axial" ~ paste(VENT, YEAR, sep = " ")
  )) %>% select(-Sample_tmp) %>% 
  unite(SAMPLE, SITE, SAMPLETYPE, sep = " ", remove = FALSE) %>% 
  group_by(FeatureID, Supergroup, SAMPLE) %>% 
    summarise(SUM = sum(AVG)) %>%
  ungroup() %>% 
  distinct(FeatureID, Supergroup, SUM, SAMPLE, .keep_all = TRUE) %>% 
  group_by(FeatureID, Supergroup) %>% 
  summarise(SAMPLE = list(SAMPLE)) %>% 
  ggplot(aes(x = SAMPLE)) +
    geom_bar(color = "black", width = 0.5, aes(fill = Supergroup)) +
    scale_x_upset(n_intersections = 15) +
  scale_y_continuous(expand = c(0,0)) +
  labs(x = "", y = "Shared ASVs") +
  theme_linedraw() +
  theme(axis.text = element_text(color="black", size=10),
        axis.title = element_text(color="black", size=10),
        legend.text = element_text(color = "black", size = 10),
        plot.margin = margin(1, 1, 1, 5, "cm")) +
    scale_fill_manual(values = c("#f1eef6", "#d7b5d8", "#df65b0", "#ce1256", "#fc9272", "#ef3b2c", 
    "#800026", "#fff7bc", "#fec44f", "#d95f0e", "#a63603", "#74c476", "#238b45", 
    "#00441b", "#7fcdbb", "#084081", "#c6dbef", "#2b8cbe", "#016c59", "#bcbddc", 
    "#807dba", "#54278f", "#bdbdbd", "black"))
```

## Top 15 taxa: MCR
```{r}
# Function usage:
# top15 <- function(df, NUM, site, level, taxa, class, category, plot_tax)
top15(insitu_asv_wClass, 15, mcr, Domain, "Eukaryota", CLASS, all_class, Phylum)
```

```{r}
# unique(insitu_asv_wClass$CLASS)
vent_only <- c("Vent only")
# Isolate top 15 vent-only ASVs from MCR - at the phylum level
top15(insitu_asv_wClass, 15, mcr, Domain, "Eukaryota", CLASS, vent_only, Phylum)

# Top 10 ciliate only taxa, as most were ciliates
top15(insitu_asv_wClass, 10, mcr, Phylum, "Ciliophora", CLASS, vent_only, Class)
```

## MCR only plot to look at parameters and sequence information

```{r}
import_mcr <- read_delim(file = "../../Mid-Cayman_Rise/midcayman-rise-microeuk/table-wcalc.txt", delim = "\t")
# head(import_mcr)
# mcr_metadata <- import_mcr %>% 
#   select(GRAZING_EFFECT_hr)
#   unite(type_site, "2020", _vent_, " ")
```

```{r, fig.width=14, fig.height=8}
# View(asv_insitu_qc)
# unique(tmp$SAMPLENAME)

plot_bubble <- function(VARIABLE){
  asv_insitu_qc %>% 
    filter(SITE %in% mcr) %>% 
    filter(Domain == "Eukaryota") %>% 
    filter(value > 0) %>% 
  # Average across replicates
      group_by(SAMPLENAME, VENT) %>% 
        summarise(SUM = sum(value),
                  ASV_COUNT = n_distinct(FeatureID),
                  TEMP_avg = mean(TEMP),
                  PROK_avg = mean(ProkConc)) %>% 
      ungroup() %>% 
  pivot_longer(cols = c(SUM, ASV_COUNT, TEMP_avg, PROK_avg)) %>% 
  filter(name == VARIABLE) %>% 
  ggplot(aes(x = SAMPLENAME, y = name, size = value)) +
    geom_point(shape = 21, color = "black", aes(size = value)) +
    scale_size_continuous(range = c(1,16)) +
    theme_void() +
    theme(axis.text.x = element_text(color = "black", angle = 45, hjust = 1, vjust = 1),
          axis.text.y = element_text(color = "black"),
          legend.title = element_blank())
}
# plot_grid(
#   plot_bubble("ASV_COUNT") + theme(axis.text.x = element_blank()),
#   plot_bubble("SUM") + theme(axis.text.x = element_blank()),
#   plot_bubble("PROK_avg") + theme(axis.text.x = element_blank()),                           
#   plot_bubble("TEMP_avg"),
#   ncol = 1,
#   align = c("hv"),
#   axis = c("lrtb")
# )
plot_bubble("ASV_COUNT") + theme(axis.text.x = element_blank()) +
  plot_bubble("SUM") + theme(axis.text.x = element_blank()) +
  plot_bubble("PROK_avg") + theme(axis.text.x = element_blank()) + 
  plot_bubble("TEMP_avg") + patchwork::plot_layout(ncol = 1)
# ?plot_grid()
```

# Gorda Ridge

 ## Bar plot relative abundance: GR

```{r Barplot GR, tidy=FALSE, fig.height=10, fig.width=11}
make_bar_relabun(insitu_asv_wClass, gr)
```

## Tile plot CLR: GR

```{r Isolate GR data and CLR, fig.height=12, fig.width=6}
make_clr_trans_tile(insitu_asv_wClass, gr)
```

## PCA: GR

```{r est eigenvalues GR, fig.height=6, fig.width=8}
make_pca(insitu_asv_wClass, gr)
```

## ASV richness: GR

```{r Plot ASV richness for GR, tidy=FALSE}
make_asv_rich(insitu_asv_wClass, gr)
```

## Presence-absence: GR

```{r, fig.width=12, fig.height=8}
make_upset_plot(insitu_asv_wClass, gr)
```

## Top 15 taxa: GR
```{r}
# Function usage:
# top15 <- function(df, NUM, site, level, taxa, class, category, plot_tax)
top15(insitu_asv_wClass, 15, gr, Domain, "Eukaryota", CLASS, all_class, Phylum)
```

```{r}
# unique(insitu_asv_wClass$CLASS)
vent_only <- c("Vent only")
# Isolate top 15 vent-only ASVs from GR - at the phylum level
top15(insitu_asv_wClass, 15, gr, Domain, "Eukaryota", CLASS, vent_only, Phylum)

# Top 10 ciliate only taxa, as most were ciliates
top15(insitu_asv_wClass, 10, gr, Phylum, "Ciliophora", CLASS, vent_only, Class)
```

# Compare all three sites

```{r}
all <- c("Axial", "VonDamm", "Piccard", "GordaRidge")
mcr <- c("VonDamm", "Piccard")
```

## Bar plot relative abundance: all

```{r Barplot all, tidy=FALSE, fig.height=10, fig.width=16}
make_bar_relabun(insitu_asv_wClass, all)
```
### Tree map - simplier taxonomic composition
```{r}
library(treemapify)
# unique(tmp$SUPERGROUP)
```

Filter data to reduce noise and show sample type to vent ecosystem variability.
```{r, fig.height=6, fig.width=7}
alv <- c("Alveolata-Ellobiopsidae", "Alveolata-Perkinsea", "Alveolata-Unknown", "Alveolata-Chrompodellids", "Alveolata-Apicomplexa")
bkgd <- c("Deep seawater", "BSW", "Shallow seawater")
plume <- c("Candelabra Plume", "Mt Edwards Plume", "Plume", "Near vent BW")

to_supergroup <- insitu_asv_wClass %>% 
  filter(SITE %in% all) %>% 
  filter(Domain == "Eukaryota") %>%
  filter(Supergroup != "Opisthokonta") %>% 
  mutate(Supergroup = ifelse(is.na(Supergroup), "Unknown Eukaryota", Supergroup),
         Phylum = ifelse(is.na(Phylum), "Unknown", Phylum),
         Phylum = ifelse(Phylum == "Alveolata_X", "Ellobiopsidae", Phylum),
         Supergroup = ifelse(Supergroup == "Alveolata", paste(Supergroup, Phylum, sep = "-"), Supergroup)) %>% 
  mutate(SUPERGROUP = case_when(
    Supergroup %in% alv ~ "Other Alveolata",
    Supergroup == "Eukaryota_X" ~ "Unknown Eukaryota",
    Phylum == "Cercozoa" ~ "Rhizaria-Cercozoa",
    Phylum == "Radiolaria" ~ "Rhizaria-Radiolaria",
    Phylum == "Ochrophyta" ~ "Stramenopiles-Ochrophyta",
    Phylum == "Opalozoa" ~ "Stramenopiles-Opalozoa",
    Phylum == "Sagenista" ~ "Stramenopiles-Sagenista",
    TRUE ~ Supergroup
  )) %>% 
  mutate(SAMPLETYPEORDER = case_when(
    VENT %in% bkgd ~ "Background",
    VENT %in% plume ~ "Plume",
    TRUE ~ "Vent"
  )) %>% 
  group_by(FeatureID, Taxon, SUPERGROUP,
           VENT, SITE, SAMPLETYPE, YEAR, DATASET, SAMPLETYPEORDER) %>% 
    summarise(SEQ_AVG_REP = mean(value)) %>% 
  ungroup() %>% 
  group_by(SITE, SUPERGROUP, SAMPLETYPEORDER) %>% 
    summarise(ASV_COUNT = n(),
      SEQ_SUM = sum(SEQ_AVG_REP))

  
# Order sample type
to_supergroup$SAMPLETYPEORDER <- factor(to_supergroup$SAMPLETYPEORDER, levels = c("Background", "Plume", "Vent"))

# View(to_supergroup)
# Remove ASVs with fewer than 200 sequences
to_supergroup %>% 
  filter(SEQ_SUM > 200) %>% 
  ggplot(aes(area = SEQ_SUM, fill = SUPERGROUP, subgroup = SUPERGROUP)) +
    geom_treemap(color = "white") +
    geom_treemap_subgroup_border(colour = "white", size = 2) +
    # geom_bar(stat = "identity", position = "fill", color = "black", width = 0.9) +
    facet_grid(SITE ~ SAMPLETYPEORDER) +
  theme_linedraw() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1),
        strip.background = element_blank(), strip.text = element_text(color = "black"),
        legend.position = "right",
        legend.title = element_blank(),
        panel.border = element_blank()) +
  scale_y_continuous(expand = c(0,0)) +
  scale_fill_manual(values = c("#fa9fb5", "#c51b8a", "#67000d", "#ef3b2c", "#ffffcc", "#feb24c", "#c7e9b4", "#1d91c0", "#253494", "#9e9ac8", "#238b45", "#54278f", "#bdbdbd", "#252525", "#fa9fb5", "#c51b8a", "#67000d", "#ef3b2c", "#ffffcc", "#feb24c", "#c7e9b4", "#1d91c0", "#253494", "#9e9ac8", "#238b45", "#54278f", "#bdbdbd", "#252525")) +
  labs(x = "", y = "Sequence proportion by Supergroup")
  
```

#### **Table S2** Percentages of all protistan supergroups

```{r}
# head(to_supergroup)
totalsum_supergroup <- sum(to_supergroup$SEQ_SUM)

to_supergroup %>% 
  group_by(SITE) %>% 
  mutate(SUM_SITE = sum(SEQ_SUM)) %>%
  ungroup() %>% 
  group_by(SITE, SAMPLETYPEORDER) %>% 
  mutate(SUM_SITE_TYPE = sum(SEQ_SUM)) %>% 
  ungroup() %>% 
mutate(Perc_seq_total = (100*(SEQ_SUM/totalsum_supergroup)),
       Perc_seq_site = (100*(SEQ_SUM/SUM_SITE)),
       Perc_seq_site_type = (100*(SEQ_SUM/SUM_SITE_TYPE))) %>% 
  select(-starts_with("SUM_")) %>% 
  gt(
    groupname_col = c("SITE", "SAMPLETYPEORDER")
    # rowname_col = "SAMPLENAME"
  ) %>% 
  fmt_number(columns = starts_with("Perc_"), decimals = 2) %>% 
  fmt_number(columns = SEQ_SUM, decimals = 0) %>% 
  fmt_number(columns = ASV_COUNT, decimals = 0)

```


## Tile plot CLR: all

A better approach down below after isolating the vent-only ASVs.
```{r Isolate all data and CLR, fig.height=16, fig.width=11}
# make_clr_trans_tile(insitu_asv_wClass, all)
```

## PCA: all

```{r est eigenvalues all, fig.height=6, fig.width=12}
make_pca(insitu_asv_wClass, all)
```

## PCA analysis at another level

Repeat, but color by Region and sample type.

```{r, fig.height=5, fig.width=6}
df_wide_asv <- insitu_asv_wClass %>%
  filter(SITE %in% all) %>%
  filter(Domain == "Eukaryota") %>% 
  # Average across replicates
    group_by(FeatureID, SAMPLENAME, VENT) %>% 
    summarise(AVG = mean(value)) %>% 
    ungroup() %>% 
  separate(SAMPLENAME, c("SITE", "SAMPLETYPE", "YEAR", "Sample_tmp"), remove = TRUE) %>% 
  mutate(REGION = case_when(
    SITE == "GordaRidge" ~ "Gorda Ridge",
    SITE %in% mcr ~ "Mid-Cayman Rise",
    SITE == "Axial" ~ "Axial")) %>% 
  mutate(VENTNAME = case_when(
    REGION == "Gorda Ridge" ~ VENT,
    REGION == "Mid-Cayman Rise" ~ paste(SITE, VENT, sep = " "),
    REGION == "Axial" ~ paste(VENT, YEAR, sep = " ")
  )) %>% select(-Sample_tmp) %>% 
  unite(SAMPLE, SAMPLETYPE, SITE, VENTNAME, sep = "_", remove = FALSE) %>% 
   group_by(FeatureID, SAMPLE) %>% 
   summarise(SUM = sum(AVG)) %>% 
  pivot_wider(names_from = FeatureID, values_from = SUM, values_fill = 0) %>% 
  column_to_rownames(var = "SAMPLE")
  # look at eigenvalues
  pca_lr <- prcomp(data.frame(compositions::clr(df_wide_asv)))
  variance_lr <- (pca_lr$sdev^2)/sum(pca_lr$sdev^2)
  ## View bar plot
  barplot(variance_lr, main = "Log-Ratio PCA Screeplot", xlab = "PC Axis", ylab = "% Variance", 
    cex.names = 1.5, cex.axis = 1.5, cex.lab = 1.5, cex.main = 1.5)
  ## Extract PCR points
  data.frame(pca_lr$x, SAMPLE = rownames(pca_lr$x)) %>% 
      separate(SAMPLE, c("SAMPLETYPE", "REGION", "VENTNAME"), sep = "_", remove = FALSE) %>% 
  ## Generate PCA plot
  ggplot(aes(x = PC1, y = PC2, shape = SAMPLETYPE, fill = REGION)) +
    geom_hline(yintercept = 0) + geom_vline(xintercept = 0, color = "#525252") +
    geom_point(size=3, stroke = 1, aes(fill = REGION)) +
    ylab(paste0('PC2 ',round(variance_lr[2]*100,2),'%')) +
    xlab(paste0('PC1 ',round(variance_lr[1]*100,2),'%')) +
    scale_shape_manual(values = c(21, 23, 24)) +
    scale_fill_manual(values = c("#fdbb84", "#31a354", "#ef3b2c", "#02818a")) +
    theme_bw() +
    theme(axis.text = element_text(color="black", size=12),
          legend.title = element_blank(),
          axis.title = element_text(color="black", size=14),
          legend.text = element_text(color = "black", size = 14),
          plot.margin = margin(2, 1, 2, 1, "cm")) +
    guides(fill = guide_legend(override.aes = list(shape = 21) ),
              shape = guide_legend(override.aes = list(fill = "black")))
```


## Dendrogram comparison
Modify sample names for dendrogram plot.
```{r}
df <- as.data.frame(t(df_wide_asv))
###
  colnames(df) <- gsub(x = names(df), pattern = "_", replacement = " ")
  colnames(df) <- gsub(x = names(df), pattern = "Vent Axial", replacement = "Axial")
  colnames(df) <- gsub(x = names(df), pattern = "Vent GordaRidge", replacement = "Gorda Ridge")
  colnames(df) <- gsub(x = names(df), pattern = "GordaRidge", replacement = "Gorda Ridge")
  colnames(df) <- gsub(x = names(df), pattern = "Plume ", replacement = "")
  colnames(df) <- gsub(x = names(df), pattern = "Vent VonDamm VonDamm", replacement = "Von Damm")
  colnames(df) <- gsub(x = names(df), pattern = "Vent Piccard Piccard", replacement = "Piccard")
  colnames(df) <- gsub(x = names(df), pattern = "Background GordaRidge", replacement = "Gorda Ridge")
  colnames(df) <- gsub(x = names(df), pattern = "VonDamm VonDamm", replacement = "Von Damm")
  colnames(df) <- gsub(x = names(df), pattern = "Piccard Piccard", replacement = "Piccard")
  colnames(df) <- gsub(x = names(df), pattern = " BSW", replacement = "")
  colnames(df) <- gsub(x = names(df), pattern = "Background Axial Deep seawater 2015", replacement = "Background Axial 2015")

# Write over same data frame - fix sample names
dendro_input <- df
# head(dendro_input)
```

Estimate Jaccard distance
```{r}
# ?vegan::decostand
# ?vegdist
dendro_jacc <- vegan::vegdist(t(dendro_input), method = "jaccard")
# head(dendro_jacc)

cluster_jacc <- hclust(dist(t(dendro_jacc)), method = "average")

library(ggdendro)
dendro_plot_df <- ggdendro::dendro_data(as.dendrogram(cluster_jacc), type = "rectangle")
label_dendro_order <- as.character(dendro_plot_df$labels$label)
# label_dendro_order
```

Plot dendrogram
```{r}
dendro_plot_output <- ggplot(segment(dendro_plot_df)) + 
  geom_segment(aes(x = x, y = y, xend = xend, yend = yend)) + 
  coord_flip() + 
  scale_y_reverse(expand = c(0.2, 0.5), breaks = c(0, 0.2, 0.4, 0.6, 0.8)) + 
  geom_text(aes(x = x, y = y, label = label, angle = 0, hjust = 0), data = label(dendro_plot_df)) + 
  theme_dendro() + 
  labs(y = "Dissimilarity", title = "Jaccard distance") + 
  theme(axis.text.x = element_text(color = "black", size = 14), 
        axis.line.x = element_line(color = "#252525"), 
        axis.ticks.x = element_line(), axis.title.x = element_text(color = "black", size = 14))
```

Add bar plot in the same order to show proportion of resident versus cosmpolitan ASVs in each sample.

### Bar plot to pair with dendrogram
```{r}
# head(insitu_asv_wClass)
# unique(insitu_asv_wClass$SITE_CLASS)
# unique(insitu_asv_wClass$CLASS)

dendro_bar <- insitu_asv_wClass %>% 
  filter(SITE %in% all) %>%
  filter(Domain == "Eukaryota") %>% 
  # Average across replicates
    group_by(FeatureID, SAMPLENAME, VENT, SITE_CLASS) %>% 
    summarise(AVG = mean(value)) %>% 
    ungroup() %>% 
  separate(SAMPLENAME, c("SITE", "SAMPLETYPE", "YEAR", "Sample_tmp"), remove = TRUE) %>% 
  mutate(REGION = case_when(
    SITE == "GordaRidge" ~ "Gorda Ridge",
    SITE %in% mcr ~ "Mid-Cayman Rise",
    SITE == "Axial" ~ "Axial")) %>% 
  mutate(VENTNAME = case_when(
    REGION == "Gorda Ridge" ~ VENT,
    REGION == "Mid-Cayman Rise" ~ paste(SITE, VENT, sep = " "),
    REGION == "Axial" ~ paste(VENT, YEAR, sep = " ")
  )) %>% select(-Sample_tmp) %>% 
  unite(SAMPLE, SAMPLETYPE, SITE, VENTNAME, sep = "_", remove = FALSE) %>%
  mutate(SITE_CLASS_2 = case_when(
    SITE_CLASS == "Piccard & Von Damm" ~ "Piccard & Von Damm only",
    SITE_CLASS == "Piccard & Axial" ~ "MCR & Axial",
    SITE_CLASS == "Piccard & Gorda Ridge" ~ "MCR & Gorda Ridge",
    SITE_CLASS == "Von Damm & Axial" ~ "MCR & Axial",
    SITE_CLASS == "Von Damm & Gorda Ridge" ~ "MCR & Gorda Ridge",
    SITE_CLASS == "Piccard, Axial, & Gorda Ridge" ~ "MCR, Axial, & Gorda Ridge",
    SITE_CLASS == "Von Damm, Axial, & Gorda Ridge" ~ "MCR, Axial, & Gorda Ridge",
    TRUE ~ SITE_CLASS
  )) %>% 
  group_by(SITE_CLASS_2, SAMPLE) %>% 
  summarise(SEQ_SUM = sum(AVG),
     ASV_COUNT = n()) %>% 
      mutate(SAMPLE = gsub(SAMPLE, pattern = "_", replacement = " ")) %>%
      mutate(SAMPLE = gsub(SAMPLE, pattern = "Vent Axial", replacement = "Axial")) %>%
      mutate(SAMPLE = gsub(SAMPLE, pattern = "Vent GordaRidge", replacement = "Gorda Ridge")) %>%
      mutate(SAMPLE = gsub(SAMPLE, pattern = "GordaRidge", replacement = "Gorda Ridge")) %>%
      mutate(SAMPLE = gsub(SAMPLE, pattern = "Plume ", replacement = "")) %>%
      mutate(SAMPLE = gsub(SAMPLE, pattern = "Vent VonDamm VonDamm", replacement = "Von Damm")) %>%
      mutate(SAMPLE = gsub(SAMPLE, pattern = "Vent Piccard Piccard", replacement = "Piccard")) %>%
      mutate(SAMPLE = gsub(SAMPLE, pattern = "Background GordaRidge", replacement = "Gorda Ridge")) %>%
      mutate(SAMPLE = gsub(SAMPLE, pattern = "VonDamm VonDamm", replacement = "Von Damm")) %>%
      mutate(SAMPLE = gsub(SAMPLE, pattern = "Piccard Piccard", replacement = "Piccard")) %>%
      mutate(SAMPLE = gsub(SAMPLE, pattern = " BSW", replacement = "")) %>%
      mutate(SAMPLE = gsub(SAMPLE, pattern = "Background Axial Deep seawater 2015", replacement = "Background Axial 2015")) 

# unique(insitu_asv_wClass$CLASS)
cosmo <- c("Vent, plume, & background", "Vent & background", "Vent & plume", "Plume & background")
dendro_res_cos_df <- insitu_asv_wClass %>% 
  filter(SITE %in% all) %>%
  filter(Domain == "Eukaryota") %>% 
  mutate(DISTRIBUTION = case_when(
    CLASS %in% cosmo ~ "Cosmopolitan",
    TRUE ~ CLASS
  )) %>% 
  # Average across replicates
    group_by(FeatureID, SAMPLENAME, VENT, DISTRIBUTION) %>% 
    summarise(AVG = mean(value)) %>% 
    ungroup() %>% 
  separate(SAMPLENAME, c("SITE", "SAMPLETYPE", "YEAR", "Sample_tmp"), remove = TRUE) %>% 
  mutate(REGION = case_when(
    SITE == "GordaRidge" ~ "Gorda Ridge",
    SITE %in% mcr ~ "Mid-Cayman Rise",
    SITE == "Axial" ~ "Axial")) %>% 
  mutate(VENTNAME = case_when(
    REGION == "Gorda Ridge" ~ VENT,
    REGION == "Mid-Cayman Rise" ~ paste(SITE, VENT, sep = " "),
    REGION == "Axial" ~ paste(VENT, YEAR, sep = " ")
  )) %>% select(-Sample_tmp) %>% 
  unite(SAMPLE, SAMPLETYPE, SITE, VENTNAME, sep = "_", remove = FALSE) %>%
  group_by(DISTRIBUTION, SAMPLE) %>% 
  summarise(SEQ_SUM = sum(AVG),
     ASV_COUNT = n()) %>% 
      mutate(SAMPLE = gsub(SAMPLE, pattern = "_", replacement = " ")) %>%
      mutate(SAMPLE = gsub(SAMPLE, pattern = "Vent Axial", replacement = "Axial")) %>%
      mutate(SAMPLE = gsub(SAMPLE, pattern = "Vent GordaRidge", replacement = "Gorda Ridge")) %>%
      mutate(SAMPLE = gsub(SAMPLE, pattern = "GordaRidge", replacement = "Gorda Ridge")) %>%
      mutate(SAMPLE = gsub(SAMPLE, pattern = "Plume ", replacement = "")) %>%
      mutate(SAMPLE = gsub(SAMPLE, pattern = "Vent VonDamm VonDamm", replacement = "Von Damm")) %>%
      mutate(SAMPLE = gsub(SAMPLE, pattern = "Vent Piccard Piccard", replacement = "Piccard")) %>%
      mutate(SAMPLE = gsub(SAMPLE, pattern = "Background GordaRidge", replacement = "Gorda Ridge")) %>%
      mutate(SAMPLE = gsub(SAMPLE, pattern = "VonDamm VonDamm", replacement = "Von Damm")) %>%
      mutate(SAMPLE = gsub(SAMPLE, pattern = "Piccard Piccard", replacement = "Piccard")) %>%
      mutate(SAMPLE = gsub(SAMPLE, pattern = " BSW", replacement = "")) %>%
      mutate(SAMPLE = gsub(SAMPLE, pattern = "Background Axial Deep seawater 2015", replacement = "Background Axial 2015")) 

dendro_bar$SAMPLE_ORDER <- factor(dendro_bar$SAMPLE, levels = label_dendro_order) 
dendro_res_cos_df$SAMPLE_ORDER <- factor(dendro_res_cos_df$SAMPLE, levels = label_dendro_order)

# dendro_bar$SITE_CLASS_ORDER <- factor(dendro_bar$SITE_CLASS, levels = c("All sites","Von Damm only","Piccard only","Piccard & Von Damm","MCR & Axial","MCR & Gorda Ridge","Piccard & Axial","Piccard & Gorda Ridge","Piccard, Axial, & Gorda Ridge","Von Damm & Axial","Von Damm & Gorda Ridge","Von Damm, Axial, & Gorda Ridge","Gorda Ridge only","Axial only","Axial & Gorda Ridge"))

dendro_bar$SITE_CLASS_ORDER <- factor(dendro_bar$SITE_CLASS_2, levels = c("All sites","Von Damm only","Piccard only","Piccard & Von Damm only","MCR & Axial","MCR & Gorda Ridge", "MCR, Axial, & Gorda Ridge", "Gorda Ridge only","Axial only","Axial & Gorda Ridge"))
# unique(dendro_bar$SITE_CLASS)
```

```{r}
dendro_bar_plot <- ggplot(dendro_bar, aes(x = SAMPLE_ORDER, y = ASV_COUNT, fill = SITE_CLASS_ORDER)) +
  geom_bar(stat = "identity", position = "fill", color = "black", width = 0.7, alpha = 0.8) +
  coord_flip() +
  scale_y_continuous(expand = c(0,0)) +
  # scale_fill_manual(values = c("grey", "#e6550d", "#fdbb84", "#31a354", "#1c9099", "#fde0dd", "#c51b8a")) +
  scale_fill_manual(values = c("#636363","#bd0026","#fed976","#fd8d3c","#a63603","#fdae6b","#c994c7","#00441b","#ce1256","#addd8e","#f7fcb9","#016c59","#41ab5d","#6a51a3","#3690c0")) +
    theme_bw() +
    theme(axis.text = element_text(color="black", size=5),
          legend.title = element_blank(),
          axis.title = element_text(color="black", size=14),
          legend.text = element_text(color = "black", size = 12),
          plot.margin = margin(1, 1, 1, 1, "cm"),
          legend.position = "top") +
  labs(x = "", y = "Proportion of ASVs")

dendro_bar_plot_res_cos <- ggplot(dendro_res_cos_df, aes(x = SAMPLE_ORDER, y = ASV_COUNT, fill = DISTRIBUTION)) +
  geom_bar(stat = "identity", position = "fill", color = "black", width = 0.7, alpha = 0.8) +
  coord_flip() +
  scale_y_continuous(expand = c(0,0)) +
  viridis::scale_fill_viridis(discrete=TRUE, option = "H") +
    theme_bw() +
    theme(axis.text = element_text(color="black", size=5),
          legend.title = element_blank(),
          axis.title = element_text(color="black", size=14),
          legend.text = element_text(color = "black", size = 12),
          plot.margin = margin(1, 1, 1, 1, "cm"),
          legend.position = "top") +
  labs(x = "", y = "Proportion of ASVs")
# dendro_bar_plot_res_cos
# ?scale_fill_viridis

dendro_bar_plot_SEQ <- ggplot(dendro_bar, aes(x = SAMPLE_ORDER, y = SEQ_SUM, fill = SITE_CLASS_ORDER)) +
  geom_bar(stat = "identity", position = "fill", color = "black", width = 0.7, alpha = 0.8) +
  coord_flip() +
  scale_y_continuous(expand = c(0,0)) +
  # scale_fill_manual(values = c("grey", "#e6550d", "#fdbb84", "#31a354", "#1c9099", "#fde0dd", "#c51b8a")) +
  scale_fill_manual(values = c("#636363","#bd0026","#fed976","#fd8d3c","#a63603","#fdae6b","#c994c7","#00441b","#ce1256","#addd8e","#f7fcb9","#016c59","#41ab5d","#6a51a3","#3690c0")) +
    theme_bw() +
    theme(axis.text = element_text(color="black", size=5),
          legend.title = element_blank(),
          axis.title = element_text(color="black", size=14),
          legend.text = element_text(color = "black", size = 12),
          plot.margin = margin(1, 1, 1, 1, "cm"),
          legend.position = "top") +
  labs(x = "", y = "Proportion of sequences")

dendro_bar_plot_res_cos_SEQ <- ggplot(dendro_res_cos_df, aes(x = SAMPLE_ORDER, y = SEQ_SUM, fill = DISTRIBUTION)) +
  geom_bar(stat = "identity", position = "fill", color = "black", width = 0.7, alpha = 0.8) +
  coord_flip() +
  scale_y_continuous(expand = c(0,0)) +
  viridis::scale_fill_viridis(discrete=TRUE, option = "H") +
    theme_bw() +
    theme(axis.text = element_text(color="black", size=5),
          legend.title = element_blank(),
          axis.title = element_text(color="black", size=14),
          legend.text = element_text(color = "black", size = 12),
          plot.margin = margin(1, 1, 1, 1, "cm"),
          legend.position = "t∂p") +
  labs(x = "", y = "Proportion of sequences")
# dendro_bar_plot
```
Combine all plots with patchwork.
```{r Combine plots dendro and bar, fig.width=18, fig.height=12}
# library(patchwork)
# svg("dendrogram_wbarplots.svg", w = 18, h = 12)
dendro_plot_output + dendro_bar_plot + dendro_bar_plot_res_cos + patchwork::plot_layout(nrow = 1, widths = c(1, 0.2, 0.2),
                  heights = c(1.5, 0.2, 0.2)) + plot_annotation(tag_levels = "a")
# dev.off()
# ?plot_layout
# ?plot_annotation
```


Create the same plot, but by sequence proportion.
```{r,fig.width=18, fig.height=12}
# svg("dendrogram_wbarplots.svg", w = 18, h = 12)
dendro_plot_output + dendro_bar_plot_SEQ + dendro_bar_plot_res_cos_SEQ + patchwork::plot_layout(nrow = 1, widths = c(1, 0.2, 0.2),
                  heights = c(1.5, 0.2, 0.2)) + plot_annotation(tag_levels = "a")
# dev.off()
```


## ASV richness: all

```{r Plot ASV richness for all, tidy=FALSE}
# make_asv_rich(insitu_asv_wClass, all)
```

ASV richness with customized color schema

```{r, fig.height=4, fig.width=6}
insitu_asv_wClass %>% 
  filter(SITE %in% all) %>%
  filter(Domain == "Eukaryota") %>% 
  # Average across replicates
    group_by(FeatureID, SAMPLENAME, VENT, Supergroup) %>% 
    summarise(AVG = mean(value)) %>% 
  separate(SAMPLENAME, c("SITE", "SAMPLETYPE", "YEAR", "Sample_tmp"), remove = TRUE) %>% 
  mutate(REGION = case_when(
    SITE == "GordaRidge" ~ "Gorda Ridge",
    SITE %in% mcr ~ "Mid-Cayman Rise",
    SITE == "Axial" ~ "Axial")) %>% 
  mutate(VENTNAME = case_when(
    REGION == "Gorda Ridge" ~ VENT,
    REGION == "Mid-Cayman Rise" ~ paste(SITE, VENT, sep = " "),
    REGION == "Axial" ~ paste(VENT, YEAR, sep = " ")
  )) %>% select(-Sample_tmp) %>% 
  unite(SAMPLE, REGION, VENTNAME, sep = " ", remove = FALSE) %>% 
  ungroup() %>% 
  group_by(SITE, REGION, SAMPLE, SAMPLETYPE) %>% 
    summarise(NUM_ASV = n()) %>% 
  ggplot(aes(x = SAMPLETYPE, y = NUM_ASV, shape = SAMPLETYPE)) +
  geom_boxplot(aes(group = SAMPLETYPE), alpha = 0.8, width = 0.4) +
  geom_jitter(size=2, width = 0.3, aes(fill = SITE)) +
  scale_shape_manual(values = c(21, 23, 24)) +
    scale_fill_manual(values = c("#fdbb84", "#31a354", "#ef3b2c", "#02818a")) +
  theme_bw() +
  theme(axis.text = element_text(color="black", size=12),
        legend.title = element_blank(),
        axis.title = element_text(color="black", size=14),
        legend.text = element_text(color = "black", size = 14)) +
  guides(fill = guide_legend(override.aes = list(shape = 21) ),
            shape = guide_legend(override.aes = list(fill = NA) ) ) +
  labs(x = "", y = "Total number of ASVs")
```

### ASV richness by sample type & taxa
```{r, fig.height=6, fig.width=12}
insitu_asv_wClass %>% 
  filter(SITE %in% all) %>%
  filter(Domain == "Eukaryota") %>% 
  mutate(Supergroup = ifelse(is.na(Supergroup), "Unknown Eukaryota", Supergroup),
         Phylum = ifelse(is.na(Phylum), "Unknown", Phylum),
         Phylum = ifelse(Phylum == "Alveolata_X", "Ellobiopsidae", Phylum),
         Supergroup = ifelse(Supergroup == "Alveolata", paste(Supergroup, Phylum, sep = "-"), Supergroup)) %>%
  # Average across replicates
    group_by(FeatureID, SAMPLENAME, VENT, Supergroup) %>% 
    summarise(AVG = mean(value)) %>% 
  separate(SAMPLENAME, c("SITE", "SAMPLETYPE", "YEAR", "Sample_tmp"), remove = TRUE) %>% 
  mutate(REGION = case_when(
    SITE == "GordaRidge" ~ "Gorda Ridge",
    SITE %in% mcr ~ "Mid-Cayman Rise",
    SITE == "Axial" ~ "Axial")) %>% 
  mutate(VENTNAME = case_when(
    REGION == "Gorda Ridge" ~ VENT,
    REGION == "Mid-Cayman Rise" ~ paste(SITE, VENT, sep = " "),
    REGION == "Axial" ~ paste(VENT, YEAR, sep = " ")
  )) %>% select(-Sample_tmp) %>% 
  unite(SAMPLE, REGION, VENTNAME, sep = " ", remove = FALSE) %>% 
  ungroup() %>% 
  group_by(SITE, REGION, SAMPLE, SAMPLETYPE, Supergroup) %>% 
    summarise(NUM_ASV = n()) %>% 
  ggplot(aes(x = SAMPLETYPE, y = NUM_ASV, shape = SAMPLETYPE)) +
  geom_boxplot(aes(group = SAMPLETYPE), alpha = 0.8, width = 0.4) +
  geom_jitter(size=2, width = 0.3, aes(fill = Supergroup)) +
  facet_wrap(.~ Supergroup, scales = "free_y") +
  scale_shape_manual(values = c(21, 23, 24)) +
    scale_fill_manual(values = c("#fa9fb5", "#c51b8a", "#67000d", "#ef3b2c", "#ffffcc", "#feb24c", "#c7e9b4", "#1d91c0", "#253494", "#9e9ac8", "#238b45", "#54278f", "#bdbdbd", "#252525", "#fa9fb5", "#c51b8a", "#67000d", "#ef3b2c", "#ffffcc", "#feb24c", "#c7e9b4", "#1d91c0", "#253494", "#9e9ac8", "#238b45", "#54278f", "#bdbdbd", "#252525")) +
  theme_bw() +
  theme(axis.text = element_text(color="black", size=12),
        legend.title = element_blank(),
        axis.title = element_text(color="black", size=14),
        legend.text = element_text(color = "black", size = 14)) +
  guides(fill = guide_legend(override.aes = list(shape = 21) ),
            shape = guide_legend(override.aes = list(fill = NA) ) ) +
  labs(x = "", y = "Total number of ASVs")
```


## Presence-absence: all

```{r upsetR all, fig.width=12, fig.height=8}
# make_upset_plot(insitu_asv_wClass, all)
# head(insitu_asv_wClass)

```

### UpsetR by ASV level
Repeat above plot, but resolve by sample location and sample type.

```{r, fig.width=15, fig.height=10}
alv <- c("Alveolata-Ellobiopsidae", "Alveolata-Perkinsea", "Alveolata-Unknown", "Alveolata-Chrompodellids", "Alveolata-Apicomplexa")
# svg("upsetR-bysite-sampletype-nov2.svg", h=9, w=15)
insitu_asv_wClass %>% 
  filter(SITE %in% all) %>%
  filter(Domain == "Eukaryota") %>% 
  filter(Supergroup != "Opisthokonta") %>% 
  mutate(Supergroup = ifelse(is.na(Supergroup), "Unknown Eukaryota", Supergroup),
         Phylum = ifelse(is.na(Phylum), "Unknown", Phylum),
         Phylum = ifelse(Phylum == "Alveolata_X", "Ellobiopsidae", Phylum),
         Supergroup = ifelse(Supergroup == "Alveolata", paste(Supergroup, Phylum, sep = "-"), Supergroup)) %>% 
  mutate(SUPERGROUP = case_when(
    Supergroup %in% alv ~ "Other Alveolata",
    Supergroup == "Eukaryota_X" ~ "Unknown Eukaryota",
    Phylum == "Cercozoa" ~ "Rhizaria-Cercozoa",
    Phylum == "Radiolaria" ~ "Rhizaria-Radiolaria",
    Phylum == "Ochrophyta" ~ "Stramenopiles-Ochrophyta",
    Phylum == "Opalozoa" ~ "Stramenopiles-Opalozoa",
    Phylum == "Sagenista" ~ "Stramenopiles-Sagenista",
    TRUE ~ Supergroup
  )) %>% 
  # Taxa to supergroup
  mutate(SupergroupPhylum = SUPERGROUP) %>%
  # Average across replicates
    group_by(FeatureID, SAMPLENAME, VENT, SupergroupPhylum) %>% 
    summarise(AVG = mean(value)) %>% 
  ungroup() %>% 
  separate(SAMPLENAME, c("SITE", "SAMPLETYPE", "YEAR", "Sample_tmp"), remove = TRUE) %>% 
  mutate(REGION = case_when(
    SITE == "GordaRidge" ~ "Gorda Ridge",
    SITE %in% mcr ~ "Mid-Cayman Rise",
    SITE == "Axial" ~ "Axial")) %>% 
  mutate(VENTNAME = case_when(
    REGION == "Gorda Ridge" ~ VENT,
    REGION == "Mid-Cayman Rise" ~ paste(SITE, VENT, sep = " "),
    REGION == "Axial" ~ paste(VENT, YEAR, sep = " ")
  )) %>% select(-Sample_tmp) %>% 
  unite(SAMPLE, SITE, SAMPLETYPE, sep = " ", remove = FALSE) %>% 
  group_by(FeatureID, SupergroupPhylum, SAMPLE) %>% 
    summarise(SUM = sum(AVG)) %>%
  # filter(SUM > 200) %>% 
  ungroup() %>% 
  distinct(FeatureID, SupergroupPhylum, SUM, SAMPLE, .keep_all = TRUE) %>% 
  group_by(FeatureID, SupergroupPhylum) %>% 
  summarise(SAMPLE = list(SAMPLE)) %>% 
  ggplot(aes(x = SAMPLE)) +
    geom_bar(color = "black", width = 0.5, aes(fill = SupergroupPhylum)) +
    scale_x_upset(n_intersections = 25) +
  scale_y_continuous(expand = c(0,0)) +
  labs(x = "", y = "Shared ASVs") +
  theme_linedraw() +
  theme(axis.text.y = element_text(color="black", size=14),
        axis.text.x = element_text(color="black", size=14),
        axis.title = element_text(color="black", size=14),
        legend.text = element_text(color = "black", size = 12),
        plot.margin = margin(1, 1, 1, 5, "cm")) +
  scale_fill_manual(values = c("#fa9fb5", "#c51b8a", "#67000d", "#ef3b2c", "#ffffcc", "#feb24c", "#c7e9b4", "#1d91c0", "#deebf7", "#253494", "#9e9ac8", "#238b45", "#54278f", "#bdbdbd", "#252525", "#fa9fb5", "#c51b8a", "#67000d", "#ef3b2c", "#ffffcc", "#feb24c", "#c7e9b4", "#1d91c0", "#253494", "#9e9ac8", "#238b45", "#54278f", "#bdbdbd", "#252525"))
# dev.off()

```

Observations regarding above plot: - Axial and Gorda Ridge vent sites
have more shared ASVs than any other pairwise comparison. After this,
there were also many ASVs shared throughout MCR (vent, plume, +
background). *May be a reflection of sample size, as MCR had more vent
sites* - a small subset of ASVs were found at all vent sites or all
samples. - ASVs within the vents had much higher unique \# of ASVs (not
shared with another habitat type) than any other sample type/location
(furtherest left bars).

### UpsetR at Genus level
Repeat upsetR plot, but summarize at genus level, rather than "species" or "strain"
```{r}
head(insitu_asv_wClass)
```

```{r, fig.width=15, fig.height=10}
alv <- c("Alveolata-Ellobiopsidae", "Alveolata-Perkinsea", "Alveolata-Unknown", "Alveolata-Chrompodellids", "Alveolata-Apicomplexa")
# svg("upsetR-bysite-sampletype-nov2.svg", h=9, w=15)
insitu_asv_wClass %>% 
  filter(SITE %in% all) %>%
  filter(Domain == "Eukaryota") %>% 
  filter(Supergroup != "Opisthokonta") %>% 
  mutate(Supergroup = ifelse(is.na(Supergroup), "Unknown Eukaryota", Supergroup),
         Phylum = ifelse(is.na(Phylum), "Unknown", Phylum),
         Phylum = ifelse(Phylum == "Alveolata_X", "Ellobiopsidae", Phylum),
         Supergroup = ifelse(Supergroup == "Alveolata", paste(Supergroup, Phylum, sep = "-"), Supergroup)) %>% 
  mutate(SUPERGROUP = case_when(
    Supergroup %in% alv ~ "Other Alveolata",
    Supergroup == "Eukaryota_X" ~ "Unknown Eukaryota",
    Phylum == "Cercozoa" ~ "Rhizaria-Cercozoa",
    Phylum == "Radiolaria" ~ "Rhizaria-Radiolaria",
    Phylum == "Ochrophyta" ~ "Stramenopiles-Ochrophyta",
    Phylum == "Opalozoa" ~ "Stramenopiles-Opalozoa",
    Phylum == "Sagenista" ~ "Stramenopiles-Sagenista",
    TRUE ~ Supergroup
  )) %>% 
  # Taxa to supergroup
  mutate(SupergroupPhylum = SUPERGROUP) %>%
  unite(GENUS, Domain:Genus, sep = ";") %>% 
  # Average across replicates
    group_by(GENUS, SAMPLENAME, VENT, SupergroupPhylum) %>% 
    summarise(AVG = mean(value)) %>% 
  ungroup() %>% 
  separate(SAMPLENAME, c("SITE", "SAMPLETYPE", "YEAR", "Sample_tmp"), remove = TRUE) %>% 
  mutate(REGION = case_when(
    SITE == "GordaRidge" ~ "Gorda Ridge",
    SITE %in% mcr ~ "Mid-Cayman Rise",
    SITE == "Axial" ~ "Axial")) %>% 
  mutate(VENTNAME = case_when(
    REGION == "Gorda Ridge" ~ VENT,
    REGION == "Mid-Cayman Rise" ~ paste(SITE, VENT, sep = " "),
    REGION == "Axial" ~ paste(VENT, YEAR, sep = " ")
  )) %>% select(-Sample_tmp) %>% 
  unite(SAMPLE, SITE, SAMPLETYPE, sep = " ", remove = FALSE) %>% 
  group_by(GENUS, SupergroupPhylum, SAMPLE) %>% 
    summarise(SUM = sum(AVG)) %>%
  # filter(SUM > 200) %>% 
  ungroup() %>% 
  distinct(GENUS, SupergroupPhylum, SUM, SAMPLE, .keep_all = TRUE) %>% 
  group_by(GENUS, SupergroupPhylum) %>% 
  summarise(SAMPLE = list(SAMPLE)) %>% 
  ggplot(aes(x = SAMPLE)) +
    geom_bar(color = "black", width = 0.5, aes(fill = SupergroupPhylum)) +
    scale_x_upset(n_intersections = 25) +
  scale_y_continuous(expand = c(0,0)) +
  labs(x = "", y = "Shared at Genus level") +
  theme_linedraw() +
  theme(axis.text.y = element_text(color="black", size=14),
        axis.text.x = element_text(color="black", size=14),
        axis.title = element_text(color="black", size=14),
        legend.text = element_text(color = "black", size = 12),
        plot.margin = margin(1, 1, 1, 5, "cm")) +
  scale_fill_manual(values = c("#fa9fb5", "#c51b8a", "#67000d", "#ef3b2c", "#ffffcc", "#feb24c", "#c7e9b4", "#1d91c0", "#deebf7", "#253494", "#9e9ac8", "#238b45", "#54278f", "#bdbdbd", "#252525", "#fa9fb5", "#c51b8a", "#67000d", "#ef3b2c", "#ffffcc", "#feb24c", "#c7e9b4", "#1d91c0", "#253494", "#9e9ac8", "#238b45", "#54278f", "#bdbdbd", "#252525"))
```

Isolate list of genus level that are shared at all sites.
```{r}
alv <- c("Alveolata-Ellobiopsidae", "Alveolata-Perkinsea", "Alveolata-Unknown", "Alveolata-Chrompodellids", "Alveolata-Apicomplexa")
# svg("upsetR-bysite-sampletype-nov2.svg", h=9, w=15)
shared_genus <- insitu_asv_wClass %>% 
  filter(SITE %in% all) %>%
  filter(Domain == "Eukaryota") %>% 
  filter(Supergroup != "Opisthokonta") %>% 
  mutate(Supergroup = ifelse(is.na(Supergroup), "Unknown Eukaryota", Supergroup),
         Phylum = ifelse(is.na(Phylum), "Unknown", Phylum),
         Phylum = ifelse(Phylum == "Alveolata_X", "Ellobiopsidae", Phylum),
         Supergroup = ifelse(Supergroup == "Alveolata", paste(Supergroup, Phylum, sep = "-"), Supergroup)) %>% 
  mutate(SUPERGROUP = case_when(
    Supergroup %in% alv ~ "Other Alveolata",
    Supergroup == "Eukaryota_X" ~ "Unknown Eukaryota",
    Phylum == "Cercozoa" ~ "Rhizaria-Cercozoa",
    Phylum == "Radiolaria" ~ "Rhizaria-Radiolaria",
    Phylum == "Ochrophyta" ~ "Stramenopiles-Ochrophyta",
    Phylum == "Opalozoa" ~ "Stramenopiles-Opalozoa",
    Phylum == "Sagenista" ~ "Stramenopiles-Sagenista",
    TRUE ~ Supergroup
  )) %>% 
  # Taxa to supergroup
  mutate(SupergroupPhylum = SUPERGROUP) %>%
  unite(GENUS, Domain:Genus, sep = ";") %>% 
  # Average across replicates
    group_by(GENUS, SAMPLENAME, VENT, SupergroupPhylum) %>% 
    summarise(AVG = mean(value)) %>% 
  ungroup() %>% 
  separate(SAMPLENAME, c("SITE", "SAMPLETYPE", "YEAR", "Sample_tmp"), remove = TRUE) %>% 
  mutate(REGION = case_when(
    SITE == "GordaRidge" ~ "Gorda Ridge",
    SITE %in% mcr ~ "Mid-Cayman Rise",
    SITE == "Axial" ~ "Axial")) %>% 
  mutate(VENTNAME = case_when(
    REGION == "Gorda Ridge" ~ VENT,
    REGION == "Mid-Cayman Rise" ~ paste(SITE, VENT, sep = " "),
    REGION == "Axial" ~ paste(VENT, YEAR, sep = " ")
  )) %>% select(-Sample_tmp) %>% 
  unite(SAMPLE, SITE, SAMPLETYPE, sep = " ", remove = FALSE) %>% 
  select(GENUS, SAMPLE, AVG) %>% 
  pivot_wider(names_from = SAMPLE, values_from = "AVG", values_fn = sum) %>%
  drop_na()

# head(shared_genus)
```


### Isolate GR and Axial

```{r}
head(insitu_asv_wClass)
gr_axial_shared <- insitu_asv_wClass %>% 
  filter(SITE_CLASS == "Axial & Gorda Ridge") %>% 
  filter(CLASS == "Vent only") %>% 
  mutate(OCEAN = "Found throughout NE Pacific vents")

length(unique(gr_axial_shared$FeatureID)) 
```

#### Distance matrix / meters - GR-Axial
Extract coordinates from Gorda Ridge and Axial Seamount, calculate distance between each vent site.
```{r}
library(geosphere)

axial_vents <- insitu_asv_wClass %>% 
  filter(SAMPLETYPE == "Vent") %>% 
  filter(SITE == "Axial")

gr_vents <- insitu_asv_wClass %>% 
  filter(SAMPLETYPE == "Vent") %>% 
  filter(SITE == "GordaRidge")

est_distances_gr_axial <- function(df){
  df_out <- df %>% 
    select(VENT, COORDINATES) %>%
    separate(COORDINATES, into = c("lat", "N", "long", "W"), sep = " ") %>%
    mutate(
      LONG_EW = as.numeric(formatC(as.numeric(long), digits = 4, format = "f")),
      LAT = as.numeric(formatC(as.numeric(lat), digits = 4, format = "f")),
           ) %>%
    mutate(LONG = case_when(
      W == "W" ~ (LONG_EW*-1),
      W == "E" ~ LONG_EW,
      W == "" ~ (LONG_EW*-1)
    )) %>%
    select(-lat, -N, -long, -W, -LONG_EW) %>%
    relocate(LONG) %>% 
    distinct(VENT, .keep_all = TRUE) %>%
    column_to_rownames(var = "VENT")
  vents <- row.names(df_out)
  distance_m <- as.data.frame(distm(df_out, fun = distHaversine))
  colnames(distance_m) <- vents
  row.names(distance_m) <- vents
  # Create matrix with distance in meters
  dist_m <- distance_m %>% 
    rownames_to_column(var = "start") %>% 
    pivot_longer(cols = vents, names_to = "end") %>%
    filter(!(start == end))
  # create wide format
  df_out_wide <- df %>%
    filter(Domain == "Eukaryota") %>% 
    # Average across replicates
      group_by(FeatureID, SAMPLENAME, VENT) %>% 
      summarise(AVG = mean(value)) %>% 
      ungroup() %>% 
    separate(SAMPLENAME, c("SITE", "SAMPLETYPE", "YEAR", "Sample_tmp"), remove = TRUE) %>% 
    mutate(REGION = case_when(
      SITE == "GordaRidge" ~ "Gorda Ridge",
      SITE %in% mcr ~ "Mid-Cayman Rise",
      SITE == "Axial" ~ "Axial")) %>% 
    mutate(VENTNAME = case_when(
      REGION == "Gorda Ridge" ~ VENT,
      REGION == "Mid-Cayman Rise" ~ paste(SITE, VENT, sep = " "),
      # REGION == "Axial" ~ paste(VENT, YEAR, sep = " ")
      REGION == "Axial" ~ VENT
    )) %>% select(-Sample_tmp) %>% 
    # unite(VENTNAME, sep = "_", remove = FALSE) %>% 
     group_by(FeatureID, VENT) %>% 
      summarise(SUM = sum(AVG)) %>% 
    pivot_wider(names_from = FeatureID, values_from = SUM, values_fill = 0) %>% 
    column_to_rownames(var = "VENT")
  # Assign row names and calc distance metric
  vents <- row.names(df_out_wide)
  dist_bray <- as.data.frame(as.matrix(vegdist(df_out_wide, method = "bray"))) %>% 
    rownames_to_column(var = "start") %>% 
    pivot_longer(cols = vents, names_to = "end") %>% 
    select(start, end, Bray_Curtis_metric = value)
  dist_jacc <- as.data.frame(as.matrix(vegdist(df_out_wide, method = "jaccard"))) %>% 
    rownames_to_column(var = "start") %>% 
    pivot_longer(cols = vents, names_to = "end") %>% 
    select(start, end, Jaccard_metric = value)
  dist_jacc_clr <- as.data.frame(as.matrix(vegdist(compositions::clr(df_out_wide), method = "euclidean"))) %>% 
    rownames_to_column(var = "start") %>% 
    pivot_longer(cols = vents, names_to = "end") %>% 
    select(start, end, Jaccard_CLR_metric = value)
  dist_ilr_euc <- as.data.frame(as.matrix(dist(compositions::ilr(df_out_wide), method = "euclidean"))) %>% 
    rownames_to_column(var = "start") %>% 
    pivot_longer(cols = vents, names_to = "end") %>% 
    select(start, end, Euclidean_ILR_metric = value)
  # Combine all distance outputs - use dist_m
  dist_compiled_output <- dist_m %>% 
    select(start, end, meters = value) %>% 
    left_join(dist_jacc_clr) %>% 
    left_join(dist_bray) %>% 
    left_join(dist_jacc) %>% 
    left_join(dist_ilr_euc) %>% 
    pivot_longer(cols = ends_with("_metric"), names_to = "comm_dist") %>% 
    filter(!(start == end))
  }

```

Estimate distance matrix among Gorda Ridge and Axial Seamount samples - compare with geographic distance. For simplifying this, Axial seamount samples have been combined.
```{r}
axial_dist <- est_distances_gr_axial(axial_vents)
gr_dist <- est_distances_gr_axial(gr_vents)
```
Compile distances in meters and as distance estimate.
```{r, fig.height=6, fig.width=6}

ggplot(axial_dist, aes(x = meters, y = value)) +
  geom_jitter(stat = "identity") +
  facet_grid(comm_dist ~ ., scales = "free") +
  geom_smooth(method="lm") +
  labs(title = "Axial Seamount", x = "Distance (m)", y = "Metric") +
  theme_linedraw() +
    ggplot(gr_dist, aes(x = meters, y = value)) +
  geom_jitter(stat = "identity") +
  facet_grid(comm_dist ~ ., scales = "free") +
  geom_smooth(method="lm") +
  labs(title = "Gorda Ridge", x = "Distance (m)", y = "Metric") +
  scale_x_log10() + theme_linedraw() +
  patchwork::plot_layout(ncol = 2)

```

#### Tile plot across GR-Axial

Look at most highly represented ASVs within the shared GR-Axial part of the community.
```{r}
# top15 <- function(df, NUM, site, level, taxa, class, category, plot_tax)
# top15(insitu_asv_wClass, 10, all, Phylum, "Ciliophora", CLASS, all_class, Class)

# length(unique(gr_axial_shared$Phylum))
# phylums <- as.character(unique(gr_axial_shared$Phylum))
# top15(gr_axial_shared, 10, all, Phylum, phylums, CLASS, all_class, Class)

# View(gr_axial_shared %>% 
#   group_by(Supergroup, Phylum, Class, Order, Family, Genus) %>% 
#   summarise(SUM = sum(value)) %>% 
#   ungroup() %>% 
#   group_by(Supergroup) %>% 
#   top_n(5, wt = SUM))
  
```


```{r}
tmp2 <- gr_axial_shared %>% 
  filter(value > 25)
hist(tmp2$value)
length(unique(tmp2$FeatureID))

more_than_25 <- as.character(unique(tmp2$FeatureID))
```


```{r, fig.height=15, fig.width=4}
# head(gr_axial_shared)

gr_axial_shared %>% 
  filter(FeatureID %in% more_than_25) %>% 
  filter(!is.na(Phylum)) %>% 
  filter(!(Phylum == "Metazoa")) %>% 
  separate(SAMPLENAME, c("SITE", "SAMPLETYPE", "YEAR", "Sample_tmp"), remove = TRUE) %>% 
    mutate(REGION = case_when(
      SITE == "GordaRidge" ~ "Gorda Ridge",
      SITE %in% mcr ~ "Mid-Cayman Rise",
      SITE == "Axial" ~ "Axial")) %>% 
    mutate(VENTNAME = case_when(
      REGION == "Gorda Ridge" ~ VENT,
      REGION == "Mid-Cayman Rise" ~ paste(SITE, VENT, sep = " "),
      REGION == "Axial" ~ paste(VENT, YEAR, sep = " ")
      # REGION == "Axial" ~ VENT
    )) %>% select(-Sample_tmp) %>% 
  group_by(SITE, YEAR, VENTNAME, FeatureID, Taxon, Domain, Supergroup, Phylum, Class) %>% 
    summarise(SUM = sum(value)) %>% 
  ungroup() %>% 
  mutate(BIN = ifelse(SUM > 0, 1, NA)) %>% 
  # pivot_wider(names_from = SAMPLENAME, values_from = BIN) %>% 
  ggplot(aes(x = VENTNAME, y = FeatureID, fill = BIN)) +
  geom_tile(fill = "grey", color = "black") +
  theme_minimal() +
  facet_grid(Phylum ~ SITE + YEAR, space = "free", scales = "free") +
  theme(axis.text.y = element_blank(),
        axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5),
        strip.text.y = element_text(angle = 0, hjust = 0, vjust = 0.5))
  
  # ?ifelse()
```



### Isolate MCR sites



Characterize putative endemic ASVs within MCR only.
```{r}
unique(insitu_asv_wClass$SITE_CLASS)
vd_picc_shared <- insitu_asv_wClass %>% 
  filter(SITE_CLASS == "Piccard & Von Damm") %>% 
  filter(CLASS == "Vent only") %>% 
  mutate(OCEAN = "Found throughout MCR vents")
length(unique(vd_picc_shared$FeatureID))
table(vd_picc_shared$Supergroup)
table(gr_axial_shared$Supergroup)
```
#### Distance matrix/meters - MCR
Isolate samples from Von Damm and Piccard vent sites only.
```{r}
vd_picc_vents <- insitu_asv_wClass %>% 
  filter(SAMPLETYPE == "Vent") %>% 
  filter(SITE == "VonDamm" | SITE == "Piccard") %>% 
  mutate(coord = case_when(
    VENT == "ShrimpHole" ~ "18.374893, -81.797441",
    TRUE ~ COORDINATES
  )) %>% 
  select(-COORDINATES) %>% 
  select(everything(), COORDINATES = coord)
  
# View(vd_picc_vents)
picc_vents <- insitu_asv_wClass %>% 
  filter(SAMPLETYPE == "Vent") %>% 
  filter(SITE == "Piccard")

vd_vents <- insitu_asv_wClass %>% 
  filter(SAMPLETYPE == "Vent") %>% 
  filter(SITE == "VonDamm") %>% 
  mutate(coord = case_when(
    VENT == "ShrimpHole" ~ "18.374893, -81.797441",
    TRUE ~ COORDINATES
  )) %>% 
  select(-COORDINATES) %>% 
  select(everything(), COORDINATES = coord)

est_distances_mcr <- function(df){
  df_out <- df %>% 
  select(VENT, COORDINATES) %>%
  separate(COORDINATES, into = c("lat", "long"), sep = ", ", convert = TRUE) %>%
  distinct(VENT, .keep_all = TRUE) %>%
  relocate(LONG = long, LAT = lat) %>%
  column_to_rownames(var = "VENT")
  vents <- row.names(df_out)
  distance_m <- as.data.frame(distm(df_out, fun = distHaversine))
  colnames(distance_m) <- vents
  row.names(distance_m) <- vents
  # Create matrix with distance in meters
  dist_m <- distance_m %>% 
    rownames_to_column(var = "start") %>% 
    pivot_longer(cols = vents, names_to = "end") %>%
    filter(!(start == end))
  # create wide format
  df_out_wide <- df %>%
    filter(Domain == "Eukaryota") %>% 
    # Average across replicates
      group_by(FeatureID, SAMPLENAME, VENT) %>% 
      summarise(AVG = mean(value)) %>% 
      ungroup() %>% 
    separate(SAMPLENAME, c("SITE", "SAMPLETYPE", "YEAR", "Sample_tmp"), remove = TRUE) %>% 
    mutate(REGION = case_when(
      SITE == "GordaRidge" ~ "Gorda Ridge",
      SITE %in% mcr ~ "Mid-Cayman Rise",
      SITE == "Axial" ~ "Axial")) %>% 
    mutate(VENTNAME = case_when(
      REGION == "Gorda Ridge" ~ VENT,
      REGION == "Mid-Cayman Rise" ~ paste(SITE, VENT, sep = " "),
      # REGION == "Axial" ~ paste(VENT, YEAR, sep = " ")
      REGION == "Axial" ~ VENT
    )) %>% select(-Sample_tmp) %>% 
    # unite(VENTNAME, sep = "_", remove = FALSE) %>% 
     group_by(FeatureID, VENT) %>% 
      summarise(SUM = sum(AVG)) %>% 
    pivot_wider(names_from = FeatureID, values_from = SUM, values_fill = 0) %>% 
    column_to_rownames(var = "VENT")
  # Assign row names and calc distance metric
  vents <- row.names(df_out_wide)
  dist_bray <- as.data.frame(as.matrix(vegdist(df_out_wide, method = "bray"))) %>% 
    rownames_to_column(var = "start") %>% 
    pivot_longer(cols = vents, names_to = "end") %>% 
    select(start, end, Bray_Curtis_metric = value)
  dist_jacc <- as.data.frame(as.matrix(vegdist(df_out_wide, method = "jaccard"))) %>% 
    rownames_to_column(var = "start") %>% 
    pivot_longer(cols = vents, names_to = "end") %>% 
    select(start, end, Jaccard_metric = value)
  dist_jacc_clr <- as.data.frame(as.matrix(vegdist(compositions::clr(df_out_wide), method = "euclidean"))) %>% 
    rownames_to_column(var = "start") %>% 
    pivot_longer(cols = vents, names_to = "end") %>% 
    select(start, end, Jaccard_CLR_metric = value)
  dist_ilr_euc <- as.data.frame(as.matrix(dist(compositions::ilr(df_out_wide), method = "euclidean"))) %>% 
    rownames_to_column(var = "start") %>% 
    pivot_longer(cols = vents, names_to = "end") %>% 
    select(start, end, Euclidean_ILR_metric = value)
  # Combine all distance outputs - use dist_m
  dist_compiled_output <- dist_m %>% 
    select(start, end, meters = value) %>% 
    left_join(dist_jacc_clr) %>% 
    left_join(dist_bray) %>% 
    left_join(dist_jacc) %>% 
    left_join(dist_ilr_euc) %>% 
    pivot_longer(cols = ends_with("_metric"), names_to = "comm_dist") %>% 
    filter(!(start == end))
  }

```

Calculate distance matrix for MCR samples
```{r}
vd_vents_dist <- est_distances_mcr(vd_vents)
picc_vents_dist <- est_distances_mcr(picc_vents)
vd_picc_dist <- est_distances_mcr(vd_picc_vents)
# View(vd_vents_dist)
# View(vd_vents)
```

Something is off about the Shrimp Hole lat/long?
```{r, fig.height=6, fig.width=6}
# head(vd_vents_dist)
ggplot(vd_vents_dist, aes(x = meters, y = value)) +
  geom_jitter(stat = "identity") +
  facet_grid(comm_dist ~ ., scales = "free") +
  geom_smooth(method="lm") +
  labs(title = "Von Damm", x = "Distance (m)", y = "Metric") +
  theme_linedraw() +
    ggplot(picc_vents_dist, aes(x = meters, y = value)) +
  geom_jitter(stat = "identity") +
  facet_grid(comm_dist ~ ., scales = "free") +
  geom_smooth(method="lm") +
  labs(title = "Piccard", x = "Distance (m)", y = "Metric") +
  theme_linedraw() +
  patchwork::plot_layout(ncol = 2)

# ggplot(vd_picc_dist, aes(x = meters, y = value)) +
  # geom_jitter(stat = "identity") +
  # facet_grid(comm_dist ~ ., scales = "free")
# View(vd_picc_dist)
```

#### Tile plot across Picc-VD

Look at top ASVs within Piccard and Von Damm
```{r}
# top15 <- function(df, NUM, site, level, taxa, class, category, plot_tax)
# top15(insitu_asv_wClass, 10, all, Phylum, "Ciliophora", CLASS, all_class, Class)

length(unique(vd_picc_shared$Phylum))
phylums <- as.character(unique(vd_picc_shared$Phylum))
top15(vd_picc_shared, 10, all, Phylum, phylums, CLASS, all_class, Class)

# View(vd_picc_shared %>%
#   group_by(Supergroup, Phylum, Class, Order, Family, Genus) %>%
#   summarise(SUM = sum(value)) %>%
#   ungroup() %>%
#   group_by(Supergroup) %>%
#   top_n(5, wt = SUM))
```


```{r}
head(vd_picc_shared)
length(unique(vd_picc_shared$FeatureID))

tmp3 <- vd_picc_shared %>% 
  filter(value > 25)
length(unique(tmp3$FeatureID))
more_than_25 <- as.character(unique(tmp3$FeatureID))
```


```{r, fig.width=4, fig.height=18}
vd_picc_shared %>% 
  filter(FeatureID %in% more_than_25) %>% 
  filter(!is.na(Phylum)) %>% 
  filter(!(Phylum == "Metazoa")) %>% 
  separate(SAMPLENAME, c("SITE", "SAMPLETYPE", "YEAR", "Sample_tmp"), remove = TRUE) %>% 
    mutate(REGION = case_when(
      SITE == "GordaRidge" ~ "Gorda Ridge",
      SITE %in% mcr ~ "Mid-Cayman Rise",
      SITE == "Axial" ~ "Axial")) %>% 
    mutate(VENTNAME = case_when(
      REGION == "Gorda Ridge" ~ VENT,
      REGION == "Mid-Cayman Rise" ~ paste(SITE, VENT, sep = " "),
      REGION == "Axial" ~ paste(VENT, YEAR, sep = " ")
      # REGION == "Axial" ~ VENT
    )) %>% select(-Sample_tmp) %>% 
  group_by(SITE, YEAR, VENTNAME, FeatureID, Taxon, Domain, Supergroup, Phylum, Class) %>% 
    summarise(SUM = sum(value)) %>% 
  ungroup() %>% 
  mutate(BIN = ifelse(SUM > 0, 1, NA)) %>% 
  # pivot_wider(names_from = SAMPLENAME, values_from = BIN) %>% 
  ggplot(aes(x = VENTNAME, y = FeatureID, fill = BIN)) +
  geom_tile(fill = "grey", color = "black") +
  theme_minimal() +
  facet_grid(Phylum ~ SITE + YEAR, space = "free", scales = "free") +
  theme(axis.text.y = element_blank(),
        axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5),
        strip.text.y = element_text(angle = 0, hjust = 0, vjust = 0.5))
  
```


## Top 15 ASVs: all
```{r}
# Function usage:
# top15 <- function(df, NUM, site, level, taxa, class, category, plot_tax)
top15(insitu_asv_wClass, 15, all, Domain, "Eukaryota", CLASS, all_class, Phylum)
```
```{r}
# top15 <- function(df, NUM, site, level, taxa, class, category, plot_tax)
top15(insitu_asv_wClass, 15, all, Domain, "Eukaryota", SITE_CLASS, "All sites", Phylum)
```

# Get overall sequence and ASV stats

## All samples
Out of all the ASVs and sequences, what percentage of ASVs were found at all 4 of my vent sites? what % of ASVs were unique to individual sites? what % of sequences?
```{r}
totalasvs <- length(unique(insitu_asv_wClass$FeatureID)); totalasvs
totalseq <- sum(insitu_asv_wClass$value); totalseq

unique(insitu_asv_wClass$SITE_CLASS)
tmp_unique <- filter(insitu_asv_wClass, grepl(" only", SITE_CLASS))
tmp_shared <- filter(insitu_asv_wClass, !(grepl(" only", SITE_CLASS)))
tmp_allshared <- filter(insitu_asv_wClass, SITE_CLASS == "All sites")

# Total number of ASVs that were found only at an individual vent site
a <- length(unique(tmp_unique$FeatureID)); a
100*(a/totalasvs)

a_sum <- sum(tmp_unique$value); a_sum
100*(a_sum/totalseq)


# Total number of ASVs that were found at more than 1 site
b <- length(unique(tmp_shared$FeatureID)); b
100*(b/totalasvs)

b_sum <- sum(tmp_shared$value); b_sum
100*(b_sum/totalseq)

# Total number of ASVs designated to be found at all sites
c <- length(unique(tmp_allshared$FeatureID)); c
100*(c/totalasvs)

c_sum <- sum(tmp_allshared$value); c_sum
100*(c_sum/totalseq)
```

## Resident
From the vent-only ASVs, what percentage of them appear only at 1 site? what percent also appear at other vent sites?
```{r}
resident_only <- filter(insitu_asv_wClass, CLASS == "Vent only")

totalasvs <- length(unique(resident_only$FeatureID)); totalasvs
totalseq <- sum(resident_only$value); totalseq

# resident includes
100*(totalasvs/length(unique(insitu_asv_wClass$FeatureID)))
100*(totalseq/length(unique(insitu_asv_wClass$FeatureID)))
  # unique(resident_only$SITE_CLASS)
tmp_unique <- filter(resident_only, grepl(" only", SITE_CLASS))
tmp_shared <- filter(resident_only, !(grepl(" only", SITE_CLASS)))
tmp_allshared <- filter(resident_only, SITE_CLASS == "All sites")

# Of the ASVs found ONLY within diffuse venting fluid, just over 90% were unique to only the individual vent site - makes up 53% of the vent only sequences
a <- length(unique(tmp_unique$FeatureID)); a
100*(a/totalasvs)

a_sum <- sum(tmp_unique$value); a_sum
100*(a_sum/totalseq)


# Of the ASVs from vent only, 9.6% were also found at other vent sites, totals to 46% of the sequences.
b <- length(unique(tmp_shared$FeatureID)); b
100*(b/totalasvs)

b_sum <- sum(tmp_shared$value); b_sum 
100*(b_sum/totalseq)

# Total number of ASVs designated to be found at all sites
c <- length(unique(tmp_allshared$FeatureID)); c
100*(c/totalasvs)

c_sum <- sum(tmp_allshared$value); c_sum
100*(c_sum/totalseq)

vent_only_allsites <- tmp_allshared
# View(vent_only_allsites)
# unique(vent_only_allsites$Taxon)
# unique(vent_only_allsites$FeatureID)
```
### View top 10 resident 
```{r}
# top15 <- function(df, NUM, site, level, taxa, class, category, plot_tax)
top15(resident_only, 10, all, Domain, "Eukaryota", CLASS, all_class, Phylum)
```

## Cosmopolitan
Isolate ASVs that were found in all habitat types! (Vent, plume, & background)

```{r}
# unique(insitu_asv_wClass$CLASS)
cosmo <- c("Vent, plume, & background", "Vent & background", "Vent & plume", "Plume & background")
# cosmo <- c("Vent, plume, & background")
cosmo_only <- filter(insitu_asv_wClass, CLASS %in% cosmo)

totalasvs <- length(unique(cosmo_only$FeatureID)); totalasvs
totalseq <- sum(cosmo_only$value); totalseq

100*(totalasvs/length(unique(insitu_asv_wClass$FeatureID)))
100*(totalseq/sum(insitu_asv_wClass$value))

# unique(cosmo_only$SITE_CLASS)
tmp_unique <- filter(cosmo_only, grepl(" only", SITE_CLASS))
tmp_shared <- filter(cosmo_only, !(grepl(" only", SITE_CLASS)))
tmp_allshared <- filter(cosmo_only, SITE_CLASS == "All sites")

# Of the ASVs found in all habitat types (vent, plume, and background) - 22% of them were unique to a given site, but found throughout, consisting of 7.6%
a <- length(unique(tmp_unique$FeatureID)); a
100*(a/totalasvs)

a_sum <- sum(tmp_unique$value); a_sum
100*(a_sum/totalseq)


# Of the ASVs found in all habitat types, 77% of them were also found at another vent site, consisting of 92% of the sequences
b <- length(unique(tmp_shared$FeatureID)); b
100*(b/totalasvs)

b_sum <- sum(tmp_shared$value); b_sum 
100*(b_sum/totalseq)

# Of the ASVS found in all habitat types, 16% of them were found at all vent sites, which made up 40% of the sequences.
c <- length(unique(tmp_allshared$FeatureID)); c
100*(c/totalasvs)

c_sum <- sum(tmp_allshared$value); c_sum
100*(c_sum/totalseq)
```

### View top 10 cosmopolitan 
```{r}
# top15 <- function(df, NUM, site, level, taxa, class, category, plot_tax)
top15(cosmo_only, 10, all, Domain, "Eukaryota", CLASS, all_class, Phylum)
```

# Compare dist metrics of resident vs. cosmopolitan

```{r}
# cosmo <- c("Vent, plume, & background", "Vent & background", "Vent & plume", "Plume & background")
cosmo <- c("Vent, plume, & background")
# cosmo_only <- filter(insitu_asv_wClass, CLASS %in% cosmo)
res <- c("Vent only")
```


```{r, fig.height=8, fig.width=8}
isolate_calc_jaccard <- function(df, slice){
  df_out_wide <- df %>%
    filter(Domain == "Eukaryota") %>% 
    filter(CLASS %in% slice) %>% 
    # Average across replicates
      group_by(FeatureID, SAMPLENAME, VENT) %>% 
      summarise(AVG = mean(value)) %>% 
      ungroup() %>% 
    separate(SAMPLENAME, c("SITE", "SAMPLETYPE", "YEAR", "Sample_tmp"), remove = FALSE) %>% 
    mutate(REGION = case_when(
      SITE == "GordaRidge" ~ "Gorda Ridge",
      SITE %in% mcr ~ "Mid-Cayman Rise",
      SITE == "Axial" ~ "Axial")) %>% 
    mutate(VENTNAME = case_when(
      REGION == "Gorda Ridge" ~ paste(SITE, VENT, sep = " "),
      REGION == "Mid-Cayman Rise" ~ paste(SITE, VENT, sep = " "),
      REGION == "Axial" ~ paste(SITE, VENT, YEAR, sep = " ")
      # REGION == "Axial" ~ VENT
    )) %>% select(-Sample_tmp) %>% 
    unite(SAMPLEID, SITE, VENT, YEAR, SAMPLETYPE, sep = "-", remove = FALSE) %>%
    unite(sample_full, SAMPLEID, VENTNAME, sep = "--") %>% 
     group_by(FeatureID, sample_full) %>% 
      summarise(SUM = sum(AVG)) %>% 
    pivot_wider(names_from = FeatureID, values_from = SUM, values_fill = 0) %>% 
    column_to_rownames(var = "sample_full")
  }

resident <- isolate_calc_jaccard(insitu_asv_wClass, res)
cosmopolitan <- isolate_calc_jaccard(insitu_asv_wClass, cosmo)


# dist_jacc <- as.data.frame(as.matrix(vegdist(df_out_wide, method = "jaccard"))) %>% 
  #   rownames_to_column(var = "start") %>% 
  #   pivot_longer(cols = vents, names_to = "end") %>% 
  #   select(start, end, Jaccard_metric = value)

  # Assign row names and calc distance metric
  # vents <- row.names(df_out_wide)
  # dist_bray <- as.data.frame(as.matrix(vegdist(df_out_wide, method = "bray"))) %>%
  #   rownames_to_column(var = "start") %>%
  #   pivot_longer(cols = vents, names_to = "end") %>%
  #   select(start, end, Bray_Curtis_metric = value) %>% 
  #   distinct()

dist_jacc_clr_res <- as.data.frame(as.matrix(vegdist(compositions::clr(resident), method = "euclidean"))) %>%
# dist_jacc_clr_res <- as.data.frame(as.matrix(vegdist(resident, method = "bray"))) %>% 
    rownames_to_column(var = "start") %>% 
    pivot_longer(cols = row.names(resident), names_to = "end") %>% 
    select(start, end, Jaccard_CLR_metric = value) %>% 
    mutate(VARS = purrr::map2_chr(start, end, ~toString(sort(c(.x, .y))))) %>%
    distinct(VARS, .keep_all = TRUE) %>%
    separate(VARS, c("X", "Y"), sep = ", ") %>% 
    select(-start, -end) %>% 
    add_column(SUBSET = "RESIDENT")
# range(dist_jacc_clr_res$Jaccard_CLR_metric)
# ggplot(dist_jacc_clr_res, aes(x = X, y = Y, fill = Jaccard_CLR_metric)) +
#   geom_tile() +
#   theme_minimal() +
#   theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5))

dist_jacc_clr_cos <- as.data.frame(as.matrix(vegdist(compositions::clr(cosmopolitan), method = "euclidean"))) %>%
# dist_jacc_clr_cos <- as.data.frame(as.matrix(vegdist(cosmopolitan, method = "bray"))) %>% 
    rownames_to_column(var = "start") %>% 
    pivot_longer(cols = row.names(cosmopolitan), names_to = "end") %>% 
    select(start, end, Jaccard_CLR_metric = value) %>% 
    mutate(VARS = purrr::map2_chr(start, end, ~toString(sort(c(.x, .y))))) %>%
    distinct(VARS, .keep_all = TRUE) %>%
  # Flip the Y and X variables for cosmopolitan - will appear at bottom
    separate(VARS, c("Y", "X"), sep = ", ") %>% 
    select(-start, -end) %>% 
    add_column(SUBSET = "COSMOPOLITAN")

###

joined_dist <- dist_jacc_clr_cos %>% 
  rbind(dist_jacc_clr_res) %>% 
  separate(Y, c("Y_FULL", "Y"), sep = "--") %>% 
    separate(Y_FULL, c("Y_SITE", "Y_VENT", "Y_YEAR"), sep = "-") %>% 
  separate(X, c("X_FULL", "X"), sep = "--") %>% 
    separate(X_FULL, c("X_SITE", "X_VENT", "X_YEAR"), sep = "-")
# head(joined_dist)
```

```{r, fig.height=8, fig.width=8}
ggplot(joined_dist, aes(x = X, y = Y, fill = Jaccard_CLR_metric)) +
  geom_tile(color = "#474440") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5),
        axis.title = element_blank()) +
  scale_fill_gradientn(colors = c("#badbdb", "#dedad2", "#e4bcad", "#df979e", "#d7658b", "#c80064")) +
  coord_fixed(ratio = 1)
  # facet_grid(Y_SITE ~ X_SITE, scale = "free", space = "free")
```
Look at lowest Jaccard metric among samples that are from separate sites.
```{r}
head(joined_dist)

# Across site similarities
# View(joined_dist %>% 
#   # Isolate pairwise comparisons from with each site
#   filter(X_SITE != Y_SITE) %>% 
#   # Isolate resident
#   filter(SUBSET == "RESIDENT") %>% 
#   arrange(Jaccard_CLR_metric))
```


# Plot ASV prevalence by sequence?
```{r Create taxonomy key for ASV data}
# Not sure if I need this
tax_key <- insitu_asv_wClass %>% 
  select(FeatureID, Taxon, Domain, Supergroup, Phylum, Class, Order, Family, Genus, Species, CLASS, SITE_CLASS) %>% 
  distinct()
```

```{r}
# head(insitu_asv_wClass)
alv <- c("Alveolata-Ellobiopsidae", "Alveolata-Perkinsea", "Alveolata-Unknown", "Alveolata-Chrompodellids", "Alveolata-Apicomplexa")

tmp <- insitu_asv_wClass %>% 
  filter(Domain == "Eukaryota") %>% 
  filter(!is.na(Supergroup)) %>% 
  # Average across replicates
    group_by(FeatureID, SAMPLENAME, VENT) %>% 
    summarise(AVG = mean(value)) %>% 
  ungroup() %>% 
  separate(SAMPLENAME, c("SITE", "SAMPLETYPE", "YEAR", "Sample_tmp"), remove = TRUE) %>% 
  mutate(REGION = case_when(
    SITE == "GordaRidge" ~ "Gorda Ridge",
    SITE %in% mcr ~ "Mid-Cayman Rise",
    SITE == "Axial" ~ "Axial")) %>% 
  mutate(VENTNAME = case_when(
    REGION == "Gorda Ridge" ~ VENT,
    REGION == "Mid-Cayman Rise" ~ paste(SITE, VENT, sep = " "),
    REGION == "Axial" ~ paste(VENT, YEAR, sep = " ")
  )) %>% select(-Sample_tmp) %>% 
  unite(SAMPLE, REGION, VENTNAME, sep = " ", remove = FALSE) %>% 
  group_by(FeatureID, SAMPLE) %>% 
  summarise(SUM = sum(AVG)) %>% 
  pivot_wider(names_from = "SAMPLE", values_from = SUM, values_fill = 0) %>% 
  column_to_rownames(var = "FeatureID") %>%
  mutate(PREVALENCE = rowSums(. > 0),
         SEQ_TOTAL = rowSums(.)) %>% 
  rownames_to_column(var = "FeatureID") %>% 
  left_join(tax_key) %>% 
   filter(Supergroup != "Opisthokonta") %>% 
  mutate(Supergroup = ifelse(is.na(Supergroup), "Unknown Eukaryota", Supergroup),
         Phylum = ifelse(is.na(Phylum), "Unknown", Phylum),
         Phylum = ifelse(Phylum == "Alveolata_X", "Ellobiopsidae", Phylum),
         Supergroup = ifelse(Supergroup == "Alveolata", paste(Supergroup, Phylum, sep = "-"), Supergroup)) %>% 
  mutate(SUPERGROUP = case_when(
    Supergroup %in% alv ~ "Other Alveolata",
    Supergroup == "Eukaryota_X" ~ "Unknown Eukaryota",
    Phylum == "Cercozoa" ~ "Rhizaria-Cercozoa",
    Phylum == "Radiolaria" ~ "Rhizaria-Radiolaria",
    Phylum == "Ochrophyta" ~ "Stramenopiles-Ochrophyta",
    Phylum == "Opalozoa" ~ "Stramenopiles-Opalozoa",
    Phylum == "Sagenista" ~ "Stramenopiles-Sagenista",
    TRUE ~ Supergroup
  ))
```

```{r, fig.height=7, fig.width=9}
# head(tmp)
tmp %>% 
  filter(SEQ_TOTAL > 0) %>% 
  ggplot(aes(x = PREVALENCE, y = SEQ_TOTAL, fill = SUPERGROUP)) +
  geom_jitter(stat = "identity", shape = 21) +
  scale_y_log10() +
  facet_wrap(SUPERGROUP ~ .) +
  scale_fill_manual(values = c("#fa9fb5", "#c51b8a", "#67000d", "#ef3b2c", "#ffffcc", "#feb24c", "#c7e9b4", "#1d91c0", "#deebf7", "#253494", "#9e9ac8", "#238b45", "#54278f", "#bdbdbd", "#252525", "#fa9fb5", "#c51b8a", "#67000d", "#ef3b2c", "#ffffcc", "#feb24c", "#c7e9b4", "#1d91c0", "#253494", "#9e9ac8", "#238b45", "#54278f", "#bdbdbd", "#252525")) +
  theme_linedraw() +
  labs(x = "Number of samples ASV appears in", y = "Total sequences (log)")
```


# Compare resident vs. cosmopolitan taxa

```{r view categories}
head(insitu_asv_wClass) # from above, where I've classified each ASV by site and occurence in sample type
unique(insitu_asv_wClass$CLASS)
unique(insitu_asv_wClass$SITE_CLASS)
unique(insitu_asv_wClass$SAMPLETYPE)
```

```{r}
tmp <- (insitu_asv_wClass %>% 
          filter(DATASET == "MCR") %>% 
       group_by(CLASS) %>% 
       summarise(SEQ = sum(value),
                      COUNT = n()))
tmp
# Vent only
587304/sum(tmp$SEQ) #33%
3953/sum(tmp$COUNT)
# Cosmo
954469/sum(tmp$SEQ)
3006/sum(tmp$COUNT)
```

33% of sequences were vent-only 42% of ASVs were vent-only

45% of sequences were cosmopolitan 32% of ASVs were cosmopolitan

> What Supergroups are associated with resident vs. endemic? what about
> to specific sites?

```{r}
make_bar_bycategory <- function(df, category, position){
  CATEGORY <- enquo(category)
  df_out <- df %>% 
  filter(Domain == "Eukaryota") %>%
  mutate(Supergroup = ifelse(is.na(Supergroup), "Unknown Eukaryota", Supergroup),
         Phylum = ifelse(is.na(Phylum), "Unknown", Phylum),
         Phylum = ifelse(Phylum == "Alveolata_X", "Ellobiopsidae", Phylum),
         Supergroup = ifelse(Supergroup == "Alveolata", paste(Supergroup, Phylum, sep = "-"), Supergroup)) %>% 
  group_by(FeatureID, Taxon, Domain, Supergroup, Phylum, Class, Order, Family, Genus, Species,
           VENT, SITE, SAMPLETYPE, YEAR, DATASET, !!CATEGORY) %>% 
    summarise(SEQ_AVG_REP = mean(value)) %>% 
  ungroup()
  ## Supergroup
  supergroup <- df_out %>% 
  group_by(Supergroup, !!CATEGORY) %>% 
    summarise(SEQ_SUM = sum(SEQ_AVG_REP)) %>% 
  ggplot(aes(x = !!CATEGORY, y = SEQ_SUM, fill = Supergroup)) +
    geom_bar(stat = "identity", position = position, color = "black", width = 0.9) +
    # facet_grid(. ~ SITE +YEAR + SAMPLETYPE, scale = "free", space = "free") +
  theme_linedraw() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1),
        strip.background = element_blank(), strip.text = element_text(color = "black"),
        legend.position = "right") +
  scale_y_continuous(expand = c(0,0)) +
  # scale_fill_brewer(palette = "Set2") +
  scale_fill_manual(values = c("#fa9fb5", "#c51b8a", "#67000d", "#ef3b2c", "#ffffcc", "#feb24c", "#c7e9b4", "#1d91c0", "#253494", "#9e9ac8", "#238b45", "#54278f", "#bdbdbd", "#252525", "#fa9fb5", "#c51b8a", "#67000d", "#ef3b2c", "#ffffcc", "#feb24c", "#c7e9b4", "#1d91c0", "#253494", "#9e9ac8", "#238b45", "#54278f", "#bdbdbd", "#252525")) +
  labs(x = "", y = "Relative abundance")
  ## Phylum
  phylum <- df_out %>% 
    unite(SupergroupPhylum, Supergroup, Phylum, sep = "-") %>% 
  group_by(SupergroupPhylum, !!CATEGORY) %>% 
    summarise(SEQ_SUM = sum(SEQ_AVG_REP)) %>% 
  ggplot(aes(x = !!CATEGORY, y = SEQ_SUM, fill = SupergroupPhylum)) +
    geom_bar(stat = "identity", position = position, color = "black", width = 0.9) +
    # facet_grid(. ~ SITE +YEAR + SAMPLETYPE, scale = "free", space = "free") +
  theme_linedraw() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1),
        strip.background = element_blank(), strip.text = element_text(color = "black"),
        legend.position = "right") +
  scale_y_continuous(expand = c(0,0)) +
  scale_fill_manual(values = c("#f1eef6", "#d7b5d8", "#df65b0", "#ce1256", "#fc9272", "#ef3b2c", 
    "#800026", "#fff7bc", "#fec44f", "#d95f0e", "#a63603", "#74c476", "#238b45", 
    "#00441b", "#7fcdbb", "#084081", "#c6dbef", "#2b8cbe", "#016c59", "#bcbddc", 
    "#807dba", "#54278f", "#bdbdbd", "black", "white", "#969696", "#525252", "#f1eef6", "#d7b5d8", "#df65b0", "#ce1256", "#fc9272", "#ef3b2c", 
    "#800026", "#fff7bc", "#fec44f", "#d95f0e", "#a63603", "#74c476", "#238b45", 
    "#00441b", "#7fcdbb", "#084081", "#c6dbef", "#2b8cbe", "#016c59", "#bcbddc", 
    "#807dba", "#54278f", "#bdbdbd", "black", "white")) +
  labs(x = "", y = "Relative abundance")
  supergroup + phylum + patchwork::plot_layout(ncol = 1)
}
```

```{r}
make_tile_bycategory <- function(df, category, position){
  CATEGORY <- enquo(category)
  df_out <- df %>% 
  filter(Domain == "Eukaryota") %>%
  mutate(Supergroup = ifelse(is.na(Supergroup), "Unknown Eukaryota", Supergroup),
         Phylum = ifelse(is.na(Phylum), "Unknown", Phylum),
         Phylum = ifelse(Phylum == "Alveolata_X", "Ellobiopsidae", Phylum),
         Supergroup = ifelse(Supergroup == "Alveolata", paste(Supergroup, Phylum, sep = "-"), Supergroup)) %>% 
  group_by(FeatureID, Taxon, Domain, Supergroup, Phylum, Class, Order, Family, Genus, Species,
           VENT, SITE, SAMPLETYPE, YEAR, DATASET, !!CATEGORY) %>% 
    summarise(SEQ_AVG_REP = mean(value)) %>% 
  ungroup()
  ## Supergroup
  supergroup <- df_out %>% 
  group_by(Supergroup, !!CATEGORY) %>% 
    summarise(SEQ_SUM = sum(SEQ_AVG_REP)) %>% 
  ggplot(aes(x = !!CATEGORY, fill = log(SEQ_SUM), y = Supergroup)) +
    geom_tile(color = "black") +
    # facet_grid(. ~ SITE +YEAR + SAMPLETYPE, scale = "free", space = "free") +
  theme_linedraw() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1),
        strip.background = element_blank(), strip.text = element_text(color = "black"),
        legend.position = "right") +
  scale_fill_gradient(low = "#ffeda0", high = "#e31a1c", na.value = "grey50") +
  labs(x = "Distribution", y = "")
  supergroup
}
```

```{r, fig.height=4, fig.width=4}
make_tile_bycategory(insitu_asv_wClass, CLASS, "fill")
```

```{r, fig.height=12, fig.width=10}
# make_bar_bycategory(insitu_asv_wClass, CLASS, "fill")
```

```{r, fig.height=12, fig.width=10}
make_bar_bycategory(insitu_asv_wClass, CLASS, "stack")
```

```{r, fig.height=12, fig.width=10}
make_bar_bycategory(insitu_asv_wClass, SITE_CLASS, "fill")
make_tile_bycategory(insitu_asv_wClass, SITE_CLASS, "fill")
```

```{r, fig.height=12, fig.width=10}
make_bar_bycategory(insitu_asv_wClass, SITE_CLASS, "stack")
```

```{r}
categories <- c("Vent only", "Vent, plume, & background")
insitu_asv_wClass %>% 
  filter(CLASS %in% categories) %>% 
  mutate(CAT = case_when(
    CLASS == "Vent only" ~ "Resident",
    TRUE ~ "Cosmopolitan"
  )) %>% 
  group_by(CAT) %>% 
  summarise(SUM = sum(value),
            COUNT = n()) %>% 
  pivot_longer(c(SUM, COUNT)) %>% 
ggplot(aes(x = name, y = value, fill = CAT))+
  geom_bar(stat = "identity", position = "dodge", color = "black") +
  theme_linedraw() +
  facet_grid(name ~ ., scales = "free")+
  scale_fill_brewer(palette = "Dark2") +
  theme(legend.title = element_blank()) +
  labs(x = "", y = "Total number of ASVs")

```

## Isolate shared ASVs among all

Isolate shared ASVs that are putative endemic taxa
```{r}
all_vents <- insitu_asv_wClass %>% 
  filter(SITE_CLASS == "All sites" & grepl("Vent", CLASS)) %>% 
  filter(SAMPLETYPE == "Vent")
# View(all_vents)
# head(insitu_asv_wClass)
# dim(all_vents)
```

Presence/absence of ASVs to the Class level across all vents.

```{r, fig.width=8, fig.height=14}
# svg("pa-ventsites.svg", w=8, h=12)
insitu_asv_wClass %>% 
  filter(grepl("Vent", CLASS)) %>% 
  filter(Domain == "Eukaryota") %>% 
  filter(SAMPLETYPE == "Vent") %>% 
  filter(!is.na(Class)) %>% 
  filter(Phylum != "Stramenopiles_X" & Phylum != "Pseudofungi" & Phylum != "Chrompodellids") %>%
  filter(Supergroup != "Opisthokonta") %>% 
  mutate(Supergroup = ifelse(is.na(Supergroup), "Unknown Eukaryota", Supergroup),
         Phylum = ifelse(is.na(Phylum), "Unknown", Phylum),
         Phylum = ifelse(Phylum == "Alveolata_X", "Ellobiopsidae", Phylum),
         Supergroup = ifelse(Supergroup == "Alveolata", paste(Supergroup, Phylum, sep = "-"), Supergroup)) %>% 
  group_by(FeatureID, Taxon, Domain, Supergroup, Phylum, Class, Order, Family, Genus, Species,
           VENT, SITE, SAMPLETYPE, YEAR, DATASET) %>% 
    summarise(SEQ_AVG_REP = mean(value)) %>% 
  ungroup() %>% 
  mutate(SITE_2 = case_when(
    SITE == "Piccard" ~ "MCR",
    SITE == "VonDamm" ~ "MCR",
    TRUE ~ SITE
  )) %>% 
  group_by(SITE_2, SAMPLETYPE, VENT, Supergroup, Class, Order) %>% 
    summarise(SEQ_SUM = sum(SEQ_AVG_REP)) %>% 
  ggplot(aes(x = VENT, y = Class, fill = log(SEQ_SUM))) +
    geom_tile(stat = "identity", color = "white", width = 0.9) +
    facet_grid(Supergroup ~ SITE_2 + SAMPLETYPE , scale = "free", space = "free") +
  # scale_fill_gradient(low = "#ffffcc", high = "#bd0026", na.value = "grey50") +
  scale_fill_gradient(low = "#bdc3c7", high = "#2c3e50", na.value = "grey50") +
  theme_linedraw() +
  theme(axis.text.x = element_text(angle = 90, hjust = 0.5, vjust = 0.5),
        strip.background = element_blank(), strip.text.x = element_text(color = "black", angle = 0),
        panel.grid = element_blank(), strip.text.y = element_text(color = "black", angle = 0))
# dev.off()
```

> What is the vent signature across the different sites?

Repeat above plot, isolate ciliate, dinos, rhizaria, and stramenopiles.

```{r, fig.width=8, fig.height=10}
phyla <- c("Alveolata-Ciliophora", "Alveolata-Dinoflagellata", "Rhizaria", "Stramenopiles")

# svg("pa-ventsites-selected.svg", w=8, h=10)
insitu_asv_wClass %>% 
  filter(grepl("Vent", CLASS)) %>% 
  filter(Domain == "Eukaryota") %>% 
  filter(SAMPLETYPE == "Vent") %>% 
  filter(!is.na(Class)) %>% 
  filter(Phylum != "Stramenopiles_X" & Phylum != "Pseudofungi" & Phylum != "Chrompodellids") %>%
  filter(Supergroup != "Opisthokonta") %>% 
  mutate(Supergroup = ifelse(is.na(Supergroup), "Unknown Eukaryota", Supergroup),
         Phylum = ifelse(is.na(Phylum), "Unknown", Phylum),
         Phylum = ifelse(Phylum == "Alveolata_X", "Ellobiopsidae", Phylum),
         Supergroup = ifelse(Supergroup == "Alveolata", paste(Supergroup, Phylum, sep = "-"), Supergroup)) %>% 
  filter(Supergroup %in% phyla) %>% 
  group_by(FeatureID, Taxon, Domain, Supergroup, Phylum, Class, Order, Family, Genus, Species,
           VENT, SITE, SAMPLETYPE, YEAR, DATASET) %>% 
    summarise(SEQ_AVG_REP = mean(value)) %>% 
  ungroup() %>% 
  mutate(SITE_2 = case_when(
    SITE == "Piccard" ~ "MCR",
    SITE == "VonDamm" ~ "MCR",
    TRUE ~ SITE
  )) %>% 
  group_by(SITE_2, SAMPLETYPE, VENT, Supergroup, Class) %>% 
    summarise(SEQ_SUM = sum(SEQ_AVG_REP)) %>% 
  ggplot(aes(x = VENT, y = Class, fill = SEQ_SUM)) +
    geom_tile(stat = "identity", color = "white", width = 0.9, fill = "black") +
    facet_grid(Supergroup ~ SITE_2 + SAMPLETYPE , scale = "free", space = "free") +
  theme_linedraw() +
  theme(axis.text.x = element_text(angle = 90, hjust = 0.5, vjust = 0.5),
        strip.background = element_blank(), strip.text.x = element_text(color = "black", angle = 0),
        panel.grid = element_blank(), strip.text.y = element_text(color = "black", angle = 0))
# dev.off()
```


## Among resident ASVs, what is distribution?

```{r}
head(insitu_asv_wClass)
unique(insitu_asv_wClass$CLASS)
unique(insitu_asv_wClass$SITE_CLASS)
unique(insitu_asv_wClass$SAMPLETYPE)
```

```{r, fig.height=4, fig.width=13}
# head(insitu_asv_wClass)
insitu_asv_wClass %>% 
  # filter(SITE %in% selection) %>%
  filter(Domain == "Eukaryota") %>% 
  filter(!is.na(Supergroup)) %>% 
  # Average across replicates
    group_by(FeatureID, SAMPLENAME, VENT, Supergroup, Phylum, CLASS, SITE_CLASS) %>% 
    summarise(AVG = mean(value)) %>% 
  separate(SAMPLENAME, c("SITE", "SAMPLETYPE", "YEAR", "Sample_tmp"), remove = TRUE) %>% 
  mutate(REGION = case_when(
    SITE == "GordaRidge" ~ "Gorda Ridge",
    SITE %in% mcr ~ "Mid-Cayman Rise",
    SITE == "Axial" ~ "Axial")) %>% 
  mutate(VENTNAME = case_when(
    REGION == "Gorda Ridge" ~ VENT,
    REGION == "Mid-Cayman Rise" ~ paste(SITE, VENT, sep = " "),
    REGION == "Axial" ~ paste(VENT, YEAR, sep = " ")
  )) %>% select(-Sample_tmp) %>% 
  unite(SAMPLE, SITE, VENTNAME, sep = " ", remove = FALSE) %>% 
  # filter(CLASS == "Vent only") %>%
  group_by(Supergroup, CLASS) %>% 
    summarise(SEQ_SUM = sum(AVG),
     ASV_COUNT = n()) %>% 
  pivot_longer(cols = c(SEQ_SUM, ASV_COUNT)) %>%
  filter(name == "SEQ_SUM") %>%
  ggplot(aes(x = CLASS, y = value, fill = Supergroup)) +
    geom_hline(yintercept = 0) +
    geom_segment(aes(x = CLASS, xend = CLASS,
                     y = 0, yend = value, color = Supergroup),
                 lineend = "butt", size = 1) +
    geom_point(size = 2, shape = 19, aes(color = Supergroup)) +
    scale_fill_manual(values = c("#fa9fb5", "#c51b8a", "#67000d", "#ef3b2c", "#ffffcc", "#feb24c", "#c7e9b4", "#1d91c0", "#253494", "#9e9ac8", "#238b45", "#54278f", "#bdbdbd", "#252525")) +
    scale_color_manual(values = c("#fa9fb5", "#c51b8a", "#67000d", "#ef3b2c", "#ffffcc", "#feb24c", "#c7e9b4", "#1d91c0", "#253494", "#9e9ac8", "#238b45", "#54278f", "#bdbdbd", "#252525")) +
    theme_bw() +
    facet_grid(. ~ Supergroup, scales = "free") +
    theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1, color = "black", size = 11),
          axis.text.y = element_text(color = "black", size = 12),
        panel.spacing.x = unit(0, "lines"),panel.spacing.y = unit(0, "lines"),
        panel.border = element_blank(),
        panel.grid = element_blank(),
        strip.background.x = element_blank(),
        strip.text = element_text(size = 11),
        legend.position = "none") +
  coord_flip() +
  labs(x = "", y ="Total sequences", title = "Number of 'vent-only' sequences by Supergroup & location")
# ?scale_fill_brewer
```

```{r, fig.height=4, fig.width=13}
# head(insitu_asv_wClass)
insitu_asv_wClass %>% 
  # filter(SITE %in% selection) %>%
  filter(Domain == "Eukaryota") %>% 
  filter(!is.na(Supergroup)) %>% 
  # Average across replicates
    group_by(FeatureID, SAMPLENAME, VENT, Supergroup, Phylum, CLASS, SITE_CLASS) %>% 
    summarise(AVG = mean(value)) %>% 
  separate(SAMPLENAME, c("SITE", "SAMPLETYPE", "YEAR", "Sample_tmp"), remove = TRUE) %>% 
  mutate(REGION = case_when(
    SITE == "GordaRidge" ~ "Gorda Ridge",
    SITE %in% mcr ~ "Mid-Cayman Rise",
    SITE == "Axial" ~ "Axial")) %>% 
  mutate(VENTNAME = case_when(
    REGION == "Gorda Ridge" ~ VENT,
    REGION == "Mid-Cayman Rise" ~ paste(SITE, VENT, sep = " "),
    REGION == "Axial" ~ paste(VENT, YEAR, sep = " ")
  )) %>% select(-Sample_tmp) %>% 
  unite(SAMPLE, REGION, VENTNAME, sep = " ", remove = FALSE) %>% 
  filter(CLASS == "Vent only") %>%
  group_by(Supergroup, SITE_CLASS) %>% 
    summarise(SEQ_SUM = sum(AVG),
     ASV_COUNT = n()) %>% 
  pivot_longer(cols = c(SEQ_SUM, ASV_COUNT)) %>%
  filter(name != "SEQ_SUM") %>%
  ggplot(aes(x = SITE_CLASS, y = value, fill = Supergroup)) +
    geom_hline(yintercept = 0) +
    geom_segment(aes(x = SITE_CLASS, xend = SITE_CLASS,
                     y = 0, yend = value, color = Supergroup),
                 lineend = "butt", size = 1) +
    geom_point(size = 2, shape = 19, aes(color = Supergroup)) +
    scale_fill_manual(values = c("#fa9fb5", "#c51b8a", "#67000d", "#ef3b2c", "#ffffcc", "#feb24c", "#c7e9b4", "#1d91c0", "#253494", "#9e9ac8", "#238b45", "#54278f", "#bdbdbd", "#252525")) +
    scale_color_manual(values = c("#fa9fb5", "#c51b8a", "#67000d", "#ef3b2c", "#ffffcc", "#feb24c", "#c7e9b4", "#1d91c0", "#253494", "#9e9ac8", "#238b45", "#54278f", "#bdbdbd", "#252525")) +
    theme_bw() +
    facet_grid(. ~ Supergroup, scales = "free") +
    theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1, color = "black", size = 11),
          axis.text.y = element_text(color = "black", size = 12),
        panel.spacing.x = unit(0, "lines"),panel.spacing.y = unit(0, "lines"),
        panel.border = element_blank(),
        panel.grid = element_blank(),
        strip.background.x = element_blank(),
        strip.text = element_text(size = 11),
        legend.position = "none") +
  coord_flip() +
  labs(x = "", y ="Total ASVs", title = "Number of 'vent-only' ASVs by Supergroup & location")
# ?scale_fill_brewer
```

### Isolate vent-only (putative endemic taxa)

Here, I've isolated almost 800,000 sequences belonging to the putative endemic ASVs (vent only), totaling to 3789 ASVs. This subset includes ASVs with 10 or more sequences (a threshold to reduce noise).

```{r}
endemic <- insitu_asv_wClass %>% 
  filter(Supergroup != "Opisthokonta") %>% 
  filter(CLASS == "Vent only") %>% 
  filter(value > 9) %>% 
  filter(!is.na(Supergroup))

# Sum of putative endemic sequences and ASVs
sum(endemic$value)
length(unique(endemic$FeatureID))
```

Tile plot by Class level? CLR?
Coord flip below and add environmental data as heatmap along side?
Combine years from Axial, group by site?
Do a better compilation of taxa... additional thresholds?
```{r, fig.height=7, fig.width=19}
alv <- c("Alveolata-Ellobiopsidae", "Alveolata-Perkinsea", "Alveolata-Unknown", "Alveolata-Chrompodellids", "Alveolata-Apicomplexa")

endemic_processed <- endemic %>% 
    filter(Domain == "Eukaryota") %>% 
    filter(Supergroup != "Opisthokonta") %>% 
   mutate(Supergroup = ifelse(is.na(Supergroup), "Unknown Eukaryota", Supergroup),
         Phylum = ifelse(is.na(Phylum), "Unknown", Phylum),
         Phylum = ifelse(Phylum == "Alveolata_X", "Ellobiopsidae", Phylum),
         Supergroup = ifelse(Supergroup == "Alveolata", paste(Supergroup, Phylum, sep = "-"), Supergroup)) %>% 
  mutate(SUPERGROUP = case_when(
    Supergroup %in% alv ~ "Other Alveolata",
    Supergroup == "Eukaryota_X" ~ "Unknown Eukaryota",
    Phylum == "Cercozoa" ~ "Rhizaria-Cercozoa",
    Phylum == "Radiolaria" ~ "Rhizaria-Radiolaria",
    Phylum == "Ochrophyta" ~ "Stramenopiles-Ochrophyta",
    Phylum == "Opalozoa" ~ "Stramenopiles-Opalozoa",
    Phylum == "Sagenista" ~ "Stramenopiles-Sagenista",
    TRUE ~ Supergroup
  )) %>% 
  mutate(PHYLUM = case_when(
    Phylum == "Unknown" ~ paste(SUPERGROUP, "Other"),
    grepl("_X", Phylum) ~ paste(SUPERGROUP, "Other"),
    is.na(Phylum) ~ paste(SUPERGROUP, "Other"),
    TRUE ~ Phylum
  )) %>% 
  mutate(CLASS = case_when(
    Class == "Unknown" ~ PHYLUM,
    grepl("_X", Class) ~ PHYLUM,
    is.na(Class) ~ Phylum,
    grepl("MAST-", Class) ~ "MAST",
    TRUE ~ Class
  )) %>% 
  filter(SUPERGROUP != "Archaeplastida") %>%
  # Average across replicates
      group_by(FeatureID, SAMPLENAME, VENT, Domain, SUPERGROUP, PHYLUM, CLASS, Order, Family, Genus, Species) %>% 
        summarise(AVG = mean(value)) %>% 
      ungroup() %>% 
    filter(!is.na(SUPERGROUP)) %>% 
    # Sum to the Order taxonomic classification
    unite(SAMPLENAME_2, SAMPLENAME, VENT, sep = "_") %>% 
      group_by(SAMPLENAME_2, SUPERGROUP, PHYLUM, CLASS) %>% 
        summarise(CLASS_SUM = sum(AVG)) %>% 
    unite(CLASS, SUPERGROUP, PHYLUM, CLASS, sep = "_") %>% 
    select(CLASS, SAMPLENAME_2, CLASS_SUM) %>% 
    pivot_wider(names_from = SAMPLENAME_2, values_from = CLASS_SUM, values_fill = 0) %>% 
    column_to_rownames(var = "CLASS")
# head(endemic_processed)
```

```{r, fig.height=6, fig.width=20}
## Take wide data frame and CLR transform, pivot to wide, and plot
# svg("tileplot-endemic-bysample.svg", h = 5, w = 19)
data.frame(compositions::clr(endemic_processed)) %>% 
      rownames_to_column(var = "CLASS") %>%
      pivot_longer(cols = starts_with(all), values_to = "CLR", names_to = "SAMPLENAME_2") %>% 
      separate(SAMPLENAME_2, c("SAMPLENAME", "VENT"), sep = "_") %>% 
      separate(SAMPLENAME, c("SITE", "SAMPLETYPE", "YEAR", "Sample_tmp"), remove = TRUE) %>% 
      mutate(VENT = str_replace_all(VENT, "\\.", " ")) %>% 
      mutate(REGION = case_when(
        SITE == "GordaRidge" ~ "Gorda Ridge",
        SITE %in% mcr ~ "Mid-Cayman Rise",
        SITE == "Axial" ~ "Axial")) %>% 
      mutate(VENTNAME = case_when(
        REGION == "Gorda Ridge" ~ VENT,
        REGION == "Mid-Cayman Rise" ~ paste(SITE, VENT, sep = " "),
        REGION == "Axial" ~ VENT
        # REGION == "Axial" ~ paste(VENT, YEAR, sep = " ")
      )) %>% select(-Sample_tmp) %>% 
    unite(SAMPLE, SITE, VENTNAME, sep = " ", remove = FALSE) %>% 
    separate(CLASS, c("Supergroup", "Phylum", "Class"), sep = "_", remove = FALSE) %>%
    ggplot(aes(x = SAMPLE, y = Class, fill = CLR)) +
      geom_tile(color = "#252525") + 
      theme(legend.position = "right", 
            panel.grid.major = element_blank(),
            panel.grid.minor = element_blank(), 
            panel.border = element_blank(), 
            panel.background = element_blank(),
            axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5, color = "black",size = 8), 
            axis.text.y = element_text(color = "black", size = 8),
            strip.background = element_blank(), 
            strip.text.y = element_text(hjust = 0, vjust = 0.5, angle = 0),
            legend.title = element_blank(),
            strip.placement = "outside") + 
      labs(x = "", y = "") +
  coord_flip() +
      # scale_fill_gradient2(low = "#4575b4", mid = "white", high = "#d73027", na.value = "grey50") +
  scale_fill_steps2(
  low = "#2166ac",
  mid = "white",
  high = "#b2182b",
  midpoint = 0,
  space = "Lab",
  na.value = "#4d4d4d",
  guide = "coloursteps",
  aesthetics = "fill"
) +
      facet_grid(SITE ~ Supergroup + Phylum, space = "free", scales = "free", switch = "both")
# dev.off()
# ?scale_fill_steps2()
```

#### Repeat CLR of endemics at ASV level
```{r, fig.height=6, fig.width=20}
alv <- c("Alveolata-Ellobiopsidae", "Alveolata-Perkinsea", "Alveolata-Unknown", "Alveolata-Chrompodellids", "Alveolata-Apicomplexa")
ciliate <- c("Ciliophora")

plot_endemic_fingerprint <- function(threshold, selection){
  endemic_processed_asv <- endemic %>% 
    filter(Domain == "Eukaryota") %>% 
    filter(Supergroup != "Opisthokonta") %>% 
   mutate(Supergroup = ifelse(is.na(Supergroup), "Unknown Eukaryota", Supergroup),
         Phylum = ifelse(is.na(Phylum), "Unknown", Phylum),
         Phylum = ifelse(Phylum == "Alveolata_X", "Ellobiopsidae", Phylum),
         Supergroup = ifelse(Supergroup == "Alveolata", paste(Supergroup, Phylum, sep = "-"), Supergroup)) %>% 
  mutate(SUPERGROUP = case_when(
    Supergroup %in% alv ~ "Other Alveolata",
    Supergroup == "Eukaryota_X" ~ "Unknown Eukaryota",
    Phylum == "Cercozoa" ~ "Rhizaria-Cercozoa",
    Phylum == "Radiolaria" ~ "Rhizaria-Radiolaria",
    Phylum == "Ochrophyta" ~ "Stramenopiles-Ochrophyta",
    Phylum == "Opalozoa" ~ "Stramenopiles-Opalozoa",
    Phylum == "Sagenista" ~ "Stramenopiles-Sagenista",
    TRUE ~ Supergroup
  )) %>% 
  mutate(PHYLUM = case_when(
    Phylum == "Unknown" ~ paste(SUPERGROUP, "Other"),
    grepl("_X", Phylum) ~ paste(SUPERGROUP, "Other"),
    is.na(Phylum) ~ paste(SUPERGROUP, "Other"),
    TRUE ~ Phylum
  )) %>% 
  mutate(CLASS = case_when(
    Class == "Unknown" ~ PHYLUM,
    grepl("_X", Class) ~ PHYLUM,
    is.na(Class) ~ Phylum,
    grepl("MAST-", Class) ~ "MAST",
    TRUE ~ Class
  )) %>% 
  # Average across replicates
      group_by(FeatureID, SAMPLENAME, VENT, Domain, SUPERGROUP, PHYLUM, CLASS, Order, Family, Genus, Species) %>% 
        summarise(AVG = mean(value)) %>% 
      ungroup() %>% 
    filter(!is.na(SUPERGROUP)) %>% 
    filter(!is.na(PHYLUM)) %>% 
    filter(AVG > threshold) %>%
    filter(PHYLUM %in% selection) %>% 
    # Sum to the Order taxonomic classification
    unite(SAMPLENAME_2, SAMPLENAME, VENT, sep = "_") %>% 
      group_by(SAMPLENAME_2, SUPERGROUP, PHYLUM, FeatureID) %>% 
        summarise(SUM = sum(AVG)) %>% 
    unite(PHYLUM_ASV, PHYLUM, FeatureID, sep = "_") %>%
    select(PHYLUM_ASV, SAMPLENAME_2, SUM) %>% 
    pivot_wider(names_from = SAMPLENAME_2, values_from = SUM, values_fill = 0) %>% 
    column_to_rownames(var = "PHYLUM_ASV")
  ###
data.frame(compositions::clr(endemic_processed_asv)) %>% 
      rownames_to_column(var = "PHYLUM_ASV") %>%
      pivot_longer(cols = starts_with(all), values_to = "CLR", names_to = "SAMPLENAME_2") %>% 
      separate(SAMPLENAME_2, c("SAMPLENAME", "VENT"), sep = "_") %>% 
      separate(SAMPLENAME, c("SITE", "SAMPLETYPE", "YEAR", "Sample_tmp"), remove = TRUE) %>% 
      mutate(VENT = str_replace_all(VENT, "\\.", " ")) %>% 
      mutate(REGION = case_when(
        SITE == "GordaRidge" ~ "Gorda Ridge",
        SITE %in% mcr ~ "Mid-Cayman Rise",
        SITE == "Axial" ~ "Axial")) %>% 
      mutate(VENTNAME = case_when(
        REGION == "Gorda Ridge" ~ VENT,
        REGION == "Mid-Cayman Rise" ~ paste(SITE, VENT, sep = " "),
        REGION == "Axial" ~ VENT
        # REGION == "Axial" ~ paste(VENT, YEAR, sep = " ")
      )) %>% select(-Sample_tmp) %>% 
    unite(SAMPLE, SITE, VENTNAME, sep = " ", remove = FALSE) %>% 
    separate(PHYLUM_ASV, c("Phylum", "FeatureID"), sep = "_", remove = FALSE) %>%
    ggplot(aes(x = SAMPLE, y = FeatureID, fill = CLR)) +
      geom_tile() + 
      theme(legend.position = "right", 
            panel.grid.major = element_blank(),
            panel.grid.minor = element_blank(), 
            panel.border = element_blank(), 
            panel.background = element_blank(),
            axis.text.x = element_blank(),
            # axis.text.y = element_blank(), 
            axis.text.y = element_text(color = "black", size = 8),
            strip.background = element_blank(), 
            strip.text.y = element_text(hjust = 0, vjust = 0.5, angle = 0),
            legend.title = element_blank(),
            axis.ticks = element_blank(),
            strip.placement = "outside") + 
      labs(x = "", y = "") +
  coord_flip() +
      # scale_fill_gradient2(low = "#4575b4", mid = "white", high = "#d73027", na.value = "grey50") +
  scale_fill_steps2(
  low = "#2166ac",
  mid = "white",
  high = "#b2182b",
  midpoint = 0,
  space = "Lab",
  na.value = "#4d4d4d",
  guide = "coloursteps",
  aesthetics = "fill"
) +
      facet_grid(SITE ~ Phylum, space = "free", scales = "free", switch = "both")
}
```

```{r, fig.height=4, fig.width=20}
## Take wide data frame and CLR transform, pivot to wide, and plot
# svg("tileplot-endemic-bysample.svg", h = 6, w = 20)
plot_endemic_fingerprint(1, ciliate)
# dev.off()
```
```{r, fig.height=6, fig.width=20}
unique(endemic$Phylum)

plot_endemic_fingerprint(0, ciliate)
```


### Community composition of vent endemics

Subset dataset to create endemic dataset and a vent inclusive dataset.

```{r, fig.height=10, fig.width=16}
make_bar_relabun(endemic, all)
```
PCA analysis for resident population within each site. Reinforces what is seen in other plots, where some vent sites are more similar to one another. 
```{r, fig.width=17, fig.height=14}
plot_grid(
  make_pca(endemic, axial),
  make_pca(endemic, mcr),
  make_pca(endemic, gr),
  make_pca(endemic, all),
  ncol = 2
)
# make_pca(endemic, all)
```

## Add metadata heat map with tile plot from above
```{r}
# colnames(endemic)
rm <- c("-", "", "nd", "bd")

# x <- c("TEMP")
endemic_env <- function(x){
  endemic %>% 
  unite(SAMPLENAME_2, SAMPLENAME, VENT, sep = "_") %>% 
  separate(SAMPLENAME_2, c("SAMPLENAME", "VENT"), sep = "_") %>% 
      separate(SAMPLENAME, c("SITE", "SAMPLETYPE", "YEAR", "Sample_tmp"), remove = TRUE) %>% 
      mutate(VENT = str_replace_all(VENT, "\\.", " ")) %>% 
      mutate(REGION = case_when(
        SITE == "GordaRidge" ~ "Gorda Ridge",
        SITE %in% mcr ~ "Mid-Cayman Rise",
        SITE == "Axial" ~ "Axial")) %>% 
      mutate(VENTNAME = case_when(
        REGION == "Gorda Ridge" ~ VENT,
        REGION == "Mid-Cayman Rise" ~ paste(SITE, VENT, sep = " "),
        REGION == "Axial" ~ VENT
      )) %>% select(-Sample_tmp) %>% 
    unite(SAMPLE, SITE, VENTNAME, sep = " ", remove = FALSE) %>%  
  select(SAMPLE, SITE, VENTNAME, DEPTH:ProkConc) %>% 
  pivot_longer(cols = TEMP:ProkConc, names_to = "MEASUREMENT", values_to = "VALUE") %>% 
    na_if(., "bd") %>% na_if(., "nd") %>% na_if(., "-") %>% na_if(., "") %>% 
  filter(MEASUREMENT == x) %>%
  # filter(!(VALUE %in% rm)) %>% 
  mutate(VALUE = as.numeric(as.character(VALUE))) %>% 
    distinct() %>% 
  ggplot(aes(x = SAMPLE, y = MEASUREMENT, fill = VALUE)) +
  geom_tile(color = "black") +
  coord_flip() +
  facet_grid(SITE ~ MEASUREMENT, switch = "both", space = "free", scale = "free") + theme_linedraw() +
    theme(axis.text.y = element_blank(),
          strip.text = element_blank(),
          strip.background = element_blank(),
          axis.ticks = element_blank(),
          strip.placement = "outside",
          legend.title = element_blank(),
          legend.position = "top",
          legend.text = element_text(size = 5),
          panel.background = element_blank(),
          panel.border = element_blank(),
          panel.grid = element_blank()) +
    labs(x = "", y = "") +
    scale_fill_distiller(palette = "Reds", direction=2, na.value = "grey50")
    # viridis::scale_fill_viridis(option = "plasma")
}
##bdc3c7 → #2c3e50
# ?scale_fill_viridis
# endemic_env("TEMP")
# ?scale_fill_distiller
```

```{r, fig.width=15, fig.height=4}
# colnames(endemic)
# svg("env-heatmap.svg", w = 7, h = 4)
plot_grid(
 endemic_env("TEMP") + 
   theme(axis.text.y = element_text(color = "black"), 
         strip.text.y = element_text(color = "black"), 
         strip.placement = "outside"),
 endemic_env("PercSeawater"),
 endemic_env("pH"),
 endemic_env("Mg"),
 # endemic_env("NO3"),
 endemic_env("H2"),
 endemic_env("CH4"),
 endemic_env("H2S"),
 endemic_env("ProkConc"),
 nrow = 1,
 rel_widths = c(5,1,1,1,1,1,1,1)
)
# dev.off()
```


# HPC analysis code below

Diversity model/estimation and network analysis to be run on HPC.

```{r Load from local, eval=FALSE}
load("asv-tables-processed-18102021.RData", verbose = T)
```

# Diversity estimators (DivNet)

DivNet package - diversity estimation hypothesis testing from Amy
Willis's group. This will also characterize the uncertainty of the
richness estimate. Richness estimation is flawed because of sample depth
and processing methods.

```{r load relevant libraries for diversity estimations}
library(phyloseq); library(breakaway); library(DivNet)
library(tidyverse)
```

This code block run on HPC.

```{r create phyloseq objects for divnet}
# Select eukaryotes only and create wide format dataframe
insitu_wide <- asv_insitu_qc %>% 
  filter(Domain == "Eukaryota") %>% 
  filter(!grepl("_Plume001_", SAMPLE)) %>% #removing "near vent background", not relevant in other data sets
  select(FeatureID, Taxon, SAMPLE, value) %>%
  pivot_wider(names_from = SAMPLE, values_from = value, values_fill = 0)

# head(insitu_wide)
insitu_samples <- as.character(colnames(insitu_wide %>% select(-Taxon, -FeatureID)))
# insitu_samples
```

```{r Isolate matrix for asvs and taxa}
insitu_tax_matrix <- insitu_wide %>%
  select(FeatureID, Taxon) %>% 
  separate(Taxon, c("Domain", "Supergroup", 
                  "Phylum", "Class", "Order",
                  "Family", "Genus", "Species"), sep = ";") %>% 
  column_to_rownames(var = "FeatureID") %>% 
  as.matrix

insitu_asv_matrix <- insitu_wide %>% 
  select(-Taxon) %>% 
  column_to_rownames(var = "FeatureID") %>% 
  as.matrix

# Align row names for each matrix
rownames(insitu_tax_matrix) <- row.names(insitu_asv_matrix)
```

Reformat metadata, but change to numeric.
```{r}
## Extract relevant metadata information
# Ensure these columns are converted to numeric
env_params <- c("DEPTH", "TEMP", "pH", "PercSeawater", "Mg", "H2", "H2S", "CH4", "ProkConc")

metadata_insitu <- metadata_formatted %>%
  na_if(., "bd") %>% na_if(., "nd") %>% na_if(., "-") %>% na_if(., "") %>% 
  filter(SAMPLE %in% insitu_samples) %>% # from reformatting df above
  # select(SAMPLE, VENT, SITE, SAMPLETYPE, YEAR) %>%
  unite(SAMPLELABEL, VENT, SITE, SAMPLETYPE, YEAR, sep = "_", remove = FALSE) %>% 
  unite(TYPE_SITE, SITE, SAMPLETYPE, sep = "_", remove = FALSE) %>% 
  mutate_at(env_params,as.numeric)

rownames(metadata_insitu) <- metadata_insitu$SAMPLE
# str(metadata_insitu)
# View(metadata_insitu)
# head(metadata_insitu)
# row.names(metadata_insitu)
```


Import taxa and ASV count matrices into phyloseq objects.

```{r Phyloseq import contined}
# Import asv and tax matrices
ASV = otu_table(insitu_asv_matrix, taxa_are_rows = TRUE)
TAX = tax_table(insitu_tax_matrix)

phylo_obj <- phyloseq(ASV, TAX)

# Import metadata as sample data in phyloseq
samplenames <- sample_data(metadata_insitu)

# join as phyloseq object
physeq_insitu = merge_phyloseq(phylo_obj, samplenames)

## Check
physeq_insitu
# head(insitu_tax_matrix)
# head(metadata_insitu)

save(physeq_insitu, metadata_insitu, file = "phyloseq-objs-180122.RData") #use below for statistical analysis
```

```{r Apply divnet, eval=FALSE}
# ?divnet()
# Glom tax levels at the Order level, then perform divnet analysis
order_divnet <- divnet(tax_glom(physeq_insitu, taxrank = "Order"), base = 3)
order_divnet_label <- divnet(tax_glom(physeq_insitu, taxrank = "Order"), X = "SAMPLELABEL", base = 3)
# Vent vs plume vs background
order_divnet_TYPE <- divnet(tax_glom(physeq_insitu, taxrank = "Order"), X = "SAMPLETYPE", base = 3)
# location and vent vs plume vs background
order_divnet_TYPE_SITE <- divnet(tax_glom(physeq_insitu, taxrank = "Order"), X = "TYPE_SITE", base = 3)

save(order_divnet, order_divnet_label, order_divnet_TYPE, order_divnet_TYPE_SITE, file = "ORDER.Rdata")

###
fam_divnet <- divnet(tax_glom(physeq_insitu, taxrank = "Family"), base = 3)
fam_divnet_label <- divnet(tax_glom(physeq_insitu, taxrank = "Family"), X = "SAMPLELABEL", base = 3)
# Vent vs plume vs background
fam_divnet_TYPE <- divnet(tax_glom(physeq_insitu, taxrank = "Family"), X = "SAMPLETYPE", base = 3)
# location and vent vs plume vs background
fam_divnet_TYPE_SITE <- divnet(tax_glom(physeq_insitu, taxrank = "Family"), X = "TYPE_SITE", base = 3)
save(fam_divnet, fam_divnet_label, fam_divnet_TYPE, fam_divnet_TYPE_SITE, file = "FAMILY.Rdata")

###
gen_divnet <- divnet(tax_glom(physeq_insitu, taxrank = "Genus"), base = 3)
gen_divnet_label <- divnet(tax_glom(physeq_insitu, taxrank = "Genus"), X = "SAMPLELABEL", base = 3)
# Vent vs plume vs background
gen_divnet_TYPE <- divnet(tax_glom(physeq_insitu, taxrank = "Genus"), X = "SAMPLETYPE", base = 3)
# location and vent vs plume vs background
gen_divnet_TYPE_SITE <- divnet(tax_glom(physeq_insitu, taxrank = "Genus"), X = "TYPE_SITE", base = 3)
save(gen_divnet, gen_divnet_label, gen_divnet_TYPE, gen_divnet_TYPE_SITE, file = "GENUS.Rdata")

###
spp_divnet <- divnet(tax_glom(physeq_insitu, taxrank = "Species"), base = 3)
spp_divnet_label <- divnet(tax_glom(physeq_insitu, taxrank = "Species"), X = "SAMPLELABEL", base = 3)
# Vent vs plume vs background
spp_divnet_TYPE <- divnet(tax_glom(physeq_insitu, taxrank = "Species"), X = "SAMPLETYPE", base = 3)
# location and vent vs plume vs background
spp_divnet_TYPE_SITE <- divnet(tax_glom(physeq_insitu, taxrank = "Species"), X = "TYPE_SITE", base = 3)
save(spp_divnet, spp_divnet_label, spp_divnet_TYPE, spp_divnet_TYPE_SITE, file = "SPECIES.Rdata")
```

Above run on HPC and RData files save so we can look at various levels
of species richness.

Function to extract shannon and simpson data from each divnet output.

```{r extract shannon and simpson from divnet output}
# ?pivot_longer()
fxn_extract_divet <- function(df){
  df$shannon %>% summary %>% 
  pivot_longer(cols = starts_with("estimate"), names_to = "ESTIMATE-shannon", values_to = "Shannon") %>%
  pivot_longer(cols = starts_with("error"), names_to = "ERROR-shannon", values_to = "Shannon-error") %>%
  pivot_longer(cols = starts_with("lower"), names_to = "LOWER-shannon", values_to = "Shannon-lower") %>%
  pivot_longer(cols = starts_with("upper"), names_to = "UPPER-shannon", values_to = "Shannon-upper") %>%
  left_join(df$simpson %>% summary %>%
      pivot_longer(cols = starts_with("estimate"), names_to = "ESTIMATE-simpson", values_to = "Simpson") %>%
      pivot_longer(cols = starts_with("error"), names_to = "ERROR-simpson", values_to = "Simpson-error") %>%
      pivot_longer(cols = starts_with("lower"), names_to = "LOWER-simpson", values_to = "Simpson-lower") %>%
      pivot_longer(cols = starts_with("upper"), names_to = "UPPER-simpson", values_to = "Simpson-upper"),
    by = c("sample_names" = "sample_names")) %>%
  left_join(metadata_insitu %>% rownames_to_column(var = "sample_names")) %>%
  select(-sample_names, -ends_with("-simpson"), -ends_with("-shannon"), -starts_with("model."), -starts_with("name.")) %>%
  distinct()
}
```

Function to create plots

```{r}
plot_sampletype <- function(df){
  plot_grid(df %>%
  # ggplot(aes(x = VENT, y = Shannon)) +
  ggplot(aes(x = SAMPLETYPE, y = Shannon, group = SAMPLETYPE)) +
  # geom_errorbar(aes(ymin = `Shannon-lower`, ymax = `Shannon-upper`), color = "#525252", width = 0.2) +
    geom_point(shape = 21, color = "#525252", size = 2, aes(fill = SAMPLETYPE)) +
  # facet_grid(. ~ SITE + SAMPLETYPE + YEAR, space = "free_x", scales = "free_x") +
    geom_violin(aes(fill = SAMPLETYPE), color = "#525252", alpha = 0.5, width = 0.5, draw_quantiles = c(0.25, 0.5, 0.75)) +
    geom_jitter(shape = 21, color = "#525252", size = 2, aes(fill = SAMPLETYPE)) +
    scale_fill_manual(values = c("#ffffff", "#969696", "#252525")) +
    # scale_fill_brewer(palette = "Set2") +
  theme_linedraw() +
  theme(axis.text.y = element_text(size = 14),
        axis.text.x = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(color = "black"),
        legend.position = "none",
        axis.ticks.x = element_blank()) +
  labs(x = "", y = "Shannon"),
  df %>%
  # ggplot(aes(x = VENT, y = Simpson)) +
    ggplot(aes(x = SAMPLETYPE, y = Shannon, group = SAMPLETYPE)) +
  # geom_errorbar(aes(ymin = `Simpson-lower`, ymax = `Simpson-upper`), color = "#525252", width = 0.2) +
    # geom_point(shape = 21, color = "#525252", size = 2, aes(fill = SAMPLETYPE)) +
  # facet_grid(. ~ SITE + SAMPLETYPE + YEAR, space = "free_x", scales = "free_x") +
    geom_violin(aes(fill = SAMPLETYPE), color = "#525252", alpha = 0.5, width = 0.5, draw_quantiles = c(0.25, 0.5, 0.75)) +
    geom_jitter(shape = 21, color = "#525252", size = 2, aes(fill = SAMPLETYPE)) +
    scale_fill_manual(values = c("#ffffff", "#969696", "#252525")) +
    # scale_fill_brewer(palette = "Set2") +
  theme_linedraw() +
  theme(axis.text.x = element_text(vjust = 1, hjust = 0.5, size = 14),
        axis.text = element_text(size = 14),
        strip.background = element_blank(),
        strip.text = element_blank(),
        legend.title = element_blank(),
        legend.position = "bottom") +
  labs(x = "", y = "Simpson"),
  ncol = 1, axis = c("lrt"), align = c("vh"))
}
```

```{r, fig.height=6, fig.width=8}
load("data-input/ORDER.Rdata", verbose = T)
order_alpha_18s <- fxn_extract_divet(order_divnet)
order_alpha_label <- fxn_extract_divet(order_divnet_label)
order_alpha_TYPE <- fxn_extract_divet(order_divnet_TYPE)
order_alpha_TYPE_SITE <- fxn_extract_divet(order_divnet_TYPE_SITE)
plot_grid(plot_sampletype(order_alpha_18s),
          plot_sampletype(order_alpha_label),
          ncol = 2)

```

```{r, fig.height=6, fig.width=8}
load("data-input/FAMILY.Rdata", verbose = T)
fam_alpha_18s <- fxn_extract_divet(fam_divnet)
fam_alpha_label <- fxn_extract_divet(fam_divnet_label)
fam_alpha_TYPE <- fxn_extract_divet(fam_divnet_TYPE)
fam_alpha_TYPE_SITE <- fxn_extract_divet(fam_divnet_TYPE_SITE)
plot_grid(plot_sampletype(fam_alpha_18s),
          plot_sampletype(fam_alpha_label),
          ncol = 2)
```

```{r, fig.height=6, fig.width=3}
load("data-input/GENUS.Rdata", verbose = T)
gen_alpha_18s <- fxn_extract_divet(gen_divnet)
gen_alpha_label <- fxn_extract_divet(gen_divnet_label)
gen_alpha_TYPE <- fxn_extract_divet(gen_divnet_TYPE)
gen_alpha_TYPE_SITE <- fxn_extract_divet(gen_divnet_TYPE_SITE)
plot_grid(plot_sampletype(gen_alpha_18s),
          plot_sampletype(gen_alpha_label),
          ncol = 2)

plot_sampletype(gen_alpha_label)
```

```{r, fig.height=6, fig.width=8}
load("data-input/SPECIES.Rdata", verbose = T)
spp_alpha_18s <- fxn_extract_divet(spp_divnet)
spp_alpha_label <- fxn_extract_divet(spp_divnet_label)
spp_alpha_TYPE <- fxn_extract_divet(spp_divnet_TYPE)
spp_alpha_TYPE_SITE <- fxn_extract_divet(spp_divnet_TYPE_SITE)
plot_grid(plot_sampletype(spp_alpha_18s),
          plot_sampletype(spp_alpha_label),
          ncol = 2)

```

```{r, fig.height=6, fig.width=4}
plot_sampletype(spp_alpha_18s)
```

```{r}
testDiversity(spp_divnet_TYPE_SITE, "shannon")
```

# Network analysis

Import data frames below and format as phyloseq. Subsample to select ASVs that appear in more than 1 sample and have a total of 100 or more sequences. This leaves 2577 total ASVs across 43 samples.

> Run below commands on HPC

```{r, eval=FALSE}
load("asv-tables-processed-18102021.RData", verbose = T)
```
```{r, eval=FALSE}
library(phyloseq)
# library(SpiecEasi)
```

```{r Input to matrix for phyloseq object, eval=FALSE}
# Select eukaryotes only and create wide format dataframe
insitu_wide_nosingle <- asv_insitu_qc %>%
  filter(Domain == "Eukaryota") %>%
  filter(!grepl("_Plume001_", SAMPLE)) %>% #removing "near vent background", not relevant in other data sets
  select(FeatureID, Taxon, SAMPLE, value) %>%
  pivot_wider(names_from = SAMPLE, values_from = value, values_fill = 0) %>%
  mutate(PREVALENCE = rowSums(select_if(., is.numeric) > 0),
         SEQ_TOTAL = rowSums(select_if(., is.numeric))) %>%
  filter(PREVALENCE >= 1) %>% 
  filter(SEQ_TOTAL >= 100)
#
# ASVs had to appear in more than 1 sample and have at least 100 sequences
length(unique(insitu_wide_nosingle$FeatureID))
length(unique(asv_insitu_qc$FeatureID))

# head(insitu_wide)
insitu_samples <- as.character(colnames(insitu_wide_nosingle %>% select(-Taxon, -FeatureID)))


# make matrices for phyloseq
insitu_tax_matrix <- insitu_wide_nosingle %>%
  select(FeatureID, Taxon) %>%
  separate(Taxon, c("Domain", "Supergroup",
                  "Phylum", "Class", "Order",
                  "Family", "Genus", "Species"), sep = ";") %>%
  column_to_rownames(var = "FeatureID") %>%
  as.matrix

insitu_asv_matrix <- insitu_wide_nosingle %>%
  select(-Taxon) %>%
  column_to_rownames(var = "FeatureID") %>%
  as.matrix


# Align row names for each matrix
rownames(insitu_tax_matrix) <- row.names(insitu_asv_matrix)

metadata_insitu <- metadata %>%
  filter(SAMPLE %in% insitu_samples) %>% # from reformatting df above
  select(SAMPLE, VENT, SITE, SAMPLETYPE, YEAR) %>%
  unite(SAMPLELABEL, VENT, SITE, SAMPLETYPE, YEAR, sep = "_", remove = FALSE) %>%
  unite(TYPE_SITE, SITE, SAMPLETYPE, sep = "_", remove = FALSE)

rownames(metadata_insitu) <- metadata_insitu$SAMPLE

# Import asv and tax matrices
ASV = otu_table(insitu_asv_matrix, taxa_are_rows = TRUE)
TAX = tax_table(insitu_tax_matrix)

phylo_obj <- phyloseq(ASV, TAX)

# Import metadata as sample data in phyloseq
samplenames <- sample_data(metadata_insitu)

## Check
physeq_insitu
```

Run SPIEC-EASI with phyloseq object.
```{r, eval=FALSE}
# Run spiec easi with glasso
pargs2 <- list(rep.num = 50, seed = 10010, ncores = 10)
spec_glasso_microeuk <- spiec.easi(physeq_insitu, method = 'glasso', lambda.min.ratio=1e-2, nlambda=20,pulsar.params=pargs2)

# save(spec_glasso_microeuk, file = "spiec-easi-output-03-12-21.RData")
```

## Results from 12-8-2021
Isolate ASV-ASV pairs of interest
```{r, eval=FALSE}
# load("spiec-easi-output-03-12-21.RData", verbose = T) # almost 5GB file!

getStability(spec_glasso_microeuk) # Target == 0.05
# [1] 0.03827056
sum(getRefit(spec_glasso_microeuk))/2
# [1] 45904.5

# spec_glasso_microeuk
# Pulsar-selected refit of sparseiCov 
# Path length: 20 
# Graph dim:   2577 
# Criterion:
#   stars... sparsity 0.0138
```

Extract weighted matrix
```{r, eval=FALSE}
# se_beta <- as.matrix(symBeta(getOptBeta(spec_glasso_microeuk)))
# df_beta <- as.data.frame(se_beta)

# Extract weight information
glasso_weight  <- cov2cor(as.matrix(getOptCov(spec_glasso_microeuk)))

colnames(glasso_weight) <- rownames(glasso_weight)# <- colnames(Networ_taxa_DF_WideMat) #here i may be able to give the verticies the taxa names if i feed it the vector of names from the levels.

weighted_adj_mat <- glasso_weight*getRefit(spec_glasso_microeuk)
df_weighted <- as.data.frame(as.matrix(weighted_adj_mat))
# Assign column and row names - from original glasso output matrix data
colnames(df_weighted) <- colnames(spec_glasso_microeuk$est$data)
row.names(df_weighted) <- colnames(spec_glasso_microeuk$est$data)
```

Work with weighted dataframe
```{r, eval=FALSE}
key <- insitu_asv_wClass %>%
  select(FeatureID, Taxon, Domain:Species, CLASS, SITE_CLASS) %>% 
  distinct()
# head(key)

df_spieceasi <- df_weighted %>% 
 rownames_to_column(var = "sideA") %>% 
  pivot_longer(cols = -sideA, names_to = "sideB") %>% 
  left_join(key, by = c(sideA = "FeatureID")) %>% 
  left_join(key, by = c(sideB = "FeatureID"), suffix = c("_sideA", "_sideB")) %>% 
  distinct()

# 6640929 total interactions
## 6 million edged?
df_spieceasi_filtered <- df_spieceasi %>% 
  filter(abs(value) > 0.01) %>% 
  mutate(Interaction = case_when(
    value < 0 ~ "negative",
    value > 0 ~ "positive"
  ))
# Leaves 91,288 interactions
## Interaction type
# negative positive 
    # 3363    87925 

save(df_spieceasi_filtered, file = "filtered-spieceasi-result-08122021.RData")  
```

## Bring results locally
```{r}
load("data-input/filtered-spieceasi-result-08122021.RData", verbose = TRUE)
```
> H0: the majority of protist-protist pairs will reveal host-parasite interactions, and then predator-prey.

Questions to ask regarding the SPIEC EASI results.
- What is the overall taxonomic composition of negative and positive co-occurring ASVs?
- What percentage of putative interactions include likely parasitic protists?

Format, get summary stats from network analyss.
```{r, fig.width=18, fig.height=7}
head(df_spieceasi_filtered)

spieceasi_rm_reps <- df_spieceasi_filtered %>% 
  mutate(TMP_ASV_REP = purrr::map2_chr(sideA, sideB, ~toString(sort(c(.x, .y))))) %>%
  select(TMP_ASV_REP, value, Interaction) %>% 
    group_by(TMP_ASV_REP, value, Interaction) %>% 
    distinct() %>% 
  ungroup() %>% 
  separate(TMP_ASV_REP, c("sideA", "sideB"), sep = ", ") %>% 
  left_join((select(df_spieceasi_filtered, ends_with("sideA")) %>% distinct())) %>% 
  left_join((select(df_spieceasi_filtered, ends_with("sideB")) %>% distinct()))

head(spieceasi_rm_reps)
```
Look at highest percentage of ASV-ASV pairs by various categories.
```{r}
# Get stats
totaloccur <- dim(spieceasi_rm_reps)[1]

spieceasi_rm_reps %>% 
  unite(CLASS_joined, CLASS_sideA, CLASS_sideB, sep = "-") %>%
  unite(SITE_CLASS_joined, SITE_CLASS_sideA, SITE_CLASS_sideB, sep = "-") %>%
  unite(PHYLUM_joined, Phylum_sideA, Phylum_sideB, remove = FALSE, sep = "-") %>%
  group_by(CLASS_joined, Interaction) %>%
  summarise(COUNT = n(),
            PERC = 100*(COUNT/totaloccur)) %>% 
  arrange(desc(PERC))
```
> ASV-ASV pairs were primarily among ASVs classified as 'Vent only' - both ASVs were resident. 36%

> Secondly, 15% of the ASV pairs were from ASVs specifically found in all sample types.

> Again, mostly all positive. The highest occurence of negative interactions were between Vent only- Vent, plume, and background. 

```{r}
asv_asv <- c("Vent only-Vent only", "Vent, plume, & background-Vent, plume, & background")

# Get stats
totaloccur <- dim(spieceasi_rm_reps)[1]

spieceasi_rm_reps %>% 
  unite(CLASS_joined, CLASS_sideA, CLASS_sideB, sep = "-") %>%
  unite(SITE_CLASS_joined, SITE_CLASS_sideA, SITE_CLASS_sideB, sep = "-") %>%
  unite(PHYLUM_joined, Phylum_sideA, Phylum_sideB, remove = FALSE, sep = "-") %>%
  filter(CLASS_joined %in% asv_asv) %>% 
  group_by(CLASS_joined) %>% 
    mutate(TOTALCOUNT = n()) %>% 
  group_by(CLASS_joined, PHYLUM_joined, Interaction, TOTALCOUNT) %>%
  summarise(COUNT = n(),
            PERC = 100*(COUNT/TOTALCOUNT)) %>%
  ungroup() %>% 
  arrange(desc(PERC)) %>% 
  distinct() %>% 
  select(CLASS_joined, PHYLUM_joined, PERC, Interaction) %>% 
  pivot_wider(names_from = CLASS_joined, values_from = PERC)
```
> Within 'Vent only-Vent only' ASV pairs, almost 15% were among ciliates (ciliate-ciliate), while for the 'Vent, plume, & backgroun' ASV pairs, almost 13% were between dinoflagellates.

Isolate the ASV-ASV pairs that appeared most frequently. These include ciliates, dinoflagellates, and radiolaria within the resident and cosmopolitan (the latter includes ASVs that appears at least once ALL sample types).
```{r}
rad_dino <- c("Dinoflagellata-Radiolaria", "Radiolaria-Dinoflagellata", "Radiolaria-Radiolaria")

ciliate_ciliate <- spieceasi_rm_reps %>% 
  unite(CLASS_joined, CLASS_sideA, CLASS_sideB, sep = "-") %>%
  unite(SITE_CLASS_joined, SITE_CLASS_sideA, SITE_CLASS_sideB, sep = "-") %>%
  unite(PHYLUM_joined, Phylum_sideA, Phylum_sideB, remove = FALSE, sep = "-") %>%
  filter(CLASS_joined %in% asv_asv) %>%
  filter(PHYLUM_joined == "Ciliophora-Ciliophora")

dino_dino <- spieceasi_rm_reps %>% 
  unite(CLASS_joined, CLASS_sideA, CLASS_sideB, sep = "-") %>%
  unite(SITE_CLASS_joined, SITE_CLASS_sideA, SITE_CLASS_sideB, sep = "-") %>%
  unite(PHYLUM_joined, Phylum_sideA, Phylum_sideB, remove = FALSE, sep = "-") %>%
  filter(CLASS_joined %in% asv_asv) %>%
  filter(PHYLUM_joined == "Dinoflagellata-Dinoflagellata")

rad_dino <- spieceasi_rm_reps %>% 
  unite(CLASS_joined, CLASS_sideA, CLASS_sideB, sep = "-") %>%
  unite(SITE_CLASS_joined, SITE_CLASS_sideA, SITE_CLASS_sideB, sep = "-") %>%
  unite(PHYLUM_joined, Phylum_sideA, Phylum_sideB, remove = FALSE, sep = "-") %>%
  filter(CLASS_joined %in% asv_asv) %>%
  filter(PHYLUM_joined %in% rad_dino)
```


### Import metadata to parse putative interaction type
Importing data from Ramond et al. 2019 - https://www.seanoe.org/data/00405/51662/
```{r}
fxn <- read.delim(file = "data-input/fxntraits-ramond.csv", sep = ";")
library(fuzzyjoin)

fxn_formatted <- fxn %>%
  mutate_at(vars(Lineage), funs(str_replace_all(., pattern = "\\|", replacement = ";"))) %>%
  select(Taxon = Lineage, SizeMin, SizeMax, Cover, Shape, Spicule, Symmetry, Polarity, Colony, Motility, Chloroplast, Plast_Origin, Ingestion, Behaviour, Mutualistic_Host, starts_with("Symbion"), ends_with("_Specialisation"), Mucilage, Chemical_Signal, Nutrient_Afinity, Oxygen_Tolerance, Salinity) %>% 
  separate(Taxon, c("Domain", "Supergroup_0", "Supergroup",
                  "Phylum", "Class", "Order",
                  "Family", "Genus", "Species"), sep = ";", remove = FALSE)
# unique(fxn_formatted$Supergroup)
```

Imported curated dataset
```{r}
fxn_curated <- read.csv("data-input/curated_ramond_subset.csv")

fxn_summarised <- fxn_curated %>% 
  separate(Taxon, c("Domain", "Supergroup_0", "Supergroup",
                  "Phylum", "Class", "Order",
                  "Family", "Genus", "Species"), sep = ";", remove = FALSE) %>% 
  filter(!is.na(Order)) %>% 
  filter(!is.na(Family)) %>% 
  unite(TAXONOMY, Supergroup_0, Supergroup, Phylum, Class, Order, sep = ";") %>% 
  unite(TAXONOMY_RESOLUTION, Family, Genus, Species, sep = ";") %>% 
  group_by(TAXONOMY) %>% 
    summarize(Species_col = paste(unique(TAXONOMY_RESOLUTION), collapse = ", "),
              over_col = paste(unique(Cover), collapse = ", "),
              Shape_col = paste(unique(Shape), collapse = ", "),
              Motility_col = paste(unique(Motility), collapse = ", "),
              Ingestion_col = paste(unique(Ingestion), collapse = ", "),
              Behaviour_col = paste(unique(Behaviour), collapse = ", "),
              Mutualistic_Host_col = paste(unique(Mutualistic_Host), collapse = ", "),
              Symbiontic_col = paste(unique(Symbiontic), collapse = ", "),
              Host_Specialisation_col = paste(unique(Host_Specialisation), collapse = ", "),
              Prey_Specialisation_col = paste(unique(Prey_Specialisation), collapse = ", "),
              Oxygen_Tolerance_col = paste(unique(Oxygen_Tolerance), collapse = ", ")
              )
dim(fxn_summarised)
# View(fxn_summarised)
# write_delim(fxn_summarised, file = "functional_trait_summary.txt", delim = "\t")
```


Create managable databases to text search for functional traits among the ASV-ASV pairs of interest.
```{r}
ciliate_unique_meta <- fxn_formatted %>% 
  filter(Phylum == "Ciliophora") %>% 
  group_by(Supergroup, Phylum, Class, Order, Family, Genus) %>% 
    summarize(Species_col = str_c(Species, collapse = ", "),
              Cover_col = str_c(Cover, collapse = ", "),
              Shape_col = str_c(Shape, collapse = ", "),
              Motility_col = str_c(Motility, collapse = ", "),
              Ingestion_col = str_c(Ingestion, collapse = ", "),
              Behaviour_col = str_c(Behaviour, collapse = ", "),
              Mutualistic_Host_col = str_c(Mutualistic_Host, collapse = ", "),
              Symbiontic_col = str_c(Symbiontic, collapse = ", "),
              Host_Specialisation_col = str_c(Host_Specialisation, collapse = ", "),
              Prey_Specialisation_col = str_c(Prey_Specialisation, collapse = ", "),
              Oxygen_Tolerance_col = str_c(Oxygen_Tolerance, collapse = ", ")
              )

dino_unique_meta <- fxn_formatted %>% 
  filter(Phylum == "Myzozoa") %>% 
  group_by(Supergroup, Phylum, Class, Order, Family, Genus) %>% 
    summarize(Species_col = str_c(Species, collapse = ", "),
              Cover_col = str_c(Cover, collapse = ", "),
              Shape_col = str_c(Shape, collapse = ", "),
              Motility_col = str_c(Motility, collapse = ", "),
              Ingestion_col = str_c(Ingestion, collapse = ", "),
              Behaviour_col = str_c(Behaviour, collapse = ", "),
              Mutualistic_Host_col = str_c(Mutualistic_Host, collapse = ", "),
              Symbiontic_col = str_c(Symbiontic, collapse = ", "),
              Host_Specialisation_col = str_c(Host_Specialisation, collapse = ", "),
              Prey_Specialisation_col = str_c(Prey_Specialisation, collapse = ", "),
              Oxygen_Tolerance_col = str_c(Oxygen_Tolerance, collapse = ", ")
              )

ret_unique_meta <- fxn_formatted %>% 
  filter(Phylum == "Retaria") %>% 
  group_by(Supergroup, Phylum, Class, Order, Family, Genus) %>% 
    summarize(Species_col = str_c(Species, collapse = ", "),
              Cover_col = str_c(Cover, collapse = ", "),
              Shape_col = str_c(Shape, collapse = ", "),
              Motility_col = str_c(Motility, collapse = ", "),
              Ingestion_col = str_c(Ingestion, collapse = ", "),
              Behaviour_col = str_c(Behaviour, collapse = ", "),
              Mutualistic_Host_col = str_c(Mutualistic_Host, collapse = ", "),
              Symbiontic_col = str_c(Symbiontic, collapse = ", "),
              Host_Specialisation_col = str_c(Host_Specialisation, collapse = ", "),
              Prey_Specialisation_col = str_c(Prey_Specialisation, collapse = ", "),
              Oxygen_Tolerance_col = str_c(Oxygen_Tolerance, collapse = ", ")
              )
# dim(ret_unique_meta)
# dim(dino_unique_meta)
# dim(ciliate_unique_meta)
# View(ret_unique_meta)
# View(dino_unique_meta)
# View(ciliate_unique_meta)
```

## Bar plot summaries of Spiec easi results

Bar plot of total interactions, highlight if % is over 1-2% and what is the ASV distribution classification

Distribution of Spiec Easi output, histogram.
```{r}
# head(spieceasi_rm_reps)
hist(spieceasi_rm_reps$value)
```

Create series of barplots that show breakdown of significantly co-occurring ASVs.
```{r, fig.height=4, fig.width=9}
# dim(spieceasi_rm_reps)

total_num <- dim(spieceasi_rm_reps)[1]
total_num
asv_asv <- c("Vent only-Vent only", "Vent, plume, & background-Vent, plume, & background")

# total_num
# svg("spieceasi-bar-base.svg", w=9, h=4)
plot_grid(spieceasi_rm_reps %>% 
  unite(CLASS_joined, CLASS_sideA, CLASS_sideB, sep = "-") %>%
  # unite(SITE_CLASS_joined, SITE_CLASS_sideA, SITE_CLASS_sideB, sep = "-") %>%
  # unite(PHYLUM_joined, Phylum_sideA, Phylum_sideB, remove = FALSE, sep = "-")
  group_by(CLASS_joined) %>% 
    summarise(COUNT = n(),
              PERC = 100*(COUNT/(dim(spieceasi_rm_reps)[1]))) %>% 
  mutate(CLASSIFICATION = case_when(
    PERC >= 10 ~ CLASS_joined,
    PERC < 10 ~ "Less than 10%"
  )) %>% 
    ungroup() %>% 
  group_by(CLASSIFICATION) %>% 
    summarise(PERC_SUM = sum(PERC)) %>%
  ggplot(aes(x = 1, y = PERC_SUM, fill = CLASSIFICATION)) +
  geom_bar(stat = "identity", width = 0.2, color = "white") +
  coord_flip() +
  # scale_y_continuous(expand = c(0,0)) +
  scale_fill_manual(values = c("#FFFA78","#F58F52", "#EB6525")) +
  theme_void() +
  theme(axis.text.x = element_text(color = "black", size = 12),
        axis.line.x = element_line(color = "black"),
        axis.ticks.x = element_line(color = "black"),
        legend.title = element_blank()),
#
#
  spieceasi_rm_reps %>% 
  unite(CLASS_joined, CLASS_sideA, CLASS_sideB, sep = "-") %>%
  group_by(Interaction) %>% 
    summarise(COUNT = n(),
              PERC = 100*(COUNT/(dim(spieceasi_rm_reps)[1]))) %>% 
  ungroup() %>% 
  group_by(Interaction) %>% 
    summarise(PERC_SUM = sum(PERC)) %>%
  ggplot(aes(x = 1, y = PERC_SUM, fill = Interaction)) +
  geom_bar(stat = "identity", width = 0.2, color = "white") +
  coord_flip() +
  # scale_x_continuous(expand = c(0,0)) +
  scale_fill_manual(values = c("#5B5B8A", "#C67B33")) +
  theme_void() +
  theme(axis.text.x = element_text(color = "black", size = 12),
        axis.line.x = element_line(color = "black"),
        axis.ticks.x = element_line(color = "black"),
        legend.title = element_blank()),
#
#
  spieceasi_rm_reps %>% 
  # unite(CLASS_joined, CLASS_sideA, CLASS_sideB, sep = "-") %>%
  unite(PHYLUM_joined, Phylum_sideA, Phylum_sideB, remove = FALSE, sep = "-") %>% 
  group_by(PHYLUM_joined) %>% 
    summarise(COUNT = n(),
              PERC = 100*(COUNT/(dim(spieceasi_rm_reps)[1]))) %>% 
  mutate(CLASSIFICATION = case_when(
    PERC >= 2 ~ PHYLUM_joined,
    PERC < 2 ~ "Less than 2%"
  )) %>% 
  ungroup() %>% 
  group_by(CLASSIFICATION) %>% 
    summarise(PERC_SUM = sum(PERC)) %>% 
  ggplot(aes(x = 1, y = PERC_SUM, fill = CLASSIFICATION)) +
  geom_bar(stat = "identity", width = 0.2, color = "white") +
  coord_flip() +
  # scale_y_continuous(expand = c(0,0)) +
  scale_fill_manual(values = c("#6FA9E4", "#A4723F","#577383", "#F8F5E5", "#D1AB8A", "#FBDB9A","#5C5E5E", "#A12531", "#59372B")) +
  theme_void() +
  theme(axis.text.x = element_text(color = "black", size = 12),
        axis.line.x = element_line(color = "black"),
        axis.ticks.x = element_line(color = "black"),
        legend.title = element_blank()),
#
  spieceasi_rm_reps %>% 
  unite(CLASS_joined, CLASS_sideA, CLASS_sideB, sep = "-") %>%
  filter(CLASS_joined %in% asv_asv) %>% 
  group_by(CLASS_joined) %>% 
    mutate(TOTAL_COUNT = n()) %>% 
  ungroup() %>% 
  unite(PHYLUM_joined, Phylum_sideA, Phylum_sideB, remove = FALSE, sep = "-") %>% 
  group_by(PHYLUM_joined, CLASS_joined) %>% 
    summarise(COUNT = n(),
              PERC = 100*(COUNT/TOTAL_COUNT)) %>% 
  mutate(CLASSIFICATION = case_when(
    PERC >= 10 ~ PHYLUM_joined,
    PERC < 10 ~ "Less than 2%"
  )) %>% 
  group_by(CLASS_joined, CLASSIFICATION) %>% 
    summarise(PERC_SUM = sum(PERC)) %>% 
  ggplot(aes(x = CLASS_joined, y = PERC_SUM, fill = CLASSIFICATION)) +
  geom_bar(stat = "identity", position = "fill", width = 0.6, color = "white") +
  coord_flip() +
  # scale_x_continuous(expand = c(0,0)) +
  scale_fill_manual(values = c("#B24236","#fed976","#5A7356")) +
  theme_void() +
  theme(axis.text.x = element_text(color = "black", size = 12),
        axis.line.x = element_line(color = "black"),
        axis.ticks.x = element_line(color = "black"),
        legend.title = element_blank()),
  ncol = 1, align = c("vh"), axis = c("lr"))
# dev.off()
# ?plot_grid()
```

Because dinos, ciliates, and radiolaria were the most common. How many ASV-ASV pairs were they a part of?
```{r}
highlighted <- c("Ciliophora-Ciliophora", "Dinoflagellata-Dinoflagellata", "Radiolaria-Radiolaria")
connect_list <- c("Cercozoa", "Ciliophora", "Dinoflagellata", "Haptophyta", "Opalozoa", "Radiolaria")

dino_cili_rad <- spieceasi_rm_reps %>% 
  unite(CLASS_joined, CLASS_sideA, CLASS_sideB, sep = "-") %>%
  unite(PHYLUM_joined, Phylum_sideA, Phylum_sideB, remove = FALSE, sep = "-") %>%
  filter(grepl("Ciliophora", PHYLUM_joined)) %>% 
  mutate(CONNECT = case_when(
    Phylum_sideA != "Ciliophora" ~ Phylum_sideA,
    Phylum_sideB != "Ciliophora" ~ Phylum_sideB,
    PHYLUM_joined == "Ciliophora-Ciliophora" ~ "Ciliophora"
  )) %>% 
  group_by(CONNECT) %>% 
  summarise(COUNT = n()) %>% 
  add_column(PRIMARY = "Ciliophora") %>% 
    rbind(spieceasi_rm_reps %>% 
    unite(CLASS_joined, CLASS_sideA, CLASS_sideB, sep = "-") %>%
    unite(PHYLUM_joined, Phylum_sideA, Phylum_sideB, remove = FALSE, sep = "-") %>%
    filter(grepl("Dinoflagellata", PHYLUM_joined)) %>% 
    mutate(CONNECT = case_when(
      Phylum_sideA != "Dinoflagellata" ~ Phylum_sideA,
      Phylum_sideB != "Dinoflagellata" ~ Phylum_sideB,
      PHYLUM_joined == "Dinoflagellata-Dinoflagellata" ~ "Dinoflagellata"
    )) %>% 
    group_by(CONNECT) %>% 
    summarise(COUNT = n()) %>% 
      add_column(PRIMARY = "Dinoflagellata")) %>%
  rbind(spieceasi_rm_reps %>% 
  unite(CLASS_joined, CLASS_sideA, CLASS_sideB, sep = "-") %>%
  unite(PHYLUM_joined, Phylum_sideA, Phylum_sideB, remove = FALSE, sep = "-") %>%
  filter(grepl("Radiolaria", PHYLUM_joined)) %>% 
  mutate(CONNECT = case_when(
    Phylum_sideA != "Radiolaria" ~ Phylum_sideA,
    Phylum_sideB != "Radiolaria" ~ Phylum_sideB,
    PHYLUM_joined == "Radiolaria-Radiolaria" ~ "Radiolaria"
  )) %>% 
  group_by(CONNECT) %>% 
  summarise(COUNT = n()) %>% 
  add_column(PRIMARY = "Radiolaria")) %>% 
  unite(TMP, CONNECT, PRIMARY, sep = "-", remove = FALSE) %>% 
  mutate(CAT_BUBBLE = case_when(
    TMP %in% highlighted ~ PRIMARY,
    CONNECT %in% connect_list ~ "NOTED"
  )) %>% select(-TMP)
# unique(dino_cili_rad$CAT_BUBBLE)
# unique(dino_cili_rad$PRIMARY)
# length(unique(dino_cili_rad$CONNECT))
# dim(dino_cili_rad)
# head(dino_cili_rad)
```

```{r, fig.height=2, fig.width=8}

ggplot(dino_cili_rad, aes(x = PRIMARY, y = CONNECT)) +
  geom_point(aes(size = COUNT, fill = CAT_BUBBLE), 
             color = "black", shape = 21, stroke = 1) +
  coord_flip() +
  theme_linedraw() +
  theme(axis.title = element_blank(),
        axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1),
        panel.grid = element_blank(),
        axis.ticks = element_blank(),
        legend.title = element_blank()) +
  scale_fill_manual(values = c("#6FA9E4", "#F8F5E5", "#e31a1c", "#6a51a3", "#000000"))

# scale_fill_manual(values = c("#6FA9E4", "#A4723F","#577383", "#F8F5E5", "#D1AB8A", "#FBDB9A","#5C5E5E", "#A12531", "#59372B"))

# scale_fill_manual(values = c("#B24236","#fed976","#5A7356"))
```

### Ciliate-ciliate
Isolate associations with ciliates
```{r, fig.height=15, fig.width=15}
# head(ciliate_ciliate)
# table(ciliate_ciliate$SITE_CLASS_joined)
# table(ciliate_ciliate)
ciliate_ciliate %>% 
  filter(CLASS_joined == "Vent only-Vent only") %>% 
  # filter(!is.na(Phylum_sideA) | !is.na(Phylum_sideB)) %>%
  ggplot(aes(x = Taxon_sideA, y = Taxon_sideB, fill = value)) +
  geom_tile(color = "black") +
  scale_fill_gradient(low = "#bfd3e6", high = "#810f7c") +
  theme_linedraw() +
  facet_grid(Class_sideB +  Order_sideB ~ Class_sideA + Order_sideA, space = "free", scales = "free") +
  theme(axis.text = element_blank(),
      panel.grid.minor = element_blank(),
      strip.text.x = element_text(angle = 90, hjust = 0, vjust = 0.5),
      strip.text.y = element_text(angle = 0, hjust = 0, vjust = 0.5),
      strip.text = element_text(color = "black", face = "bold"),
      strip.background = element_blank())
```
Isolate functional traits from these groups of ciliates
```{r}
order <- c("Plagiopylea", "Nassophorea", "Litostomatea")
family <- c("Scuticociliatia", "Euplotia", "Suctoria", "Peritrichia")
genus <- c("Strombidiidae")
```


Tile plot of vent-only (resident) ciliate-ciliate ASV co-occurrences.
```{r, fig.height=8, fig.width=10}
# head(ciliate_ciliate)
# hist(ciliate_ciliate$value)

ciliate_ciliate %>% 
       filter(value >= 0.4) %>% 
       select(starts_with(c("Class_", "Order_")), PHYLUM_joined) %>% 
       select(-CLASS_joined) %>% 
       # select(Taxon_sideA, Taxon_sideB, PHYLUM_joined) %>% 
       unite(SIDEA, Class_sideA, Order_sideA, sep = ";", remove = FALSE) %>%
        unite(SIDEB, Class_sideB, Order_sideB, sep = ";", remove = FALSE) %>% 
      group_by(SIDEA, SIDEB, Class_sideA, Class_sideB) %>% 
      summarise(COUNT = n()) %>% 
  ggplot(aes(x = SIDEA, y = SIDEB, fill = COUNT)) +
    geom_tile(color = "black") +
  scale_fill_gradient(low = "#bfd3e6", high = "#810f7c") +
  theme_linedraw() +
  facet_grid(Class_sideB ~ Class_sideA, space = "free", scales = "free") +
  theme(panel.grid.minor = element_blank(),
      strip.text.x = element_text(angle = 90, hjust = 0, vjust = 0.5),
      strip.text.y = element_text(angle = 0, hjust = 0, vjust = 0.5),
      strip.text = element_text(color = "black", face = "bold"),
      strip.background = element_blank(),
      axis.title = element_blank(),
      axis.text.x = element_text(angle = 90, hjust = 0, vjust = 0.5))
      
```



### Dinoflagellate
```{r, fig.height=7, fig.width=8}
dino_dino %>% 
  filter(CLASS_joined == "Vent only-Vent only") %>% 
  # filter(!is.na(Phylum_sideA) | !is.na(Phylum_sideB)) %>%
  ggplot(aes(x = Taxon_sideA, y = Taxon_sideB, fill = value)) +
  geom_tile(color = "black") +
  scale_fill_gradient(low = "#bfd3e6", high = "#810f7c") +
  theme_linedraw() +
  facet_grid(Class_sideB +  Order_sideB ~ Class_sideA + Order_sideA, space = "free", scales = "free") +
  theme(axis.text = element_blank(),
      panel.grid.minor = element_blank(),
      strip.text.x = element_text(angle = 90, hjust = 0, vjust = 0.5),
      strip.text.y = element_text(angle = 0, hjust = 0, vjust = 0.5),
      strip.text = element_text(color = "black", face = "bold"),
      strip.background = element_blank())
```
Dinos and others
Get stats on Ciliate-other ASV pairs for a table
```{r}

```

### Retaria and dino

```{r, fig.height=12, fig.width=12}
rad_dino %>% 
  filter(CLASS_joined == "Vent only-Vent only") %>% 
  # filter(!is.na(Phylum_sideA) | !is.na(Phylum_sideB)) %>%
  ggplot(aes(x = Taxon_sideA, y = Taxon_sideB, fill = value)) +
  geom_tile(color = "black") +
  scale_fill_gradient(low = "#bfd3e6", high = "#810f7c") +
  theme_linedraw() +
  facet_grid(Class_sideB +  Order_sideB ~ Class_sideA + Order_sideA, space = "free", scales = "free") +
  theme(axis.text = element_blank(),
      panel.grid.minor = element_blank(),
      strip.text.x = element_text(angle = 90, hjust = 0, vjust = 0.5),
      strip.text.y = element_text(angle = 0, hjust = 0, vjust = 0.5),
      strip.text = element_text(color = "black", face = "bold"),
      strip.background = element_blank())
```
Retaria and others
Get stats on Ciliate-other ASV pairs for a table
```{r}
head(spieceasi_rm_reps)
```


### Isolate list of ASV interactions of interest
```{r}
# unique(df_spieceasi_filtered$Class_sideA)

# rm <- c(NA, "Metazoa", "")

# Filter to remove ASVs not classified to the Phylum level at least
# compare_inter <- df_spieceasi_filtered %>% 
#   filter(!(Phylum_sideA %in% rm | Phylum_sideB %in% rm)) %>% 
#   select()
# 
# compare_inter <- select(df_spieceasi_filtered, FeatureID = sideA, Taxon = Taxon_sideA) %>% 
#   rbind(select(df_spieceasi_filtered, FeatureID = sideB, Taxon = Taxon_sideB)) %>% 
#   distinct()
# 
# compare_fxn <- fxn_formatted

# write_delim(compare_inter, file = "interactions_unique_IDs.txt", delim = "\t")
# write_delim(fxn_formatted, file = "ramond_IDs.txt", delim = "\t")
```


Create tile plot with highly significantly co-occurring ASVs - helping to narrow down what is discussed in text.
```{r}
head(spieceasi_rm_reps)
```
```{r}
sig_0.4 <- spieceasi_rm_reps %>% 
  filter(abs(value) >= 0.4) %>%
  unite(PHYLUM_joined, Phylum_sideA, Phylum_sideB, remove = FALSE, sep = "-")
dim(sig_0.4)
# head(sig_0.4)
# write_delim(sig_0.4, file = "significant_interactions_by04.txt", delim = "\t")

# View(select(sig_0.4, starts_with("Taxon")) %>% 
       # distinct())
```

Function to isolate highest total number of interactions
```{r}
check_tax_interact <- function(tax){
  tmp <- spieceasi_rm_reps %>% 
    filter(abs(value) >= 0.4) %>%
    unite(PHYLUM_joined, Phylum_sideA, Phylum_sideB, remove = FALSE, sep = "-") %>% 
    filter(grepl(tax, PHYLUM_joined)) %>% 
    select(starts_with("Taxon"), PHYLUM_joined) %>% 
    # filter(grepl(compare1, PHYLUM_joined)) %>% 
    distinct()
  View(tmp)
  table(sort(tmp$PHYLUM_joined))
  }

# check_tax_interact("Ciliophora")

```


```{r, fig.height=10, fig.width=13}

# spieceasi_rm_reps %>% 
#   filter(abs(value) >= 0.4) %>% 
#   unite(PHYLUM_joined, Phylum_sideA, Phylum_sideB, remove = FALSE, sep = "-") %>%
#   # filter(grepl("Ciliophora", PHYLUM_joined)) %>% 
#   select(starts_with(c("Phylum_", "Class_", "Order_")), PHYLUM_joined) %>% 
#        # select(Taxon_sideA, Taxon_sideB, PHYLUM_joined) %>% 
#   unite(SIDEA, Class_sideA, Order_sideA, sep = ";", remove = FALSE) %>%
#   unite(SIDEB, Class_sideB, Order_sideB, sep = ";", remove = FALSE) %>% 
#     group_by(SIDEA, SIDEB, Phylum_sideA, Phylum_sideB) %>% 
#       summarise(COUNT = n()) %>% 
#   ggplot(aes(x = SIDEA, y = SIDEB, fill = COUNT)) +
#     geom_tile(color = "black") +
#   scale_fill_gradient(low = "#bfd3e6", high = "#810f7c") +
#   theme_linedraw() +
#   facet_grid(Phylum_sideB ~ Phylum_sideA, space = "free", scales = "free") +
#   theme(panel.grid.minor = element_blank(),
#       strip.text.x = element_text(angle = 90, hjust = 0, vjust = 0.5),
#       strip.text.y = element_text(angle = 0, hjust = 0, vjust = 0.5),
#       strip.text = element_text(color = "black", face = "bold"),
#       strip.background = element_blank(),
#       axis.title = element_blank(),
#       axis.text.x = element_text(angle = 90, hjust = 0, vjust = 0.5))
```



# Test influence of abiotic/biotic factors on community composition.



## Import and format dataset
Import phyloseq-saved RObjects.
```{r}
# library(tidyverse); library(phyloseq); library(compositions); library(vegan)
# load("asv-tables-processed-18102021.RData", verbose = TRUE)
load("phyloseq-objs-180122.RData", verbose = TRUE)
```


Isolate ASV count table using phyloseq commands and perform a CLR transformation.
```{r}
# Extract phyloseq object & CLR transform
count_table_clr <- compositions::clr(otu_table(physeq_insitu))
```

Check all samples considered and estimate distance matrix (Euclidean)
```{r}
# transpose and make dist matrix
sample_data(physeq_insitu)

# CLR-transformed data, transpose and create euclidean dist matrix
dist_euc <- vegdist(t(count_table_clr), method = "euclidean")
# ?vegdist()
```

### Test for homogeneity of variance with betadisper
```{r}
# ?betadisper()
beta_SITE <- betadisper(dist_euc, sample_data(physeq_insitu)$SITE)
beta_SAMPLETYPE <- betadisper(dist_euc, sample_data(physeq_insitu)$SAMPLETYPE)
beta_TYPE_SITE <- betadisper(dist_euc, sample_data(physeq_insitu)$TYPE_SITE)
```

SITE
```{r}
anova(beta_SITE)
TukeyHSD(beta_SITE)
plot(beta_SITE)
```
Sampletype
```{r}
anova(beta_SAMPLETYPE)
TukeyHSD(beta_SAMPLETYPE)
plot(beta_SAMPLETYPE)
```
Type site - too much resolution? This isn't really how I'm setting up the comparison groups anyway, so its fine.

```{r}
anova(beta_TYPE_SITE)
TukeyHSD(beta_TYPE_SITE)
plot(beta_TYPE_SITE)
```

Permutest outputs
```{r}
permutest(beta_SITE, pairwise = TRUE, permutations = 999)
```


```{r}
permutest(beta_SAMPLETYPE, pairwise = TRUE, permutations = 999)
```

```{r}
# permutest(beta_TYPE_SITE, pairwise = TRUE, permutations = 999)
```

Summary of disperson tests: Site and sample type has an influence on the disperson of all samples, but it is to be expected. For now, will perform analysis with all samples. 

### Adonis
Use ```adonis()``` to examine categorical significance
```{r}
set.seed(1212)

adonis(dist_euc ~ sample_data(physeq_insitu)$SITE, permutations = 999) # Categorical with vent site
```
> Influence of vent site on sample communities is significant.

```{r}
adonis(dist_euc ~ sample_data(physeq_insitu)$SITE + sample_data(physeq_insitu)$SAMPLETYPE, permutations = 999) # Categorical with vent site
```


## Assign parameters for testing

Isolate numerics for each parameter to be tested below.

```{r}
# names(sample_data(physeq_insitu))
# Numerics - geochem
temp <- sample_data(physeq_insitu)$TEMP
ph <- sample_data(physeq_insitu)$pH
persea <- sample_data(physeq_insitu)$PercSeawater
mg <- sample_data(physeq_insitu)$Mg
h2 <- sample_data(physeq_insitu)$H2
h2s <- sample_data(physeq_insitu)$H2S
ch4 <- sample_data(physeq_insitu)$CH4
prok <- sample_data(physeq_insitu)$ProkConc

# categorical (logic)
sampletype <- sample_data(physeq_insitu)$SAMPLETYPE
site <- sample_data(physeq_insitu)$SITE
```

## Distance-based redundancy analysis

Legendre and Anderson (1999), RDA that looks for linear relationships on dissimilarity matrices. Constrained ordination with non-Euclidean distance measures. (Legendre P, Anderson MJ (1999) Distance-based redundancy analysis: testing multispecies responses in multifactorial ecological experiments. Ecol Monogr. 69:1–24.)

Below ```dbrda()``` statement inputs all samples as a distance matrix (Euclidean), with the RH side variables as numeric data following the ~. 

```{r}
# ?dbrda()
vent_microeuk_rda <- dbrda(dist_euc ~ temp + ph + persea + mg + h2 + h2s + ch4 + prok, dist = 'euc', na.action = na.exclude)

# na.exclude: includes NAs for scores of missing observations (use this)
# na.omit: only non-missing sites scores are shown

# Print anova() output with significance values
anova(vent_microeuk_rda) #overall significance
## The model is significant
```


```{r}
# Axes; set x and y equal to variance of axes.
xy <- anova(vent_microeuk_rda, by = 'axis')
x <- round(xy$Variance[1], 1); y <- round(xy$Variance[2], 1)

# by terms
# anova(vent_microeuk_rda, by = 'terms')

# by margin
anova(vent_microeuk_rda, by = 'margin')
```

> Output shows temperature and methane as significant environmental parameters.

Samples visualized with respect to biplot direction of environmental parameters.
```{r, fig.width=6, fig.height=6}
plot(vent_microeuk_rda, xlab = x, ylab = y)
# plot(vent_microeuk_rda, type = 'points', xlab = x, ylab = y)
```
### Account for differences in sample type and site

Try with condition set to sample type. QUESTION: Null for the below... if we account for differences in Condition X... are the parameters still significant? Below results suggest that when we account for vent site or sample type, the parameters are not significant. 

```{r}
vent_microeuk_rda_sampletype <- dbrda(dist_euc ~ temp + ph + persea + mg + h2 + h2s + ch4 + prok + Condition(sampletype), dist = 'euc', na.action = na.exclude)

anova(vent_microeuk_rda_sampletype, by = 'margin')
```

Try with condition set to site.
```{r}
vent_microeuk_rda_site <- dbrda(dist_euc ~ temp + ph + persea + mg + h2 + h2s + ch4 + prok + Condition(site), dist = 'euc', na.action = na.exclude)

anova(vent_microeuk_rda_site, by = 'margin')
```


## Functions for DBRDA analysis

```{r}
subset_df <- subset_samples(physeq_insitu, SITE == "Axial")
# head
```


```{r}
dbrda_subset <- function(subset_df){
  # category <- enquo(category)
  # subset_df <- subset_samples(physeq_insitu, category %in% selection)
  clr_out <- compositions::clr(otu_table(subset_df))
  euc_out <- vegdist(t(clr_out), method = "euclidean")
  # Params
  temp <- sample_data(subset_df)$TEMP
  ph <- sample_data(subset_df)$pH
  persea <- sample_data(subset_df)$PercSeawater
  mg <- sample_data(subset_df)$Mg
  h2 <- sample_data(subset_df)$H2
  h2s <- sample_data(subset_df)$H2S
  ch4 <- sample_data(subset_df)$CH4
  prok <- sample_data(subset_df)$ProkConc
  # DBRDA
  rda_out <- dbrda(euc_out ~ temp + ph + persea + mg + h2 + h2s + ch4 + prok, dist = 'euc', na.action = na.exclude)
  # Overall significance
  # anova(rda_out)
  # Print results by margin
  anova(rda_out, by = 'margin')
  # Plot
  # xy <- anova(rda_out, by = 'axis')
  # x <- round(xy$Variance[1], 1); y <- round(xy$Variance[2], 1)
  # plot(rda_out, xlab = x, ylab = y)
}
# dbrda_subset(physeq_insitu)
```

Check sample data categories fpr further testing.
```{r}
sample_data(physeq_insitu)
```


## Stats for vent sites
```{r}
dbrda_subset(subset_samples(physeq_insitu, SAMPLETYPE == "Vent"))
```

Temperature, methane, and then H2.

## Stats for Axial
Subset to only vent sites - check significance of year.
Subset to all sites - check significant of year again.
```{r}
AXIAL <- subset_samples(physeq_insitu, SITE == "Axial")
# AXIAL <- subset_samples(physeq_insitu, (SITE == "Axial" & SAMPLETYPE == "Vent"))
clr_out <- compositions::clr(otu_table(AXIAL))
euc_out <- vegdist(t(clr_out), method = "euclidean")
  
adonis(euc_out ~ sample_data(AXIAL)$YEAR, permutations = 999) 
```

Year was not significant at Axial. Although, year to year sampling was not consistent.

## Stats for MCR
Subset to vent sites only - check significance of Piccard vs. VD. Geochem?
```{r}
MCR <- subset_samples(physeq_insitu, (SITE == "Piccard" | SITE == "VonDamm"))
# MCR <- subset_samples(physeq_insitu, (SITE == "Axial" & SAMPLETYPE == "Vent"))
clr_out <- compositions::clr(otu_table(MCR))
euc_out <- vegdist(t(clr_out), method = "euclidean")
# sample_data(MCR)  

adonis(euc_out ~ sample_data(MCR)$SITE, permutations = 999) 
```

Site was not significant factor within the 2 Mid-Cayman Rise vent sites.

## Isolate resident only

Check significance of all environmental parameters with the resident population.


Isolate vent-only ASVs
```{r}
tmp <- filter(insitu_asv_wClass, CLASS == "Vent only")
list_of_resident <- as.character(unique(tmp$FeatureID))
length(list_of_resident)

res_vent <- prune_taxa(list_of_resident, physeq_insitu)
```

```{r}
dbrda_subset(subset_samples(res_vent, SAMPLETYPE == "Vent"))
```

## Isolate individual taxa
```{r}
head(tax_table(physeq_insitu))
```
Ciliates
```{r}
dbrda_subset(subset_taxa(physeq_insitu, Phylum == "Ciliophora"))
```
Stramenopiles, MAST mostly.
```{r}
dbrda_subset(subset_taxa(physeq_insitu, Phylum == "Sagenista"))
```
Radiolaria
```{r}
dbrda_subset(subset_taxa(physeq_insitu, Phylum == "Radiolaria"))
```


## Visualize samples via temperature & prokaryote cell concentration

Plot sequence relative abundance by temperature and prokaryote
concentration. These two parameters were chosen because I have the most metadata from them. If a sample was not countable or had no temperature
record, it was removed.

### Plot as a factor of relative sequence abundance

```{r, fig.height=15, fig.width=20, tidy=FALSE}
# asv_insitu_qc %>% 
#     # filter(SITE %in% selection) %>% 
#     filter(!is.na(TEMP)) %>% 
#     filter(!is.na(ProkConc)) %>% 
#     filter(Domain == "Eukaryota") %>%
#     mutate(Supergroup = ifelse(is.na(Supergroup), "Unknown Eukaryota", Supergroup),
#            Phylum = ifelse(is.na(Phylum), "Unknown", Phylum),
#            Phylum = ifelse(Phylum == "Alveolata_X", "Ellobiopsidae", Phylum),
#            Supergroup = ifelse(Supergroup == "Alveolata", paste(Supergroup, Phylum, sep = "-"), Supergroup)) %>% 
#     group_by(FeatureID, Taxon, Domain, Supergroup, Phylum, Class, Order, Family, Genus, Species,
#              VENT, SITE, SAMPLETYPE, YEAR, DATASET, TEMP, ProkConc) %>% 
#     summarise(SEQ_AVG_REP = mean(value)) %>% 
#     ungroup() %>% 
#     unite(SupergroupPhylum, Supergroup, Phylum, sep = "-") %>% 
#     group_by(SITE, SAMPLETYPE, VENT, YEAR, SupergroupPhylum, TEMP, ProkConc) %>% 
#       summarise(SEQ_SUM = sum(SEQ_AVG_REP)) %>% 
#     ggplot(aes(x = ProkConc, y = as.numeric(TEMP), fill = SITE, shape = SAMPLETYPE)) +
#       geom_point(color = "black", aes(size = SEQ_SUM)) +
#       scale_size_continuous(range = c(4,9)) +
#       scale_shape_manual(values = c(21, 23, 24)) +
#       scale_x_log10() +
#       facet_wrap(SupergroupPhylum ~., scale = "free") +
#       theme_linedraw() +
#       theme(axis.text = element_text(color = "black", size = 12),
#           strip.background = element_blank(), strip.text = element_text(color = "black"),
#           legend.position = "right") +
#     # scale_y_continuous(expand = c(0,0)) +
#     scale_fill_manual(values = c("#feb24c", "#addd8e", "#de2d26", "#1c9099")) +
#     guides(fill = guide_legend(override.aes = list(shape = c(21))),
#            shape = guide_legend(override.aes = list(fill = "black"))) +
#     labs(x = bquote("Cells "~mL^-1 ~hr^-1), y = "Temperature (C)")
```

### Plot as a factor of CLR transformed data

Repeat above plot, but with CLR transformed data.

```{r, fig.height=15, fig.width=20, tidy=FALSE}
# df_wide_tmp <- asv_insitu_qc %>% 
#     filter(!is.na(TEMP)) %>% 
#     filter(!is.na(ProkConc)) %>% 
#     filter(Domain == "Eukaryota") %>% 
#     filter(value > 0) %>% 
#   # Average across replicates
#       group_by(FeatureID, SAMPLENAME, VENT, Domain, Supergroup, Phylum, Class, Order, Family, Genus, Species, TEMP, ProkConc) %>% 
#         summarise(AVG = mean(value)) %>% 
#       ungroup() %>% 
#     # Sum to the Order taxonomic classification
#     unite(SAMPLENAME_2, SAMPLENAME, VENT, TEMP, ProkConc, sep = "_") %>% 
#     unite(TAX, FeatureID, Supergroup, Phylum, sep = " ") %>% 
#     select(TAX, SAMPLENAME_2, AVG) %>% 
#     pivot_wider(names_from = SAMPLENAME_2, values_from = AVG, values_fill = 0) %>% 
#     column_to_rownames(var = "TAX")
# 
#   ## Take wide data frame and CLR transform, pivot to wide, and plot
#   clr_long_df <- data.frame(compositions::clr(df_wide_tmp)) %>% 
#       rownames_to_column(var = "TAX") %>%
#       pivot_longer(cols = starts_with(all), values_to = "CLR", names_to = "SAMPLENAME_2") %>% 
#       separate(SAMPLENAME_2, c("SAMPLENAME", "VENT", "TEMP", "ProkConc"), sep = "_") %>% 
#       separate(SAMPLENAME, c("SITE", "SAMPLETYPE", "YEAR", "Sample_tmp"), remove = TRUE) %>% 
#       mutate(VENT = str_replace_all(VENT, "\\.", " ")) %>% 
#       mutate(REGION = case_when(
#         SITE == "GordaRidge" ~ "Gorda Ridge",
#         SITE %in% mcr ~ "Mid-Cayman Rise",
#         SITE == "Axial" ~ "Axial")) %>% 
#       mutate(VENTNAME = case_when(
#         REGION == "Gorda Ridge" ~ VENT,
#         REGION == "Mid-Cayman Rise" ~ paste(SITE, VENT, sep = " "),
#         REGION == "Axial" ~ paste(VENT, YEAR, sep = " ")
#       )) %>% select(-Sample_tmp) %>% 
#     unite(SAMPLE, REGION, VENTNAME, sep = " ", remove = FALSE) %>% 
#     separate(TAX, c("ASVid","Supergroup", "Phylum"), sep = " ", remove = TRUE) %>% 
#     unite(SupergroupPhylum, Supergroup, Phylum, sep = "-")
# 
#   # head(clr_long_df)
#   ## Plot
#   clr_long_df %>% 
#     filter(SAMPLETYPE == "Vent") %>% 
#     ggplot(aes(x = as.numeric(ProkConc), y = as.numeric(TEMP), fill = CLR, shape = REGION)) +
#       geom_point(color = "black", size = 3, aes(fill = CLR, shape = REGION)) +
#       scale_fill_gradient2(low = "#4575b4", mid = "white", high = "#d73027", na.value = "grey50") +
#       scale_shape_manual(values = c(21, 23, 24, 25)) +
#       scale_x_log10() +
#       facet_wrap(SupergroupPhylum ~ ., scale = "free") +
#       theme_linedraw() +
#       theme(axis.text = element_text(color = "black", size = 12),
#           strip.background = element_blank(), strip.text = element_text(color = "black"),
#           legend.position = "right") +
#     guides(fill = guide_legend(override.aes = list(shape = c(21))),
#            shape = guide_legend(override.aes = list(fill = "black"))) +
#     labs(x = bquote("Cells "~mL^-1 ~hr^-1), y = "Temperature (C)")
```


# Session Info
```{r}
sessionInfo()
```

